<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Matt work summary - Extra tips</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title">Extra tips</h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Matt work summary</a> 
    </div>
      </div>
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">Home</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./whole_muscle_scrnaseq.html" class="sidebar-item-text sidebar-link">Whole muscle single nuclear</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./myob_single_nuc.html" class="sidebar-item-text sidebar-link">Myoblast single nuclear</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./atac_seq.html" class="sidebar-item-text sidebar-link">ATACseq</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./conda_envs.html" class="sidebar-item-text sidebar-link">Conda environments</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./extra_tips.html" class="sidebar-item-text sidebar-link active">Extra tips</a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">Single cell</a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./single_cell.html" class="sidebar-item-text sidebar-link">Overview</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pseudotime.html" class="sidebar-item-text sidebar-link">Pseudotime</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pseudobulk.html" class="sidebar-item-text sidebar-link">Pseudobulk</a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#cycling-through-files" id="toc-cycling-through-files" class="nav-link active" data-scroll-target="#cycling-through-files">Cycling through files</a></li>
  <li><a href="#looking-for-script" id="toc-looking-for-script" class="nav-link" data-scroll-target="#looking-for-script">Looking for script</a></li>
  <li><a href="#general-enrichment" id="toc-general-enrichment" class="nav-link" data-scroll-target="#general-enrichment">General enrichment</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title d-none d-lg-block">Extra tips</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="cycling-through-files" class="level1">
<h1>Cycling through files</h1>
<p>Often you will need to work a lot of files through a single function i.e.&nbsp;looping over names. There are a couple of ways of doing this below, both of which work on roughly the same principal. They will take a name such as MB18b_S2_R2_001.fastq.gz, split it into ‘fields’, demarcated by ‘_’, so MB18b is field 1 and S2 is field 2. This is done by <code>cut -d'_' -f 1-2</code> , where f1-2 are the fields 1 and 2. You need to pick the fields which let you separate out your samples from each other.</p>
<p>in the single nuclear you can do a below (complicated). <code>declare -p newArr</code> will show all the names that are going to be looped over.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1"></a><span class="va">arr</span><span class="op">=</span><span class="va">($id)</span></span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="co"># Get central bit of filename</span></span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="va">arr_names</span><span class="op">=</span><span class="va">()</span></span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="cf">for</span> i <span class="kw">in</span> <span class="st">"</span><span class="va">${arr</span><span class="op">[@]</span><span class="va">}</span><span class="st">"</span></span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="cf">do</span></span>
<span id="cb1-6"><a href="#cb1-6"></a>    <span class="va">arr_names</span><span class="op">+=</span><span class="va">(</span><span class="st">"</span><span class="va">$(</span><span class="bu">echo</span> <span class="va">$i</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-d</span><span class="st">'_'</span> <span class="at">-f</span> 1-3<span class="va">)</span><span class="st">"</span><span class="va">)</span></span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="cf">done</span></span>
<span id="cb1-8"><a href="#cb1-8"></a><span class="bu">declare</span> <span class="at">-p</span> <span class="va">arr_names</span></span>
<span id="cb1-9"><a href="#cb1-9"></a></span>
<span id="cb1-10"><a href="#cb1-10"></a><span class="co"># Get only the unique filenames</span></span>
<span id="cb1-11"><a href="#cb1-11"></a><span class="va">newArr</span><span class="op">=</span><span class="va">()</span><span class="kw">;</span> </span>
<span id="cb1-12"><a href="#cb1-12"></a><span class="cf">while</span> <span class="va">IFS</span><span class="op">=</span> <span class="bu">read</span> <span class="at">-r</span> <span class="at">-d</span> <span class="st">''</span> <span class="va">x</span><span class="kw">;</span> </span>
<span id="cb1-13"><a href="#cb1-13"></a><span class="cf">do</span> <span class="va">newArr</span><span class="op">+=</span><span class="va">(</span><span class="st">"</span><span class="va">$x</span><span class="st">"</span><span class="va">)</span><span class="kw">;</span> </span>
<span id="cb1-14"><a href="#cb1-14"></a><span class="cf">done</span> <span class="op">&lt;</span> <span class="op">&lt;(</span><span class="bu">printf</span> <span class="st">"%s\0"</span> <span class="st">"</span><span class="va">${arr_names</span><span class="op">[@]</span><span class="va">}</span><span class="st">"</span> <span class="kw">|</span> <span class="fu">sort</span> <span class="at">-uz</span><span class="op">)</span></span>
<span id="cb1-15"><a href="#cb1-15"></a><span class="bu">declare</span> <span class="at">-p</span> <span class="va">newArr</span></span>
<span id="cb1-16"><a href="#cb1-16"></a></span>
<span id="cb1-17"><a href="#cb1-17"></a><span class="co"># Loop over those names in the function</span></span>
<span id="cb1-18"><a href="#cb1-18"></a><span class="cf">for</span> i <span class="kw">in</span> <span class="va">${newArr</span><span class="op">[@]</span><span class="va">}</span></span>
<span id="cb1-19"><a href="#cb1-19"></a><span class="cf">do</span></span>
<span id="cb1-20"><a href="#cb1-20"></a></span>
<span id="cb1-21"><a href="#cb1-21"></a><span class="ex">fastp</span> <span class="dt">\</span></span>
<span id="cb1-22"><a href="#cb1-22"></a>-w 16 <span class="dt">\</span></span>
<span id="cb1-23"><a href="#cb1-23"></a>-p <span class="dt">\</span></span>
<span id="cb1-24"><a href="#cb1-24"></a>-Q <span class="dt">\</span></span>
<span id="cb1-25"><a href="#cb1-25"></a>-l 50 <span class="dt">\</span></span>
<span id="cb1-26"><a href="#cb1-26"></a>-i /scratch/moh1u21/single_nuclear/bsg-ftp.well.ox.ac.uk/220427_A00711_0538_BH7VMTDMXY/fastqs/<span class="va">${i}</span>_2.fastq.gz <span class="dt">\</span></span>
<span id="cb1-27"><a href="#cb1-27"></a>-o results/raw/trimmed/trimmed_<span class="va">${i}</span>_2.fastq.gz <span class="dt">\</span></span>
<span id="cb1-28"><a href="#cb1-28"></a>--adapter_sequence=AGCAGTGGTATCAACGCAGAGTGAATGGG <span class="dt">\</span></span>
<span id="cb1-29"><a href="#cb1-29"></a>--adapter_fasta /scratch/moh1u21/23-01-30-whole_muscle_snrnaseq/data/adapters.fasta <span class="dt">\</span></span>
<span id="cb1-30"><a href="#cb1-30"></a>--trim_poly_g <span class="dt">\</span></span>
<span id="cb1-31"><a href="#cb1-31"></a>--trim_poly_x <span class="dt">\</span></span>
<span id="cb1-32"><a href="#cb1-32"></a>--cut_tail <span class="dt">\</span></span>
<span id="cb1-33"><a href="#cb1-33"></a>--cut_mean_quality 30 <span class="dt">\</span></span>
<span id="cb1-34"><a href="#cb1-34"></a>--html results/raw/fastp_reports/trimmed_<span class="va">${i}</span>_fastp_report.html <span class="dt">\</span></span>
<span id="cb1-35"><a href="#cb1-35"></a>--json results/raw/fastp_reports/trimmed_<span class="va">${i}</span>_fastp_report.json <span class="op">&gt;&gt;</span> <span class="va">$fastp_logfile</span> <span class="dv">2</span><span class="op">&gt;&amp;</span><span class="dv">1</span></span>
<span id="cb1-36"><a href="#cb1-36"></a></span>
<span id="cb1-37"><a href="#cb1-37"></a><span class="ex">seqkit</span> pair <span class="dt">\</span></span>
<span id="cb1-38"><a href="#cb1-38"></a>--read1 /scratch/moh1u21/single_nuclear/bsg-ftp.well.ox.ac.uk/220427_A00711_0538_BH7VMTDMXY/fastqs/<span class="va">${i}</span>_1.fastq.gz <span class="dt">\</span></span>
<span id="cb1-39"><a href="#cb1-39"></a>--read2 <span class="st">"</span><span class="va">$project_folder</span><span class="st">"</span>/results/raw/trimmed/trimmed_<span class="va">${i}</span>_2.fastq.gz <span class="dt">\</span></span>
<span id="cb1-40"><a href="#cb1-40"></a>--out-dir <span class="st">"</span><span class="va">$project_folder</span><span class="st">"</span>/results/raw/final_fqs</span>
<span id="cb1-41"><a href="#cb1-41"></a></span>
<span id="cb1-42"><a href="#cb1-42"></a><span class="cf">done</span> </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Alternatively you can use a package called parallel, which effectively gets all the names, puts them in a file and will put them into a function iteratively wherever you put a {} as below.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1"></a><span class="co"># Below will get only the names of files (not folders) in the specified directory. </span></span>
<span id="cb2-2"><a href="#cb2-2"></a><span class="fu">find</span> /scratch/moh1u21/ATAC_seq_test/data/demultiplexed_reads/ATAC_seq_test/ <span class="at">-maxdepth</span> 1 <span class="at">-type</span> f <span class="kw">|</span> <span class="fu">awk</span> <span class="at">-F</span>/ <span class="st">'{print $NF}'</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-d</span><span class="st">"_"</span> <span class="at">-f1-2</span> <span class="kw">|</span> <span class="fu">sort</span> <span class="kw">|</span> <span class="fu">uniq</span> <span class="op">&gt;</span> fast_files.txt</span>
<span id="cb2-3"><a href="#cb2-3"></a></span>
<span id="cb2-4"><a href="#cb2-4"></a><span class="co"># cat fast_files.txt | parallel \</span></span>
<span id="cb2-5"><a href="#cb2-5"></a><span class="co"># fastq_info final_fqs/{}_1.fastq.gz final_fqs/trimmed_{}_2.fastq.gz '&gt;&gt;' Results.txt '2&gt;&amp;1'</span></span>
<span id="cb2-6"><a href="#cb2-6"></a></span>
<span id="cb2-7"><a href="#cb2-7"></a><span class="fu">cat</span> fast_files.txt <span class="kw">|</span> <span class="ex">parallel</span> <span class="dt">\</span></span>
<span id="cb2-8"><a href="#cb2-8"></a>fastp <span class="dt">\</span></span>
<span id="cb2-9"><a href="#cb2-9"></a>-w 16 <span class="dt">\</span></span>
<span id="cb2-10"><a href="#cb2-10"></a>-p <span class="dt">\</span></span>
<span id="cb2-11"><a href="#cb2-11"></a>--length_required 50 <span class="dt">\</span></span>
<span id="cb2-12"><a href="#cb2-12"></a>-i data/demultiplexed_reads/ATAC_seq_test/{}_R1_001.fastq.gz <span class="dt">\</span></span>
<span id="cb2-13"><a href="#cb2-13"></a>-I data/demultiplexed_reads/ATAC_seq_test/{}_R2_001.fastq.gz <span class="dt">\</span></span>
<span id="cb2-14"><a href="#cb2-14"></a>-o data/demultiplexed_reads/trimmed/trimmed_{}_R1_001.fastq.gz <span class="dt">\</span></span>
<span id="cb2-15"><a href="#cb2-15"></a>-O data/demultiplexed_reads/trimmed/trimmed_{}_R2_001.fastq.gz <span class="dt">\</span></span>
<span id="cb2-16"><a href="#cb2-16"></a>--detect_adapter_for_pe <span class="dt">\</span></span>
<span id="cb2-17"><a href="#cb2-17"></a>--correction <span class="dt">\</span></span>
<span id="cb2-18"><a href="#cb2-18"></a>--trim_poly_g <span class="dt">\</span></span>
<span id="cb2-19"><a href="#cb2-19"></a>--trim_poly_x <span class="dt">\</span></span>
<span id="cb2-20"><a href="#cb2-20"></a>--cut_tail <span class="dt">\</span></span>
<span id="cb2-21"><a href="#cb2-21"></a>--cut_mean_quality 30 <span class="dt">\</span></span>
<span id="cb2-22"><a href="#cb2-22"></a>--html results/reports/fastp/{}_trimmed_fastp_report.html <span class="dt">\</span></span>
<span id="cb2-23"><a href="#cb2-23"></a>--json results/reports/fastp/{}_trimmed_fastp_report.json <span class="st">'&gt;'</span> <span class="va">$fastp_logfile</span> <span class="st">'2&gt;&amp;1'</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="looking-for-script" class="level1">
<h1>Looking for script</h1>
<p>If you ever have a file and you need to find the script which generated it, you can use the below code to find the file in which it is written.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb3-2"><a href="#cb3-2"></a></span>
<span id="cb3-3"><a href="#cb3-3"></a><span class="co"># Set the directory path and search string</span></span>
<span id="cb3-4"><a href="#cb3-4"></a><span class="va">DIRECTORY</span><span class="op">=</span><span class="st">"path/to/your/folder"</span></span>
<span id="cb3-5"><a href="#cb3-5"></a><span class="va">SEARCH_STRING</span><span class="op">=</span><span class="st">"your_line_of_code"</span></span>
<span id="cb3-6"><a href="#cb3-6"></a></span>
<span id="cb3-7"><a href="#cb3-7"></a><span class="co"># Search for the specific line of code in R script files</span></span>
<span id="cb3-8"><a href="#cb3-8"></a><span class="fu">find</span> ./ <span class="at">-name</span> <span class="st">"*.R"</span> <span class="at">-type</span> f <span class="at">-exec</span> awk <span class="at">-v</span> search_string=<span class="st">"enrichment_results"</span> <span class="st">'</span></span>
<span id="cb3-9"><a href="#cb3-9"></a><span class="st">  $0 ~ search_string {</span></span>
<span id="cb3-10"><a href="#cb3-10"></a><span class="st">    printf "%s:%s:%s\n", FILENAME, FNR, $0</span></span>
<span id="cb3-11"><a href="#cb3-11"></a><span class="st">  }'</span> {} <span class="dt">\;</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="general-enrichment" class="level1">
<h1>General enrichment</h1>
<ul>
<li><p>The code takes a table with the most differentially expressed markers for a single cell dataset by cluster. The universe in this case is all protein-coding genes in the genome. An example of the input markers object is hashed below.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="co">#                p_val avg_log2FC pct.1 pct.2     p_val_adj cluster   gene</span></span>
<span id="cb4-2"><a href="#cb4-2"></a><span class="co"># TNNT1  1.510251e-192  1.4501320 0.961 0.376 2.587512e-188       0  TNNT1</span></span>
<span id="cb4-3"><a href="#cb4-3"></a><span class="co"># ATP2A2  7.443596e-87  0.7266991 0.667 0.242  1.275311e-82       0 ATP2A2</span></span>
<span id="cb4-4"><a href="#cb4-4"></a><span class="co"># TPM3    2.308168e-84  0.5415437 0.867 0.366  3.954584e-80       0   TPM3</span></span>
<span id="cb4-5"><a href="#cb4-5"></a><span class="co"># LGR5    9.075674e-71  0.6019044 0.452 0.118  1.554935e-66       0   LGR5</span></span>
<span id="cb4-6"><a href="#cb4-6"></a><span class="co"># BICC1   1.202617e-64  0.6783576 0.549 0.202  2.060444e-60       0  BICC1</span></span>
<span id="cb4-7"><a href="#cb4-7"></a><span class="co"># MYBPC1  2.295684e-59  0.5507988 0.996 0.895  3.933196e-55       0 MYBPC1</span></span>
<span id="cb4-8"><a href="#cb4-8"></a></span>
<span id="cb4-9"><a href="#cb4-9"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb4-10"><a href="#cb4-10"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb4-11"><a href="#cb4-11"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb4-12"><a href="#cb4-12"></a><span class="fu">library</span>(AnnotationDbi)</span>
<span id="cb4-13"><a href="#cb4-13"></a></span>
<span id="cb4-14"><a href="#cb4-14"></a>markers <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"data/wholemuscle_sn_allmarkers.rds"</span>)</span>
<span id="cb4-15"><a href="#cb4-15"></a></span>
<span id="cb4-16"><a href="#cb4-16"></a><span class="co"># Set up my universe ------------------------------------------------------</span></span>
<span id="cb4-17"><a href="#cb4-17"></a><span class="do">## Get a list of all Entrez Gene IDs</span></span>
<span id="cb4-18"><a href="#cb4-18"></a>entrez_ids <span class="ot">&lt;-</span> <span class="fu">keys</span>(org.Hs.eg.db, <span class="at">keytype =</span> <span class="st">"ENTREZID"</span>)</span>
<span id="cb4-19"><a href="#cb4-19"></a></span>
<span id="cb4-20"><a href="#cb4-20"></a><span class="do">## Get the corresponding gene symbols</span></span>
<span id="cb4-21"><a href="#cb4-21"></a>gene_info <span class="ot">&lt;-</span> </span>
<span id="cb4-22"><a href="#cb4-22"></a>  AnnotationDbi<span class="sc">::</span><span class="fu">select</span>(org.Hs.eg.db, </span>
<span id="cb4-23"><a href="#cb4-23"></a>                        <span class="at">keys =</span> entrez_ids, </span>
<span id="cb4-24"><a href="#cb4-24"></a>                        <span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"SYMBOL"</span>, <span class="st">"GENETYPE"</span>), </span>
<span id="cb4-25"><a href="#cb4-25"></a>                        <span class="at">keytype =</span> <span class="st">"ENTREZID"</span>)</span>
<span id="cb4-26"><a href="#cb4-26"></a></span>
<span id="cb4-27"><a href="#cb4-27"></a>protein_coding_genes <span class="ot">&lt;-</span> gene_info[gene_info<span class="sc">$</span>GENETYPE <span class="sc">==</span> <span class="st">"protein-coding"</span>,]</span>
<span id="cb4-28"><a href="#cb4-28"></a>universe <span class="ot">&lt;-</span> <span class="fu">pull</span>(protein_coding_genes, SYMBOL)</span>
<span id="cb4-29"><a href="#cb4-29"></a></span>
<span id="cb4-30"><a href="#cb4-30"></a></span>
<span id="cb4-31"><a href="#cb4-31"></a><span class="co"># Process data ------------------------------------------------------------</span></span>
<span id="cb4-32"><a href="#cb4-32"></a>nested <span class="ot">&lt;-</span> </span>
<span id="cb4-33"><a href="#cb4-33"></a>  markers <span class="sc">%&gt;%</span></span>
<span id="cb4-34"><a href="#cb4-34"></a>  <span class="fu">filter</span>(p_val_adj<span class="sc">&lt;</span><span class="fl">0.05</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-35"><a href="#cb4-35"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span> </span>
<span id="cb4-36"><a href="#cb4-36"></a>  <span class="fu">group_map</span>(<span class="sc">~</span><span class="fu">pull</span>(.x, gene))</span>
<span id="cb4-37"><a href="#cb4-37"></a></span>
<span id="cb4-38"><a href="#cb4-38"></a><span class="co"># universe &lt;- shrink.d11$GeneID</span></span>
<span id="cb4-39"><a href="#cb4-39"></a></span>
<span id="cb4-40"><a href="#cb4-40"></a>enrichment_results <span class="ot">&lt;-</span> </span>
<span id="cb4-41"><a href="#cb4-41"></a>  <span class="fu">map</span>(</span>
<span id="cb4-42"><a href="#cb4-42"></a>    nested,</span>
<span id="cb4-43"><a href="#cb4-43"></a>    <span class="sc">~</span> <span class="fu">enrichGO</span>(</span>
<span id="cb4-44"><a href="#cb4-44"></a>      <span class="at">gene =</span> .x,</span>
<span id="cb4-45"><a href="#cb4-45"></a>      <span class="at">universe =</span> universe, </span>
<span id="cb4-46"><a href="#cb4-46"></a>      <span class="at">OrgDb =</span> org.Hs.eg.db,</span>
<span id="cb4-47"><a href="#cb4-47"></a>      <span class="at">keyType =</span> <span class="st">"SYMBOL"</span>,</span>
<span id="cb4-48"><a href="#cb4-48"></a>      <span class="at">ont =</span> <span class="st">"ALL"</span>,</span>
<span id="cb4-49"><a href="#cb4-49"></a>      <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb4-50"><a href="#cb4-50"></a>      <span class="at">readable =</span> <span class="cn">TRUE</span></span>
<span id="cb4-51"><a href="#cb4-51"></a>    )</span>
<span id="cb4-52"><a href="#cb4-52"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb4-53"><a href="#cb4-53"></a>  <span class="fu">map</span>(as_tibble) <span class="sc">%&gt;%</span></span>
<span id="cb4-54"><a href="#cb4-54"></a>  <span class="fu">map</span>( <span class="sc">~</span> <span class="fu">filter</span>(.x, Count <span class="sc">&gt;=</span> <span class="dv">3</span>)) </span>
<span id="cb4-55"><a href="#cb4-55"></a></span>
<span id="cb4-56"><a href="#cb4-56"></a><span class="fu">names</span>(enrichment_results) <span class="ot">&lt;-</span> <span class="fu">str_c</span>(<span class="st">"Cluster_"</span>, <span class="fu">seq</span>(<span class="dv">0</span>, <span class="fu">length</span>(enrichment_results)<span class="sc">-</span><span class="dv">1</span>, <span class="dv">1</span>))</span>
<span id="cb4-57"><a href="#cb4-57"></a>enrichment_results_export <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(enrichment_results, <span class="at">.id=</span><span class="st">"cluster"</span>)</span>
<span id="cb4-58"><a href="#cb4-58"></a></span>
<span id="cb4-59"><a href="#cb4-59"></a>norbs <span class="ot">&lt;-</span> </span>
<span id="cb4-60"><a href="#cb4-60"></a>  enrichment_results_export <span class="sc">%&gt;%</span> </span>
<span id="cb4-61"><a href="#cb4-61"></a>  <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span> </span>
<span id="cb4-62"><a href="#cb4-62"></a>  <span class="fu">arrange</span>(cluster, p.adjust, <span class="at">.by_group =</span> T)</span>
<span id="cb4-63"><a href="#cb4-63"></a></span>
<span id="cb4-64"><a href="#cb4-64"></a><span class="fu">table</span>(norbs<span class="sc">$</span>cluster)</span>
<span id="cb4-65"><a href="#cb4-65"></a></span>
<span id="cb4-66"><a href="#cb4-66"></a>abbles <span class="ot">&lt;-</span> </span>
<span id="cb4-67"><a href="#cb4-67"></a>  norbs <span class="sc">%&gt;%</span> </span>
<span id="cb4-68"><a href="#cb4-68"></a>  <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span> </span>
<span id="cb4-69"><a href="#cb4-69"></a>  <span class="fu">slice</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>)</span>
<span id="cb4-70"><a href="#cb4-70"></a></span>
<span id="cb4-71"><a href="#cb4-71"></a><span class="fu">write_csv</span>(norbs, <span class="st">"results/enrichment_results_ss.csv"</span>)</span>
<span id="cb4-72"><a href="#cb4-72"></a><span class="fu">write_csv</span>(abbles, <span class="st">"results/cluster_markers_top10s.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></li>
<li><p>The code could be modified as below to run a single set of genes - NOT RUN -</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb5-2"><a href="#cb5-2"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb5-3"><a href="#cb5-3"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb5-4"><a href="#cb5-4"></a><span class="fu">library</span>(AnnotationDbi)</span>
<span id="cb5-5"><a href="#cb5-5"></a></span>
<span id="cb5-6"><a href="#cb5-6"></a>markers <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"data/wholemuscle_sn_allmarkers.rds"</span>)</span>
<span id="cb5-7"><a href="#cb5-7"></a></span>
<span id="cb5-8"><a href="#cb5-8"></a><span class="co"># Set up my universe ------------------------------------------------------</span></span>
<span id="cb5-9"><a href="#cb5-9"></a><span class="do">## Get a list of all Entrez Gene IDs</span></span>
<span id="cb5-10"><a href="#cb5-10"></a>entrez_ids <span class="ot">&lt;-</span> <span class="fu">keys</span>(org.Hs.eg.db, <span class="at">keytype =</span> <span class="st">"ENTREZID"</span>)</span>
<span id="cb5-11"><a href="#cb5-11"></a></span>
<span id="cb5-12"><a href="#cb5-12"></a><span class="do">## Get the corresponding gene symbols</span></span>
<span id="cb5-13"><a href="#cb5-13"></a>gene_info <span class="ot">&lt;-</span> </span>
<span id="cb5-14"><a href="#cb5-14"></a>  AnnotationDbi<span class="sc">::</span><span class="fu">select</span>(org.Hs.eg.db, </span>
<span id="cb5-15"><a href="#cb5-15"></a>                        <span class="at">keys =</span> entrez_ids, </span>
<span id="cb5-16"><a href="#cb5-16"></a>                        <span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"SYMBOL"</span>, <span class="st">"GENETYPE"</span>), </span>
<span id="cb5-17"><a href="#cb5-17"></a>                        <span class="at">keytype =</span> <span class="st">"ENTREZID"</span>)</span>
<span id="cb5-18"><a href="#cb5-18"></a></span>
<span id="cb5-19"><a href="#cb5-19"></a>protein_coding_genes <span class="ot">&lt;-</span> gene_info[gene_info<span class="sc">$</span>GENETYPE <span class="sc">==</span> <span class="st">"protein-coding"</span>,]</span>
<span id="cb5-20"><a href="#cb5-20"></a>universe <span class="ot">&lt;-</span> <span class="fu">pull</span>(protein_coding_genes, SYMBOL)</span>
<span id="cb5-21"><a href="#cb5-21"></a></span>
<span id="cb5-22"><a href="#cb5-22"></a></span>
<span id="cb5-23"><a href="#cb5-23"></a><span class="co"># Process data ------------------------------------------------------------</span></span>
<span id="cb5-24"><a href="#cb5-24"></a></span>
<span id="cb5-25"><a href="#cb5-25"></a>enrichment_results <span class="ot">&lt;-</span> </span>
<span id="cb5-26"><a href="#cb5-26"></a>  <span class="fu">enrichGO</span>(</span>
<span id="cb5-27"><a href="#cb5-27"></a>      <span class="at">gene =</span> <span class="st">"&lt;character vector containing genes of interest&gt;"</span>,</span>
<span id="cb5-28"><a href="#cb5-28"></a>      <span class="at">universe =</span> universe, </span>
<span id="cb5-29"><a href="#cb5-29"></a>      <span class="at">OrgDb =</span> org.Hs.eg.db,</span>
<span id="cb5-30"><a href="#cb5-30"></a>      <span class="at">keyType =</span> <span class="st">"SYMBOL"</span>,</span>
<span id="cb5-31"><a href="#cb5-31"></a>      <span class="at">ont =</span> <span class="st">"ALL"</span>,</span>
<span id="cb5-32"><a href="#cb5-32"></a>      <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb5-33"><a href="#cb5-33"></a>      <span class="at">readable =</span> <span class="cn">TRUE</span></span>
<span id="cb5-34"><a href="#cb5-34"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb5-35"><a href="#cb5-35"></a>  as_tibble <span class="sc">%&gt;%</span></span>
<span id="cb5-36"><a href="#cb5-36"></a>  <span class="fu">filter</span>(Count <span class="sc">&gt;=</span> <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></li>
</ul>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>