<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Matt work summary - Pseudobulk</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title">Pseudobulk</h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Matt work summary</a> 
    </div>
      </div>
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">Home</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./whole_muscle_scrnaseq.html" class="sidebar-item-text sidebar-link">Whole muscle single nuclear</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./myob_single_nuc.html" class="sidebar-item-text sidebar-link">Myoblast single nuclear</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./atac_seq.html" class="sidebar-item-text sidebar-link">ATACseq</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./conda_envs.html" class="sidebar-item-text sidebar-link">Conda environments</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./extra_tips.html" class="sidebar-item-text sidebar-link">Extra tips</a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">Single cell</a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./single_cell.html" class="sidebar-item-text sidebar-link">Overview</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pseudotime.html" class="sidebar-item-text sidebar-link">Pseudotime</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pseudobulk.html" class="sidebar-item-text sidebar-link active">Pseudobulk</a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#summary" id="toc-summary" class="nav-link active" data-scroll-target="#summary">Summary</a></li>
  <li><a href="#setup" id="toc-setup" class="nav-link" data-scroll-target="#setup">Setup</a></li>
  <li><a href="#processing-script" id="toc-processing-script" class="nav-link" data-scroll-target="#processing-script">Processing script</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title d-none d-lg-block">Pseudobulk</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="summary" class="level1">
<h1>Summary</h1>
<p>Pseudobulk aggregates all single cell count data together to form a single set of counts for each sample, rather than counts from each cell.</p>
</section>
<section id="setup" class="level1">
<h1>Setup</h1>
<ul>
<li><p>The project folder is <code>/Users/fluentin44/Library/CloudStorage/OneDrive-UniversityofSouthampton/21-11-18-single_cell_seq</code></p></li>
<li><p>The script used to generate the file <code>22-08-09-pseudobulk_by_cluster.xlsx</code> in the shared area is made by the script below which is located at <code>~/Library/CloudStorage/OneDrive-UniversityofSouthampton/21-11-18-single_cell_seq/src/22-08-09-pseudobulk_by_cluster.R</code> .</p></li>
</ul>
</section>
<section id="processing-script" class="level1">
<h1>Processing script</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="co"># Libraries ---------------------------------------------------------------</span></span>
<span id="cb1-2"><a href="#cb1-2"></a></span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(tidyverse))</span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(SingleCellExperiment))</span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(Seurat))</span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(scater))</span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(ExperimentHub))</span>
<span id="cb1-8"><a href="#cb1-8"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(edgeR))</span>
<span id="cb1-9"><a href="#cb1-9"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(DESeq2))</span>
<span id="cb1-10"><a href="#cb1-10"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(batchelor))</span>
<span id="cb1-11"><a href="#cb1-11"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(pheatmap))</span>
<span id="cb1-12"><a href="#cb1-12"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(RColorBrewer))</span>
<span id="cb1-13"><a href="#cb1-13"></a></span>
<span id="cb1-14"><a href="#cb1-14"></a></span>
<span id="cb1-15"><a href="#cb1-15"></a><span class="co"># Functions ---------------------------------------------------------------</span></span>
<span id="cb1-16"><a href="#cb1-16"></a></span>
<span id="cb1-17"><a href="#cb1-17"></a>gene_ex_heatmap <span class="ot">&lt;-</span> <span class="cf">function</span>(dds,</span>
<span id="cb1-18"><a href="#cb1-18"></a>                            voi,</span>
<span id="cb1-19"><a href="#cb1-19"></a>                            variables,</span>
<span id="cb1-20"><a href="#cb1-20"></a>                            rld,</span>
<span id="cb1-21"><a href="#cb1-21"></a>                            cluster_rows,</span>
<span id="cb1-22"><a href="#cb1-22"></a>                            cluster_cols) {</span>
<span id="cb1-23"><a href="#cb1-23"></a>  niff <span class="ot">&lt;-</span> <span class="fu">c</span>(voi, variables)</span>
<span id="cb1-24"><a href="#cb1-24"></a>  select <span class="ot">&lt;-</span> <span class="fu">order</span>(<span class="fu">rowMeans</span>(<span class="fu">counts</span>(dds, <span class="at">normalized =</span> <span class="cn">TRUE</span>)),</span>
<span id="cb1-25"><a href="#cb1-25"></a>                  <span class="at">decreasing =</span> <span class="cn">TRUE</span>)[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]</span>
<span id="cb1-26"><a href="#cb1-26"></a>  df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(dds)[, niff])</span>
<span id="cb1-27"><a href="#cb1-27"></a>  <span class="co">#rownames(df) &lt;- colnames(counts(dds))</span></span>
<span id="cb1-28"><a href="#cb1-28"></a>  <span class="fu">pheatmap</span>(</span>
<span id="cb1-29"><a href="#cb1-29"></a>    <span class="fu">assay</span>(rld)[select, ],</span>
<span id="cb1-30"><a href="#cb1-30"></a>    <span class="at">cluster_rows =</span> cluster_rows,</span>
<span id="cb1-31"><a href="#cb1-31"></a>    <span class="at">show_rownames =</span> T,</span>
<span id="cb1-32"><a href="#cb1-32"></a>    <span class="at">cluster_cols =</span> cluster_cols,</span>
<span id="cb1-33"><a href="#cb1-33"></a>    <span class="at">annotation_col =</span> df</span>
<span id="cb1-34"><a href="#cb1-34"></a>  )</span>
<span id="cb1-35"><a href="#cb1-35"></a>}</span>
<span id="cb1-36"><a href="#cb1-36"></a></span>
<span id="cb1-37"><a href="#cb1-37"></a></span>
<span id="cb1-38"><a href="#cb1-38"></a></span>
<span id="cb1-39"><a href="#cb1-39"></a>dea <span class="ot">&lt;-</span> <span class="cf">function</span>(y, </span>
<span id="cb1-40"><a href="#cb1-40"></a>                voi, </span>
<span id="cb1-41"><a href="#cb1-41"></a>                variables, </span>
<span id="cb1-42"><a href="#cb1-42"></a>                <span class="co">#continuous = NULL, </span></span>
<span id="cb1-43"><a href="#cb1-43"></a>                variable_format,</span>
<span id="cb1-44"><a href="#cb1-44"></a>                <span class="at">wald=</span>T){</span>
<span id="cb1-45"><a href="#cb1-45"></a>  </span>
<span id="cb1-46"><a href="#cb1-46"></a>  variable_format <span class="ot">&lt;-</span> <span class="fu">ensyms</span>(variable_format)</span>
<span id="cb1-47"><a href="#cb1-47"></a>  </span>
<span id="cb1-48"><a href="#cb1-48"></a>  <span class="co"># countsToUse &lt;- counts(sce)</span></span>
<span id="cb1-49"><a href="#cb1-49"></a>  <span class="co"># #colnames(countsToUse) &lt;- colData(sce)$ID</span></span>
<span id="cb1-50"><a href="#cb1-50"></a>  <span class="co"># y &lt;- DGEList(countsToUse, samples = colData(sce))</span></span>
<span id="cb1-51"><a href="#cb1-51"></a>  <span class="co"># </span></span>
<span id="cb1-52"><a href="#cb1-52"></a>  <span class="co"># for (i in variables) {</span></span>
<span id="cb1-53"><a href="#cb1-53"></a>  <span class="co">#   lose &lt;- is.na(y$samples[[i]])</span></span>
<span id="cb1-54"><a href="#cb1-54"></a>  <span class="co">#   y$samples &lt;- y$samples[!lose, ]</span></span>
<span id="cb1-55"><a href="#cb1-55"></a>  <span class="co">#   y$counts &lt;- y$counts[,colnames(y$counts) %in% rownames(y$samples)]</span></span>
<span id="cb1-56"><a href="#cb1-56"></a>  <span class="co"># }</span></span>
<span id="cb1-57"><a href="#cb1-57"></a>  <span class="co"># </span></span>
<span id="cb1-58"><a href="#cb1-58"></a>  <span class="co"># lose &lt;- is.na(y$samples[[voi]])</span></span>
<span id="cb1-59"><a href="#cb1-59"></a>  <span class="co"># y$samples &lt;- y$samples[!lose, ]</span></span>
<span id="cb1-60"><a href="#cb1-60"></a>  <span class="co"># y$counts &lt;- y$counts[,colnames(y$counts) %in% rownames(y$samples)]</span></span>
<span id="cb1-61"><a href="#cb1-61"></a>  <span class="co"># </span></span>
<span id="cb1-62"><a href="#cb1-62"></a>  <span class="co"># discarded &lt;- y$samples$ncells &lt; 20</span></span>
<span id="cb1-63"><a href="#cb1-63"></a>  <span class="co"># y &lt;- y[,!discarded]</span></span>
<span id="cb1-64"><a href="#cb1-64"></a>  <span class="co"># print(summary(discarded))</span></span>
<span id="cb1-65"><a href="#cb1-65"></a>  <span class="co"># enframe(summary(discarded))</span></span>
<span id="cb1-66"><a href="#cb1-66"></a>  <span class="co"># </span></span>
<span id="cb1-67"><a href="#cb1-67"></a>  <span class="co"># keep &lt;- filterByExpr(y, group=y$samples[[voi]]) </span></span>
<span id="cb1-68"><a href="#cb1-68"></a>  <span class="co"># y &lt;- y[keep, , keep.lib.sizes=FALSE] </span></span>
<span id="cb1-69"><a href="#cb1-69"></a>  <span class="co"># print(summary(keep))</span></span>
<span id="cb1-70"><a href="#cb1-70"></a>  <span class="co"># enframe(summary(keep))</span></span>
<span id="cb1-71"><a href="#cb1-71"></a>  <span class="co"># </span></span>
<span id="cb1-72"><a href="#cb1-72"></a>  <span class="co"># if(length(unique(y$samples$[[voi]])) &gt; 1</span></span>
<span id="cb1-73"><a href="#cb1-73"></a>  </span>
<span id="cb1-74"><a href="#cb1-74"></a>  <span class="cf">if</span>(wald){</span>
<span id="cb1-75"><a href="#cb1-75"></a>    mod <span class="ot">&lt;-</span> <span class="fu">as.formula</span>(<span class="fu">paste</span>(<span class="st">"~"</span>, <span class="fu">paste</span>(variables, <span class="at">collapse=</span><span class="st">"+"</span>), <span class="st">"+"</span>, voi, <span class="at">sep=</span><span class="st">""</span>))</span>
<span id="cb1-76"><a href="#cb1-76"></a>    </span>
<span id="cb1-77"><a href="#cb1-77"></a>    dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(y<span class="sc">$</span>counts, </span>
<span id="cb1-78"><a href="#cb1-78"></a>                                  <span class="at">colData =</span> y<span class="sc">$</span>samples, </span>
<span id="cb1-79"><a href="#cb1-79"></a>                                  <span class="at">design =</span> mod)</span>
<span id="cb1-80"><a href="#cb1-80"></a>    </span>
<span id="cb1-81"><a href="#cb1-81"></a>    <span class="co"># rld &lt;- vst(dds, blind=TRUE)</span></span>
<span id="cb1-82"><a href="#cb1-82"></a>    </span>
<span id="cb1-83"><a href="#cb1-83"></a>    <span class="co"># Input is a matrix of log transformed values</span></span>
<span id="cb1-84"><a href="#cb1-84"></a>    <span class="co"># https://hbctraining.github.io/DGE_workshop/lessons/03_DGE_QC_analysis.html</span></span>
<span id="cb1-85"><a href="#cb1-85"></a>    rld <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds, <span class="at">blind=</span>T)</span>
<span id="cb1-86"><a href="#cb1-86"></a>    rld_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(rld)</span>
<span id="cb1-87"><a href="#cb1-87"></a>    rld_cor <span class="ot">&lt;-</span> <span class="fu">cor</span>(rld_mat)</span>
<span id="cb1-88"><a href="#cb1-88"></a>    pca <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(rld_mat))</span>
<span id="cb1-89"><a href="#cb1-89"></a>    </span>
<span id="cb1-90"><a href="#cb1-90"></a>    <span class="co"># Create data frame with metadata and PC3 and PC4 values for input to ggplot</span></span>
<span id="cb1-91"><a href="#cb1-91"></a>    <span class="fu">all</span>(<span class="fu">rownames</span>(<span class="fu">colData</span>(rld)) <span class="sc">==</span> <span class="fu">rownames</span>(pca<span class="sc">$</span>x))</span>
<span id="cb1-92"><a href="#cb1-92"></a>    df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">cbind</span>(<span class="fu">colData</span>(rld), pca<span class="sc">$</span>x))</span>
<span id="cb1-93"><a href="#cb1-93"></a>    <span class="fu">ggplot</span>(df) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>PC1, <span class="at">y=</span>PC2, <span class="at">color =</span> voi))</span>
<span id="cb1-94"><a href="#cb1-94"></a>    </span>
<span id="cb1-95"><a href="#cb1-95"></a>    DESeq2<span class="sc">::</span><span class="fu">plotPCA</span>(rld, <span class="at">intgroup =</span> voi)</span>
<span id="cb1-96"><a href="#cb1-96"></a>    </span>
<span id="cb1-97"><a href="#cb1-97"></a>    pheatmap<span class="sc">::</span><span class="fu">pheatmap</span>(rld_cor, <span class="at">annotation =</span> y<span class="sc">$</span>samples[, <span class="fu">c</span>(voi), <span class="at">drop=</span>F])</span>
<span id="cb1-98"><a href="#cb1-98"></a>    </span>
<span id="cb1-99"><a href="#cb1-99"></a>    <span class="co"># Run DESeq2 differential expression analysis</span></span>
<span id="cb1-100"><a href="#cb1-100"></a>    dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span>
<span id="cb1-101"><a href="#cb1-101"></a>    </span>
<span id="cb1-102"><a href="#cb1-102"></a>    <span class="co"># Plot dispersion estimates</span></span>
<span id="cb1-103"><a href="#cb1-103"></a>    <span class="fu">plotDispEsts</span>(dds)</span>
<span id="cb1-104"><a href="#cb1-104"></a>    </span>
<span id="cb1-105"><a href="#cb1-105"></a>    <span class="co"># Check the coefficients for the comparison</span></span>
<span id="cb1-106"><a href="#cb1-106"></a>    <span class="fu">print</span>(<span class="fu">resultsNames</span>(dds))</span>
<span id="cb1-107"><a href="#cb1-107"></a>    </span>
<span id="cb1-108"><a href="#cb1-108"></a>    <span class="fu">gene_ex_heatmap</span>(</span>
<span id="cb1-109"><a href="#cb1-109"></a>      dds,</span>
<span id="cb1-110"><a href="#cb1-110"></a>      <span class="at">voi =</span> voi,</span>
<span id="cb1-111"><a href="#cb1-111"></a>      <span class="at">variables =</span> variables,</span>
<span id="cb1-112"><a href="#cb1-112"></a>      rld,</span>
<span id="cb1-113"><a href="#cb1-113"></a>      <span class="at">cluster_rows =</span> F,</span>
<span id="cb1-114"><a href="#cb1-114"></a>      <span class="at">cluster_cols =</span> T</span>
<span id="cb1-115"><a href="#cb1-115"></a>    )</span>
<span id="cb1-116"><a href="#cb1-116"></a>    </span>
<span id="cb1-117"><a href="#cb1-117"></a>    <span class="cf">if</span>(variable_format <span class="sc">==</span> <span class="st">"continuous"</span>){</span>
<span id="cb1-118"><a href="#cb1-118"></a>      <span class="co"># x &lt;- readline(prompt="Please choose a coefficient ")</span></span>
<span id="cb1-119"><a href="#cb1-119"></a>      </span>
<span id="cb1-120"><a href="#cb1-120"></a>      res <span class="ot">&lt;-</span> <span class="fu">results</span>(dds,</span>
<span id="cb1-121"><a href="#cb1-121"></a>                     <span class="at">name =</span> voi,</span>
<span id="cb1-122"><a href="#cb1-122"></a>                     <span class="at">alpha =</span> <span class="fl">0.05</span>)</span>
<span id="cb1-123"><a href="#cb1-123"></a>      </span>
<span id="cb1-124"><a href="#cb1-124"></a>      res <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds,</span>
<span id="cb1-125"><a href="#cb1-125"></a>                       <span class="at">coef =</span> voi,</span>
<span id="cb1-126"><a href="#cb1-126"></a>                       <span class="at">res=</span>res,</span>
<span id="cb1-127"><a href="#cb1-127"></a>                       <span class="at">type =</span> <span class="st">"apeglm"</span>)</span>
<span id="cb1-128"><a href="#cb1-128"></a>      </span>
<span id="cb1-129"><a href="#cb1-129"></a>      <span class="co"># Turn the results object into a tibble for use with tidyverse functions</span></span>
<span id="cb1-130"><a href="#cb1-130"></a>      res_tbl <span class="ot">&lt;-</span> </span>
<span id="cb1-131"><a href="#cb1-131"></a>        res <span class="sc">%&gt;%</span></span>
<span id="cb1-132"><a href="#cb1-132"></a>        <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb1-133"><a href="#cb1-133"></a>        <span class="fu">rownames_to_column</span>(<span class="at">var=</span><span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb1-134"><a href="#cb1-134"></a>        <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb1-135"><a href="#cb1-135"></a>        <span class="fu">arrange</span>(padj, pvalue)</span>
<span id="cb1-136"><a href="#cb1-136"></a>      </span>
<span id="cb1-137"><a href="#cb1-137"></a>      dds <span class="ot">&lt;-</span> res_tbl</span>
<span id="cb1-138"><a href="#cb1-138"></a>    }<span class="cf">else</span> <span class="cf">if</span>(variable_format <span class="sc">==</span> <span class="st">"catagorical"</span>){</span>
<span id="cb1-139"><a href="#cb1-139"></a>      <span class="co"># x &lt;- readline(prompt="Please choose a coefficient ")</span></span>
<span id="cb1-140"><a href="#cb1-140"></a>      interest <span class="ot">&lt;-</span> <span class="fu">resultsNames</span>(dds)[<span class="fu">length</span>(<span class="fu">resultsNames</span>(dds))]</span>
<span id="cb1-141"><a href="#cb1-141"></a>      </span>
<span id="cb1-142"><a href="#cb1-142"></a>      res <span class="ot">&lt;-</span> <span class="fu">results</span>(dds,</span>
<span id="cb1-143"><a href="#cb1-143"></a>                     <span class="at">name =</span> interest,</span>
<span id="cb1-144"><a href="#cb1-144"></a>                     <span class="at">alpha =</span> <span class="fl">0.05</span>)</span>
<span id="cb1-145"><a href="#cb1-145"></a>      </span>
<span id="cb1-146"><a href="#cb1-146"></a>      res <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds,</span>
<span id="cb1-147"><a href="#cb1-147"></a>                       <span class="at">coef =</span> interest,</span>
<span id="cb1-148"><a href="#cb1-148"></a>                       <span class="at">res=</span>res,</span>
<span id="cb1-149"><a href="#cb1-149"></a>                       <span class="at">type =</span> <span class="st">"apeglm"</span>)</span>
<span id="cb1-150"><a href="#cb1-150"></a>      </span>
<span id="cb1-151"><a href="#cb1-151"></a>      <span class="co"># Turn the results object into a tibble for use with tidyverse functions</span></span>
<span id="cb1-152"><a href="#cb1-152"></a>      res_tbl <span class="ot">&lt;-</span> </span>
<span id="cb1-153"><a href="#cb1-153"></a>        res <span class="sc">%&gt;%</span></span>
<span id="cb1-154"><a href="#cb1-154"></a>        <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb1-155"><a href="#cb1-155"></a>        <span class="fu">rownames_to_column</span>(<span class="at">var=</span><span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb1-156"><a href="#cb1-156"></a>        <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb1-157"><a href="#cb1-157"></a>        <span class="fu">arrange</span>(padj, pvalue)</span>
<span id="cb1-158"><a href="#cb1-158"></a>      </span>
<span id="cb1-159"><a href="#cb1-159"></a>      dds <span class="ot">&lt;-</span> res_tbl</span>
<span id="cb1-160"><a href="#cb1-160"></a>    }</span>
<span id="cb1-161"><a href="#cb1-161"></a>  }<span class="cf">else</span>{</span>
<span id="cb1-162"><a href="#cb1-162"></a>    mod <span class="ot">&lt;-</span> <span class="fu">as.formula</span>(<span class="fu">paste</span>(<span class="st">"~"</span>, <span class="fu">paste</span>(variables, <span class="at">collapse=</span><span class="st">"+"</span>), <span class="st">"+"</span>, voi, <span class="at">sep=</span><span class="st">""</span>))</span>
<span id="cb1-163"><a href="#cb1-163"></a>    </span>
<span id="cb1-164"><a href="#cb1-164"></a>    dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(y<span class="sc">$</span>counts, </span>
<span id="cb1-165"><a href="#cb1-165"></a>                                  <span class="at">colData =</span> y<span class="sc">$</span>samples, </span>
<span id="cb1-166"><a href="#cb1-166"></a>                                  <span class="at">design =</span> mod)</span>
<span id="cb1-167"><a href="#cb1-167"></a>    </span>
<span id="cb1-168"><a href="#cb1-168"></a>    <span class="co"># rld &lt;- vst(dds, blind=TRUE)</span></span>
<span id="cb1-169"><a href="#cb1-169"></a>    </span>
<span id="cb1-170"><a href="#cb1-170"></a>    <span class="co"># Input is a matrix of log transformed values</span></span>
<span id="cb1-171"><a href="#cb1-171"></a>    <span class="co"># https://hbctraining.github.io/DGE_workshop/lessons/03_DGE_QC_analysis.html</span></span>
<span id="cb1-172"><a href="#cb1-172"></a>    rld <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds, <span class="at">blind=</span>T)</span>
<span id="cb1-173"><a href="#cb1-173"></a>    rld_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(rld)</span>
<span id="cb1-174"><a href="#cb1-174"></a>    rld_cor <span class="ot">&lt;-</span> <span class="fu">cor</span>(rld_mat)</span>
<span id="cb1-175"><a href="#cb1-175"></a>    pca <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(rld_mat))</span>
<span id="cb1-176"><a href="#cb1-176"></a>    </span>
<span id="cb1-177"><a href="#cb1-177"></a>    <span class="co"># Create data frame with metadata and PC3 and PC4 values for input to ggplot</span></span>
<span id="cb1-178"><a href="#cb1-178"></a>    <span class="fu">all</span>(<span class="fu">rownames</span>(<span class="fu">colData</span>(rld)) <span class="sc">==</span> <span class="fu">rownames</span>(pca<span class="sc">$</span>x))</span>
<span id="cb1-179"><a href="#cb1-179"></a>    df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">cbind</span>(<span class="fu">colData</span>(rld), pca<span class="sc">$</span>x))</span>
<span id="cb1-180"><a href="#cb1-180"></a>    <span class="fu">ggplot</span>(df) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>PC1, <span class="at">y=</span>PC2, <span class="at">color =</span> voi))</span>
<span id="cb1-181"><a href="#cb1-181"></a>    </span>
<span id="cb1-182"><a href="#cb1-182"></a>    DESeq2<span class="sc">::</span><span class="fu">plotPCA</span>(rld, <span class="at">intgroup =</span> voi)</span>
<span id="cb1-183"><a href="#cb1-183"></a>    </span>
<span id="cb1-184"><a href="#cb1-184"></a>    pheatmap<span class="sc">::</span><span class="fu">pheatmap</span>(rld_cor, <span class="at">annotation =</span> y<span class="sc">$</span>samples[, <span class="fu">c</span>(voi), <span class="at">drop=</span>F])</span>
<span id="cb1-185"><a href="#cb1-185"></a>    </span>
<span id="cb1-186"><a href="#cb1-186"></a>    <span class="co"># Run DESeq2 differential expression analysis</span></span>
<span id="cb1-187"><a href="#cb1-187"></a>    <span class="co"># dds &lt;- DESeq(dds)</span></span>
<span id="cb1-188"><a href="#cb1-188"></a>    </span>
<span id="cb1-189"><a href="#cb1-189"></a>    design <span class="ot">&lt;-</span> <span class="fu">as.formula</span>(<span class="fu">paste</span>(<span class="st">"~"</span>, <span class="fu">paste</span>(variables, <span class="at">collapse=</span><span class="st">"+"</span>), <span class="st">"+"</span>, voi, <span class="at">sep=</span><span class="st">""</span>))</span>
<span id="cb1-190"><a href="#cb1-190"></a>    reduced <span class="ot">&lt;-</span> <span class="fu">as.formula</span>(<span class="fu">paste</span>(<span class="st">"~"</span>, <span class="fu">paste</span>(variables, <span class="at">collapse=</span><span class="st">"+"</span>), <span class="at">sep=</span><span class="st">""</span>))</span>
<span id="cb1-191"><a href="#cb1-191"></a>    </span>
<span id="cb1-192"><a href="#cb1-192"></a>    design</span>
<span id="cb1-193"><a href="#cb1-193"></a>    reduced</span>
<span id="cb1-194"><a href="#cb1-194"></a>    </span>
<span id="cb1-195"><a href="#cb1-195"></a>    <span class="co"># Run DESeq2 differential expression analysis</span></span>
<span id="cb1-196"><a href="#cb1-196"></a>    <span class="co"># Rather than evaluating whether a gene’s expression is up- or down-regulated </span></span>
<span id="cb1-197"><a href="#cb1-197"></a>    <span class="co"># in one class compared to another, the LRT identifies genes that are changing </span></span>
<span id="cb1-198"><a href="#cb1-198"></a>    <span class="co"># in expresssion in any direction across the different sample classes.</span></span>
<span id="cb1-199"><a href="#cb1-199"></a>    dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds, </span>
<span id="cb1-200"><a href="#cb1-200"></a>                 <span class="at">test=</span><span class="st">"LRT"</span>, </span>
<span id="cb1-201"><a href="#cb1-201"></a>                 <span class="at">full =</span> design,</span>
<span id="cb1-202"><a href="#cb1-202"></a>                 <span class="at">reduced =</span> reduced)</span>
<span id="cb1-203"><a href="#cb1-203"></a>    </span>
<span id="cb1-204"><a href="#cb1-204"></a>    <span class="co"># Plot dispersion estimates</span></span>
<span id="cb1-205"><a href="#cb1-205"></a>    <span class="fu">plotDispEsts</span>(dds)</span>
<span id="cb1-206"><a href="#cb1-206"></a>    </span>
<span id="cb1-207"><a href="#cb1-207"></a>    <span class="co"># Check the coefficients for the comparison</span></span>
<span id="cb1-208"><a href="#cb1-208"></a>    <span class="fu">print</span>(<span class="fu">resultsNames</span>(dds))</span>
<span id="cb1-209"><a href="#cb1-209"></a>    </span>
<span id="cb1-210"><a href="#cb1-210"></a>    <span class="fu">gene_ex_heatmap</span>(</span>
<span id="cb1-211"><a href="#cb1-211"></a>      dds,</span>
<span id="cb1-212"><a href="#cb1-212"></a>      <span class="at">voi =</span> voi,</span>
<span id="cb1-213"><a href="#cb1-213"></a>      <span class="at">variables =</span> variables,</span>
<span id="cb1-214"><a href="#cb1-214"></a>      rld,</span>
<span id="cb1-215"><a href="#cb1-215"></a>      <span class="at">cluster_rows =</span> F,</span>
<span id="cb1-216"><a href="#cb1-216"></a>      <span class="at">cluster_cols =</span> T</span>
<span id="cb1-217"><a href="#cb1-217"></a>    )</span>
<span id="cb1-218"><a href="#cb1-218"></a>  </span>
<span id="cb1-219"><a href="#cb1-219"></a>      <span class="co"># x &lt;- readline(prompt="Please choose a coefficient ")</span></span>
<span id="cb1-220"><a href="#cb1-220"></a>      </span>
<span id="cb1-221"><a href="#cb1-221"></a>      <span class="co"># res &lt;- results(dds,</span></span>
<span id="cb1-222"><a href="#cb1-222"></a>      <span class="co">#                name = voi,</span></span>
<span id="cb1-223"><a href="#cb1-223"></a>      <span class="co">#                alpha = 0.05)</span></span>
<span id="cb1-224"><a href="#cb1-224"></a>      </span>
<span id="cb1-225"><a href="#cb1-225"></a>      res <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">alpha =</span> <span class="fl">0.05</span>)</span>
<span id="cb1-226"><a href="#cb1-226"></a>      </span>
<span id="cb1-227"><a href="#cb1-227"></a>      <span class="co"># res &lt;- lfcShrink(dds,</span></span>
<span id="cb1-228"><a href="#cb1-228"></a>      <span class="co">#                  coef = voi,</span></span>
<span id="cb1-229"><a href="#cb1-229"></a>      <span class="co">#                  res=res,</span></span>
<span id="cb1-230"><a href="#cb1-230"></a>      <span class="co">#                  type = "apeglm")</span></span>
<span id="cb1-231"><a href="#cb1-231"></a>      </span>
<span id="cb1-232"><a href="#cb1-232"></a>      <span class="co"># Turn the results object into a tibble for use with tidyverse functions</span></span>
<span id="cb1-233"><a href="#cb1-233"></a>      res_tbl <span class="ot">&lt;-</span> </span>
<span id="cb1-234"><a href="#cb1-234"></a>        res <span class="sc">%&gt;%</span></span>
<span id="cb1-235"><a href="#cb1-235"></a>        <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb1-236"><a href="#cb1-236"></a>        <span class="fu">rownames_to_column</span>(<span class="at">var=</span><span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb1-237"><a href="#cb1-237"></a>        <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb1-238"><a href="#cb1-238"></a>        <span class="fu">arrange</span>(padj, pvalue)</span>
<span id="cb1-239"><a href="#cb1-239"></a>      </span>
<span id="cb1-240"><a href="#cb1-240"></a>      dds <span class="ot">&lt;-</span> res_tbl</span>
<span id="cb1-241"><a href="#cb1-241"></a>  }</span>
<span id="cb1-242"><a href="#cb1-242"></a>  <span class="fu">return</span>(dds)</span>
<span id="cb1-243"><a href="#cb1-243"></a>}</span>
<span id="cb1-244"><a href="#cb1-244"></a></span>
<span id="cb1-245"><a href="#cb1-245"></a>summary_df <span class="ot">&lt;-</span> <span class="cf">function</span>(x){</span>
<span id="cb1-246"><a href="#cb1-246"></a>  <span class="fu">data.frame</span>(<span class="at">less_0.05 =</span> <span class="fu">nrow</span>(<span class="fu">filter</span>(x, padj<span class="sc">&lt;</span><span class="fl">0.05</span>)),</span>
<span id="cb1-247"><a href="#cb1-247"></a>             <span class="at">less_0.10 =</span> <span class="fu">nrow</span>(<span class="fu">filter</span>(x, padj<span class="sc">&lt;</span><span class="fl">0.10</span>)),</span>
<span id="cb1-248"><a href="#cb1-248"></a>             <span class="at">less_0.15 =</span> <span class="fu">nrow</span>(<span class="fu">filter</span>(x, padj<span class="sc">&lt;</span><span class="fl">0.15</span>)),</span>
<span id="cb1-249"><a href="#cb1-249"></a>             <span class="at">less_0.20 =</span> <span class="fu">nrow</span>(<span class="fu">filter</span>(x, padj<span class="sc">&lt;</span><span class="fl">0.20</span>)),</span>
<span id="cb1-250"><a href="#cb1-250"></a>             <span class="at">less_0.25 =</span> <span class="fu">nrow</span>(<span class="fu">filter</span>(x, padj<span class="sc">&lt;</span><span class="fl">0.25</span>)))</span>
<span id="cb1-251"><a href="#cb1-251"></a>}</span>
<span id="cb1-252"><a href="#cb1-252"></a></span>
<span id="cb1-253"><a href="#cb1-253"></a>dea_qc <span class="ot">&lt;-</span> <span class="cf">function</span>(sce, voi, variables){</span>
<span id="cb1-254"><a href="#cb1-254"></a>  </span>
<span id="cb1-255"><a href="#cb1-255"></a>  countsToUse <span class="ot">&lt;-</span> <span class="fu">counts</span>(sce)</span>
<span id="cb1-256"><a href="#cb1-256"></a>  <span class="co">#colnames(countsToUse) &lt;- colData(sce)$ID</span></span>
<span id="cb1-257"><a href="#cb1-257"></a>  y <span class="ot">&lt;-</span> <span class="fu">DGEList</span>(countsToUse, <span class="at">samples =</span> <span class="fu">colData</span>(sce))</span>
<span id="cb1-258"><a href="#cb1-258"></a>  </span>
<span id="cb1-259"><a href="#cb1-259"></a>  <span class="cf">for</span> (i <span class="cf">in</span> variables) {</span>
<span id="cb1-260"><a href="#cb1-260"></a>    lose <span class="ot">&lt;-</span> <span class="fu">is.na</span>(y<span class="sc">$</span>samples[[i]])</span>
<span id="cb1-261"><a href="#cb1-261"></a>    y<span class="sc">$</span>samples <span class="ot">&lt;-</span> y<span class="sc">$</span>samples[<span class="sc">!</span>lose, ]</span>
<span id="cb1-262"><a href="#cb1-262"></a>    y<span class="sc">$</span>counts <span class="ot">&lt;-</span> y<span class="sc">$</span>counts[,<span class="fu">colnames</span>(y<span class="sc">$</span>counts) <span class="sc">%in%</span> <span class="fu">rownames</span>(y<span class="sc">$</span>samples)]</span>
<span id="cb1-263"><a href="#cb1-263"></a>  }</span>
<span id="cb1-264"><a href="#cb1-264"></a>  </span>
<span id="cb1-265"><a href="#cb1-265"></a>  lose <span class="ot">&lt;-</span> <span class="fu">is.na</span>(y<span class="sc">$</span>samples[[voi]])</span>
<span id="cb1-266"><a href="#cb1-266"></a>  y<span class="sc">$</span>samples <span class="ot">&lt;-</span> y<span class="sc">$</span>samples[<span class="sc">!</span>lose, ]</span>
<span id="cb1-267"><a href="#cb1-267"></a>  y<span class="sc">$</span>counts <span class="ot">&lt;-</span> y<span class="sc">$</span>counts[,<span class="fu">colnames</span>(y<span class="sc">$</span>counts) <span class="sc">%in%</span> <span class="fu">rownames</span>(y<span class="sc">$</span>samples)]</span>
<span id="cb1-268"><a href="#cb1-268"></a>  </span>
<span id="cb1-269"><a href="#cb1-269"></a>  discarded <span class="ot">&lt;-</span> y<span class="sc">$</span>samples<span class="sc">$</span>ncells <span class="sc">&lt;</span> <span class="dv">20</span></span>
<span id="cb1-270"><a href="#cb1-270"></a>  y <span class="ot">&lt;-</span> y[,<span class="sc">!</span>discarded]</span>
<span id="cb1-271"><a href="#cb1-271"></a>  <span class="fu">print</span>(<span class="fu">summary</span>(discarded))</span>
<span id="cb1-272"><a href="#cb1-272"></a>  </span>
<span id="cb1-273"><a href="#cb1-273"></a>  keep <span class="ot">&lt;-</span> <span class="fu">filterByExpr</span>(y, <span class="at">group=</span>y<span class="sc">$</span>samples[[voi]]) </span>
<span id="cb1-274"><a href="#cb1-274"></a>  y <span class="ot">&lt;-</span> y[keep, , keep.lib.sizes<span class="ot">=</span><span class="cn">FALSE</span>] </span>
<span id="cb1-275"><a href="#cb1-275"></a>  <span class="fu">print</span>(<span class="fu">summary</span>(keep))</span>
<span id="cb1-276"><a href="#cb1-276"></a>  </span>
<span id="cb1-277"><a href="#cb1-277"></a>  vars <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb1-278"><a href="#cb1-278"></a>  <span class="cf">for</span>(i <span class="cf">in</span> variables){</span>
<span id="cb1-279"><a href="#cb1-279"></a>    vars[i] <span class="ot">&lt;-</span> i</span>
<span id="cb1-280"><a href="#cb1-280"></a>  }</span>
<span id="cb1-281"><a href="#cb1-281"></a>  vars[[voi]] <span class="ot">&lt;-</span> voi</span>
<span id="cb1-282"><a href="#cb1-282"></a>  </span>
<span id="cb1-283"><a href="#cb1-283"></a>  naps <span class="ot">&lt;-</span> <span class="fu">map_lgl</span>(vars, <span class="sc">~</span>{<span class="fu">length</span>(<span class="fu">unique</span>(y<span class="sc">$</span>samples[[.x]]))<span class="sc">&gt;</span><span class="dv">1</span>})</span>
<span id="cb1-284"><a href="#cb1-284"></a>  </span>
<span id="cb1-285"><a href="#cb1-285"></a>  <span class="cf">if</span>(<span class="fu">any</span>(naps<span class="sc">==</span>F)){</span>
<span id="cb1-286"><a href="#cb1-286"></a>    <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb1-287"><a href="#cb1-287"></a>  }<span class="cf">else</span>{</span>
<span id="cb1-288"><a href="#cb1-288"></a>    <span class="fu">return</span>(y)</span>
<span id="cb1-289"><a href="#cb1-289"></a>  }</span>
<span id="cb1-290"><a href="#cb1-290"></a>  }</span>
<span id="cb1-291"><a href="#cb1-291"></a></span>
<span id="cb1-292"><a href="#cb1-292"></a><span class="co"># Read in data ------------------------------------------------------------</span></span>
<span id="cb1-293"><a href="#cb1-293"></a></span>
<span id="cb1-294"><a href="#cb1-294"></a>master <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"data/processed/master_seurat_obj.rds"</span>)</span>
<span id="cb1-295"><a href="#cb1-295"></a></span>
<span id="cb1-296"><a href="#cb1-296"></a><span class="co"># Code --------------------------------------------------------------------</span></span>
<span id="cb1-297"><a href="#cb1-297"></a></span>
<span id="cb1-298"><a href="#cb1-298"></a><span class="fu">set.seed</span>(<span class="dv">01001001</span>)</span>
<span id="cb1-299"><a href="#cb1-299"></a>bobs <span class="ot">&lt;-</span> <span class="fu">as.SingleCellExperiment</span>(master)</span>
<span id="cb1-300"><a href="#cb1-300"></a>sce <span class="ot">&lt;-</span> <span class="fu">aggregateAcrossCells</span>(bobs, <span class="at">id=</span><span class="fu">colData</span>(bobs)[,<span class="fu">c</span>(<span class="st">"ID"</span>)])</span>
<span id="cb1-301"><a href="#cb1-301"></a></span>
<span id="cb1-302"><a href="#cb1-302"></a>countsToUse <span class="ot">&lt;-</span> <span class="fu">counts</span>(sce)</span>
<span id="cb1-303"><a href="#cb1-303"></a><span class="fu">colnames</span>(countsToUse) <span class="ot">&lt;-</span> <span class="fu">colData</span>(sce)<span class="sc">$</span>ID</span>
<span id="cb1-304"><a href="#cb1-304"></a>y <span class="ot">&lt;-</span> <span class="fu">DGEList</span>(countsToUse, <span class="at">samples =</span> <span class="fu">colData</span>(sce))</span>
<span id="cb1-305"><a href="#cb1-305"></a><span class="co"># aggregated_results_all &lt;- list()</span></span>
<span id="cb1-306"><a href="#cb1-306"></a></span>
<span id="cb1-307"><a href="#cb1-307"></a></span>
<span id="cb1-308"><a href="#cb1-308"></a><span class="co"># Per cluster -------------------------------------------------------------</span></span>
<span id="cb1-309"><a href="#cb1-309"></a></span>
<span id="cb1-310"><a href="#cb1-310"></a>clusters <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">levels</span>(master<span class="sc">$</span>cluster_id))</span>
<span id="cb1-311"><a href="#cb1-311"></a>master_clusters <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb1-312"><a href="#cb1-312"></a><span class="cf">for</span>(i <span class="cf">in</span> clusters){</span>
<span id="cb1-313"><a href="#cb1-313"></a>  master_clusters[[i]] <span class="ot">&lt;-</span> <span class="fu">subset</span>(<span class="at">x =</span> master, <span class="at">subset =</span> cluster_id <span class="sc">==</span> i)</span>
<span id="cb1-314"><a href="#cb1-314"></a>}</span>
<span id="cb1-315"><a href="#cb1-315"></a>master_clusters <span class="ot">&lt;-</span> <span class="fu">map</span>(master_clusters, as.SingleCellExperiment)</span>
<span id="cb1-316"><a href="#cb1-316"></a></span>
<span id="cb1-317"><a href="#cb1-317"></a><span class="co"># nab &lt;- subset(x = master, subset = cluster_id == 0)</span></span>
<span id="cb1-318"><a href="#cb1-318"></a><span class="co"># babs &lt;- as.SingleCellExperiment(nab)</span></span>
<span id="cb1-319"><a href="#cb1-319"></a><span class="co"># sce &lt;- aggregateAcrossCells(bobs, id=colData(bobs)[,c("ID")])</span></span>
<span id="cb1-320"><a href="#cb1-320"></a></span>
<span id="cb1-321"><a href="#cb1-321"></a>sce <span class="ot">&lt;-</span> <span class="fu">map</span>(master_clusters, <span class="sc">~</span><span class="fu">aggregateAcrossCells</span>(.x, <span class="at">id=</span><span class="fu">colData</span>(.x)[,<span class="fu">c</span>(<span class="st">"ID"</span>)]))</span>
<span id="cb1-322"><a href="#cb1-322"></a><span class="fu">names</span>(sce) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"cluster_"</span>, <span class="fu">names</span>(sce))</span>
<span id="cb1-323"><a href="#cb1-323"></a></span>
<span id="cb1-324"><a href="#cb1-324"></a></span>
<span id="cb1-325"><a href="#cb1-325"></a><span class="co"># Continuous analyses -----------------------------------------------------</span></span>
<span id="cb1-326"><a href="#cb1-326"></a></span>
<span id="cb1-327"><a href="#cb1-327"></a>ark <span class="ot">&lt;-</span> </span>
<span id="cb1-328"><a href="#cb1-328"></a>  <span class="fu">map</span>(sce, <span class="sc">~</span><span class="fu">dea_qc</span>(</span>
<span id="cb1-329"><a href="#cb1-329"></a>    .x,</span>
<span id="cb1-330"><a href="#cb1-330"></a>    <span class="at">voi =</span> <span class="st">"almhsq_scaled"</span>,</span>
<span id="cb1-331"><a href="#cb1-331"></a>    <span class="at">variables =</span> <span class="fu">c</span>(<span class="st">"Sex"</span>, <span class="st">"age_scaled"</span>)</span>
<span id="cb1-332"><a href="#cb1-332"></a>  )) <span class="sc">%&gt;%</span> </span>
<span id="cb1-333"><a href="#cb1-333"></a>  <span class="fu">discard</span>(is.null) <span class="sc">%&gt;%</span> </span>
<span id="cb1-334"><a href="#cb1-334"></a>  <span class="fu">map</span>(<span class="sc">~</span><span class="fu">dea</span>(</span>
<span id="cb1-335"><a href="#cb1-335"></a>    .x,</span>
<span id="cb1-336"><a href="#cb1-336"></a>    <span class="at">voi =</span> <span class="st">"almhsq_scaled"</span>,</span>
<span id="cb1-337"><a href="#cb1-337"></a>    <span class="at">variables =</span> <span class="fu">c</span>(<span class="st">"Sex"</span>, <span class="st">"age_scaled"</span>),</span>
<span id="cb1-338"><a href="#cb1-338"></a>    <span class="at">variable_format =</span> continuous</span>
<span id="cb1-339"><a href="#cb1-339"></a>  ))</span>
<span id="cb1-340"><a href="#cb1-340"></a></span>
<span id="cb1-341"><a href="#cb1-341"></a>almi_cont_results <span class="ot">&lt;-</span> ark</span>
<span id="cb1-342"><a href="#cb1-342"></a></span>
<span id="cb1-343"><a href="#cb1-343"></a>almi_cont <span class="ot">&lt;-</span> <span class="fu">map</span>(ark, summary_df) <span class="sc">%&gt;%</span> </span>
<span id="cb1-344"><a href="#cb1-344"></a>  <span class="fu">bind_rows</span>(<span class="at">.id=</span><span class="st">"cluster"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1-345"><a href="#cb1-345"></a>  <span class="fu">mutate</span>(<span class="at">analysis =</span> <span class="st">"almi_cont"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1-346"><a href="#cb1-346"></a>  <span class="fu">relocate</span>(analysis, <span class="at">.before =</span> <span class="fu">everything</span>())</span>
<span id="cb1-347"><a href="#cb1-347"></a></span>
<span id="cb1-348"><a href="#cb1-348"></a></span>
<span id="cb1-349"><a href="#cb1-349"></a>ark <span class="ot">&lt;-</span> </span>
<span id="cb1-350"><a href="#cb1-350"></a>  <span class="fu">map</span>(sce, <span class="sc">~</span><span class="fu">dea_qc</span>(</span>
<span id="cb1-351"><a href="#cb1-351"></a>    .x,</span>
<span id="cb1-352"><a href="#cb1-352"></a>    <span class="at">voi =</span> <span class="st">"xamaxgrip_scaled"</span>,</span>
<span id="cb1-353"><a href="#cb1-353"></a>    <span class="at">variables =</span> <span class="fu">c</span>(<span class="st">"Sex"</span>, <span class="st">"age_scaled"</span>)</span>
<span id="cb1-354"><a href="#cb1-354"></a>  )) <span class="sc">%&gt;%</span> </span>
<span id="cb1-355"><a href="#cb1-355"></a>  <span class="fu">discard</span>(is.null) <span class="sc">%&gt;%</span> </span>
<span id="cb1-356"><a href="#cb1-356"></a>  <span class="fu">map</span>(<span class="sc">~</span><span class="fu">dea</span>(</span>
<span id="cb1-357"><a href="#cb1-357"></a>    .x,</span>
<span id="cb1-358"><a href="#cb1-358"></a>    <span class="at">voi =</span> <span class="st">"xamaxgrip_scaled"</span>,</span>
<span id="cb1-359"><a href="#cb1-359"></a>    <span class="at">variables =</span> <span class="fu">c</span>(<span class="st">"Sex"</span>, <span class="st">"age_scaled"</span>),</span>
<span id="cb1-360"><a href="#cb1-360"></a>    <span class="at">variable_format =</span> continuous</span>
<span id="cb1-361"><a href="#cb1-361"></a>  ))</span>
<span id="cb1-362"><a href="#cb1-362"></a></span>
<span id="cb1-363"><a href="#cb1-363"></a>grip_cont_results <span class="ot">&lt;-</span> ark</span>
<span id="cb1-364"><a href="#cb1-364"></a></span>
<span id="cb1-365"><a href="#cb1-365"></a>grip_cont <span class="ot">&lt;-</span> <span class="fu">map</span>(ark, summary_df) <span class="sc">%&gt;%</span> </span>
<span id="cb1-366"><a href="#cb1-366"></a>  <span class="fu">bind_rows</span>(<span class="at">.id=</span><span class="st">"cluster"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1-367"><a href="#cb1-367"></a>  <span class="fu">mutate</span>(<span class="at">analysis =</span> <span class="st">"grip_cont"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1-368"><a href="#cb1-368"></a>  <span class="fu">relocate</span>(analysis, <span class="at">.before =</span> <span class="fu">everything</span>())</span>
<span id="cb1-369"><a href="#cb1-369"></a></span>
<span id="cb1-370"><a href="#cb1-370"></a></span>
<span id="cb1-371"><a href="#cb1-371"></a>ark <span class="ot">&lt;-</span> </span>
<span id="cb1-372"><a href="#cb1-372"></a>  <span class="fu">map</span>(sce, <span class="sc">~</span><span class="fu">dea_qc</span>(</span>
<span id="cb1-373"><a href="#cb1-373"></a>    .x,</span>
<span id="cb1-374"><a href="#cb1-374"></a>    <span class="at">voi =</span> <span class="st">"xwspdms_scaled"</span>,</span>
<span id="cb1-375"><a href="#cb1-375"></a>    <span class="at">variables =</span> <span class="fu">c</span>(<span class="st">"Sex"</span>, <span class="st">"age_scaled"</span>)</span>
<span id="cb1-376"><a href="#cb1-376"></a>  )) <span class="sc">%&gt;%</span> </span>
<span id="cb1-377"><a href="#cb1-377"></a>  <span class="fu">discard</span>(is.null) <span class="sc">%&gt;%</span> </span>
<span id="cb1-378"><a href="#cb1-378"></a>  <span class="fu">map</span>(<span class="sc">~</span><span class="fu">dea</span>(</span>
<span id="cb1-379"><a href="#cb1-379"></a>    .x,</span>
<span id="cb1-380"><a href="#cb1-380"></a>    <span class="at">voi =</span> <span class="st">"xwspdms_scaled"</span>,</span>
<span id="cb1-381"><a href="#cb1-381"></a>    <span class="at">variables =</span> <span class="fu">c</span>(<span class="st">"Sex"</span>, <span class="st">"age_scaled"</span>),</span>
<span id="cb1-382"><a href="#cb1-382"></a>    <span class="at">variable_format =</span> continuous</span>
<span id="cb1-383"><a href="#cb1-383"></a>  ))</span>
<span id="cb1-384"><a href="#cb1-384"></a></span>
<span id="cb1-385"><a href="#cb1-385"></a>gait_cont_results <span class="ot">&lt;-</span> ark</span>
<span id="cb1-386"><a href="#cb1-386"></a></span>
<span id="cb1-387"><a href="#cb1-387"></a>gait_cont <span class="ot">&lt;-</span> </span>
<span id="cb1-388"><a href="#cb1-388"></a>  <span class="fu">map</span>(ark, summary_df) <span class="sc">%&gt;%</span> </span>
<span id="cb1-389"><a href="#cb1-389"></a>  <span class="fu">bind_rows</span>(<span class="at">.id=</span><span class="st">"cluster"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1-390"><a href="#cb1-390"></a>  <span class="fu">mutate</span>(<span class="at">analysis =</span> <span class="st">"gait_cont"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1-391"><a href="#cb1-391"></a>  <span class="fu">relocate</span>(analysis, <span class="at">.before =</span> <span class="fu">everything</span>())</span>
<span id="cb1-392"><a href="#cb1-392"></a></span>
<span id="cb1-393"><a href="#cb1-393"></a></span>
<span id="cb1-394"><a href="#cb1-394"></a></span>
<span id="cb1-395"><a href="#cb1-395"></a><span class="co"># Quartiile analyses ------------------------------------------------------</span></span>
<span id="cb1-396"><a href="#cb1-396"></a></span>
<span id="cb1-397"><a href="#cb1-397"></a>ark <span class="ot">&lt;-</span> </span>
<span id="cb1-398"><a href="#cb1-398"></a>  <span class="fu">map</span>(sce, <span class="sc">~</span><span class="fu">dea_qc</span>(</span>
<span id="cb1-399"><a href="#cb1-399"></a>    .x,</span>
<span id="cb1-400"><a href="#cb1-400"></a>    <span class="at">voi =</span> <span class="st">"almhsq_quart"</span>,</span>
<span id="cb1-401"><a href="#cb1-401"></a>    <span class="at">variables =</span> <span class="fu">c</span>(<span class="st">"Sex"</span>, <span class="st">"age_scaled"</span>)</span>
<span id="cb1-402"><a href="#cb1-402"></a>  )) </span>
<span id="cb1-403"><a href="#cb1-403"></a>  </span>
<span id="cb1-404"><a href="#cb1-404"></a>  </span>
<span id="cb1-405"><a href="#cb1-405"></a><span class="co">#ark$cluster_6 &lt;- NULL </span></span>
<span id="cb1-406"><a href="#cb1-406"></a>ark<span class="sc">$</span>cluster_9 <span class="ot">&lt;-</span> <span class="cn">NULL</span> </span>
<span id="cb1-407"><a href="#cb1-407"></a></span>
<span id="cb1-408"><a href="#cb1-408"></a>ark_fin <span class="ot">&lt;-</span> </span>
<span id="cb1-409"><a href="#cb1-409"></a>  ark <span class="sc">%&gt;%</span> </span>
<span id="cb1-410"><a href="#cb1-410"></a>  <span class="fu">discard</span>(is.null) <span class="sc">%&gt;%</span> </span>
<span id="cb1-411"><a href="#cb1-411"></a>  <span class="fu">map</span>(<span class="sc">~</span><span class="fu">dea</span>(</span>
<span id="cb1-412"><a href="#cb1-412"></a>    .x,</span>
<span id="cb1-413"><a href="#cb1-413"></a>    <span class="at">voi =</span> <span class="st">"almhsq_quart"</span>,</span>
<span id="cb1-414"><a href="#cb1-414"></a>    <span class="at">variables =</span> <span class="fu">c</span>(<span class="st">"Sex"</span>, <span class="st">"age_scaled"</span>),</span>
<span id="cb1-415"><a href="#cb1-415"></a>    <span class="at">wald =</span> F</span>
<span id="cb1-416"><a href="#cb1-416"></a>  ))</span>
<span id="cb1-417"><a href="#cb1-417"></a></span>
<span id="cb1-418"><a href="#cb1-418"></a>almi_quart_results <span class="ot">&lt;-</span> ark_fin</span>
<span id="cb1-419"><a href="#cb1-419"></a></span>
<span id="cb1-420"><a href="#cb1-420"></a>almi_quart <span class="ot">&lt;-</span> <span class="fu">map</span>(ark_fin, summary_df) <span class="sc">%&gt;%</span> </span>
<span id="cb1-421"><a href="#cb1-421"></a>  <span class="fu">bind_rows</span>(<span class="at">.id=</span><span class="st">"cluster"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1-422"><a href="#cb1-422"></a>  <span class="fu">mutate</span>(<span class="at">analysis =</span> <span class="st">"almi_quart"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1-423"><a href="#cb1-423"></a>  <span class="fu">relocate</span>(analysis, <span class="at">.before =</span> <span class="fu">everything</span>())</span>
<span id="cb1-424"><a href="#cb1-424"></a></span>
<span id="cb1-425"><a href="#cb1-425"></a></span>
<span id="cb1-426"><a href="#cb1-426"></a>ark <span class="ot">&lt;-</span> </span>
<span id="cb1-427"><a href="#cb1-427"></a>  <span class="fu">map</span>(sce, <span class="sc">~</span><span class="fu">dea_qc</span>(</span>
<span id="cb1-428"><a href="#cb1-428"></a>    .x,</span>
<span id="cb1-429"><a href="#cb1-429"></a>    <span class="at">voi =</span> <span class="st">"xamaxgrip_quart"</span>,</span>
<span id="cb1-430"><a href="#cb1-430"></a>    <span class="at">variables =</span> <span class="fu">c</span>(<span class="st">"Sex"</span>, <span class="st">"age_scaled"</span>)</span>
<span id="cb1-431"><a href="#cb1-431"></a>  ))</span>
<span id="cb1-432"><a href="#cb1-432"></a></span>
<span id="cb1-433"><a href="#cb1-433"></a>ark<span class="sc">$</span>cluster_9 <span class="ot">&lt;-</span> <span class="cn">NULL</span> </span>
<span id="cb1-434"><a href="#cb1-434"></a></span>
<span id="cb1-435"><a href="#cb1-435"></a>ark_fin <span class="ot">&lt;-</span> </span>
<span id="cb1-436"><a href="#cb1-436"></a>  ark <span class="sc">%&gt;%</span> </span>
<span id="cb1-437"><a href="#cb1-437"></a>  <span class="fu">discard</span>(is.null) <span class="sc">%&gt;%</span> </span>
<span id="cb1-438"><a href="#cb1-438"></a>  <span class="fu">map</span>(<span class="sc">~</span><span class="fu">dea</span>(</span>
<span id="cb1-439"><a href="#cb1-439"></a>    .x,</span>
<span id="cb1-440"><a href="#cb1-440"></a>    <span class="at">voi =</span> <span class="st">"xamaxgrip_quart"</span>,</span>
<span id="cb1-441"><a href="#cb1-441"></a>    <span class="at">variables =</span> <span class="fu">c</span>(<span class="st">"Sex"</span>, <span class="st">"age_scaled"</span>),</span>
<span id="cb1-442"><a href="#cb1-442"></a>    <span class="at">wald =</span> F</span>
<span id="cb1-443"><a href="#cb1-443"></a>  ))</span>
<span id="cb1-444"><a href="#cb1-444"></a></span>
<span id="cb1-445"><a href="#cb1-445"></a>grip_quart_results <span class="ot">&lt;-</span> ark_fin</span>
<span id="cb1-446"><a href="#cb1-446"></a></span>
<span id="cb1-447"><a href="#cb1-447"></a>grip_quart <span class="ot">&lt;-</span> <span class="fu">map</span>(ark_fin, summary_df) <span class="sc">%&gt;%</span> </span>
<span id="cb1-448"><a href="#cb1-448"></a>  <span class="fu">bind_rows</span>(<span class="at">.id=</span><span class="st">"cluster"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1-449"><a href="#cb1-449"></a>  <span class="fu">mutate</span>(<span class="at">analysis =</span> <span class="st">"grip_quart"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1-450"><a href="#cb1-450"></a>  <span class="fu">relocate</span>(analysis, <span class="at">.before =</span> <span class="fu">everything</span>())</span>
<span id="cb1-451"><a href="#cb1-451"></a></span>
<span id="cb1-452"><a href="#cb1-452"></a></span>
<span id="cb1-453"><a href="#cb1-453"></a>ark <span class="ot">&lt;-</span> </span>
<span id="cb1-454"><a href="#cb1-454"></a>  <span class="fu">map</span>(sce, <span class="sc">~</span><span class="fu">dea_qc</span>(</span>
<span id="cb1-455"><a href="#cb1-455"></a>    .x,</span>
<span id="cb1-456"><a href="#cb1-456"></a>    <span class="at">voi =</span> <span class="st">"xwspdms_quart"</span>,</span>
<span id="cb1-457"><a href="#cb1-457"></a>    <span class="at">variables =</span> <span class="fu">c</span>(<span class="st">"Sex"</span>, <span class="st">"age_scaled"</span>)</span>
<span id="cb1-458"><a href="#cb1-458"></a>  ))</span>
<span id="cb1-459"><a href="#cb1-459"></a></span>
<span id="cb1-460"><a href="#cb1-460"></a>ark<span class="sc">$</span>cluster_9 <span class="ot">&lt;-</span> <span class="cn">NULL</span> </span>
<span id="cb1-461"><a href="#cb1-461"></a></span>
<span id="cb1-462"><a href="#cb1-462"></a>ark_fin <span class="ot">&lt;-</span> </span>
<span id="cb1-463"><a href="#cb1-463"></a>  ark <span class="sc">%&gt;%</span> </span>
<span id="cb1-464"><a href="#cb1-464"></a>  <span class="fu">discard</span>(is.null) <span class="sc">%&gt;%</span> </span>
<span id="cb1-465"><a href="#cb1-465"></a>  <span class="fu">map</span>(<span class="sc">~</span><span class="fu">dea</span>(</span>
<span id="cb1-466"><a href="#cb1-466"></a>    .x,</span>
<span id="cb1-467"><a href="#cb1-467"></a>    <span class="at">voi =</span> <span class="st">"xwspdms_quart"</span>,</span>
<span id="cb1-468"><a href="#cb1-468"></a>    <span class="at">variables =</span> <span class="fu">c</span>(<span class="st">"Sex"</span>, <span class="st">"age_scaled"</span>),</span>
<span id="cb1-469"><a href="#cb1-469"></a>    <span class="at">wald =</span> F</span>
<span id="cb1-470"><a href="#cb1-470"></a>  ))</span>
<span id="cb1-471"><a href="#cb1-471"></a></span>
<span id="cb1-472"><a href="#cb1-472"></a>gait_quart_results <span class="ot">&lt;-</span> ark_fin</span>
<span id="cb1-473"><a href="#cb1-473"></a></span>
<span id="cb1-474"><a href="#cb1-474"></a>gait_quart <span class="ot">&lt;-</span> <span class="fu">map</span>(ark_fin, summary_df) <span class="sc">%&gt;%</span> </span>
<span id="cb1-475"><a href="#cb1-475"></a>  <span class="fu">bind_rows</span>(<span class="at">.id=</span><span class="st">"cluster"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1-476"><a href="#cb1-476"></a>  <span class="fu">mutate</span>(<span class="at">analysis =</span> <span class="st">"gait_quart"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1-477"><a href="#cb1-477"></a>  <span class="fu">relocate</span>(analysis, <span class="at">.before =</span> <span class="fu">everything</span>())</span>
<span id="cb1-478"><a href="#cb1-478"></a></span>
<span id="cb1-479"><a href="#cb1-479"></a></span>
<span id="cb1-480"><a href="#cb1-480"></a></span>
<span id="cb1-481"><a href="#cb1-481"></a><span class="co"># Sarcopenia --------------------------------------------------------------</span></span>
<span id="cb1-482"><a href="#cb1-482"></a></span>
<span id="cb1-483"><a href="#cb1-483"></a>ark <span class="ot">&lt;-</span> </span>
<span id="cb1-484"><a href="#cb1-484"></a>  <span class="fu">map</span>(sce, <span class="sc">~</span><span class="fu">dea_qc</span>(</span>
<span id="cb1-485"><a href="#cb1-485"></a>    .x,</span>
<span id="cb1-486"><a href="#cb1-486"></a>    <span class="at">voi =</span> <span class="st">"status_binary"</span>,</span>
<span id="cb1-487"><a href="#cb1-487"></a>    <span class="at">variables =</span> <span class="fu">c</span>(<span class="st">"Sex"</span>, <span class="st">"age_scaled"</span>)</span>
<span id="cb1-488"><a href="#cb1-488"></a>  )) </span>
<span id="cb1-489"><a href="#cb1-489"></a></span>
<span id="cb1-490"><a href="#cb1-490"></a>ark<span class="sc">$</span>cluster_9 <span class="ot">&lt;-</span> <span class="cn">NULL</span> </span>
<span id="cb1-491"><a href="#cb1-491"></a></span>
<span id="cb1-492"><a href="#cb1-492"></a>ark_fin <span class="ot">&lt;-</span> </span>
<span id="cb1-493"><a href="#cb1-493"></a>ark <span class="sc">%&gt;%</span> </span>
<span id="cb1-494"><a href="#cb1-494"></a>  <span class="fu">discard</span>(is.null) <span class="sc">%&gt;%</span> </span>
<span id="cb1-495"><a href="#cb1-495"></a>  <span class="fu">map</span>(<span class="sc">~</span><span class="fu">dea</span>(</span>
<span id="cb1-496"><a href="#cb1-496"></a>    .x,</span>
<span id="cb1-497"><a href="#cb1-497"></a>    <span class="at">voi =</span> <span class="st">"status_binary"</span>,</span>
<span id="cb1-498"><a href="#cb1-498"></a>    <span class="at">variables =</span> <span class="fu">c</span>(<span class="st">"Sex"</span>, <span class="st">"age_scaled"</span>),</span>
<span id="cb1-499"><a href="#cb1-499"></a>    <span class="at">variable_format =</span> catagorical</span>
<span id="cb1-500"><a href="#cb1-500"></a>  ))</span>
<span id="cb1-501"><a href="#cb1-501"></a></span>
<span id="cb1-502"><a href="#cb1-502"></a>status_binary_results <span class="ot">&lt;-</span> ark_fin</span>
<span id="cb1-503"><a href="#cb1-503"></a></span>
<span id="cb1-504"><a href="#cb1-504"></a>status_binary <span class="ot">&lt;-</span> <span class="fu">map</span>(ark_fin, summary_df) <span class="sc">%&gt;%</span> </span>
<span id="cb1-505"><a href="#cb1-505"></a>  <span class="fu">bind_rows</span>(<span class="at">.id=</span><span class="st">"cluster"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1-506"><a href="#cb1-506"></a>  <span class="fu">mutate</span>(<span class="at">analysis =</span> <span class="st">"status_binary"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1-507"><a href="#cb1-507"></a>  <span class="fu">relocate</span>(analysis, <span class="at">.before =</span> <span class="fu">everything</span>())</span>
<span id="cb1-508"><a href="#cb1-508"></a></span>
<span id="cb1-509"><a href="#cb1-509"></a></span>
<span id="cb1-510"><a href="#cb1-510"></a>final <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(almi_cont, </span>
<span id="cb1-511"><a href="#cb1-511"></a>                   gait_cont, </span>
<span id="cb1-512"><a href="#cb1-512"></a>                   grip_cont, </span>
<span id="cb1-513"><a href="#cb1-513"></a>                   almi_quart, </span>
<span id="cb1-514"><a href="#cb1-514"></a>                   gait_quart, </span>
<span id="cb1-515"><a href="#cb1-515"></a>                   grip_quart, </span>
<span id="cb1-516"><a href="#cb1-516"></a>                   status_binary)</span>
<span id="cb1-517"><a href="#cb1-517"></a></span>
<span id="cb1-518"><a href="#cb1-518"></a><span class="fu">write_csv</span>(final, <span class="st">"results/22-08-10-pseudobulk_by_cluster.csv"</span>)</span>
<span id="cb1-519"><a href="#cb1-519"></a></span>
<span id="cb1-520"><a href="#cb1-520"></a></span>
<span id="cb1-521"><a href="#cb1-521"></a></span>
<span id="cb1-522"><a href="#cb1-522"></a><span class="co"># Pulling &lt;0.05 -----------------------------------------------------------</span></span>
<span id="cb1-523"><a href="#cb1-523"></a></span>
<span id="cb1-524"><a href="#cb1-524"></a>pull_sigs <span class="ot">&lt;-</span> <span class="cf">function</span>(results_list, analysis){</span>
<span id="cb1-525"><a href="#cb1-525"></a>  marf <span class="ot">&lt;-</span> </span>
<span id="cb1-526"><a href="#cb1-526"></a>    <span class="fu">map</span>(results_list, <span class="sc">~</span>{.x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(padj<span class="sc">&lt;</span><span class="fl">0.05</span>)}) <span class="sc">%&gt;%</span> </span>
<span id="cb1-527"><a href="#cb1-527"></a>    <span class="fu">bind_rows</span>(<span class="at">.id=</span><span class="st">"cluster"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1-528"><a href="#cb1-528"></a>    <span class="fu">mutate</span>(<span class="at">analysis =</span> analysis) <span class="sc">%&gt;%</span> </span>
<span id="cb1-529"><a href="#cb1-529"></a>    <span class="fu">relocate</span>(analysis, <span class="at">.before =</span> <span class="fu">everything</span>())</span>
<span id="cb1-530"><a href="#cb1-530"></a>  <span class="fu">return</span>(marf)</span>
<span id="cb1-531"><a href="#cb1-531"></a>}</span>
<span id="cb1-532"><a href="#cb1-532"></a></span>
<span id="cb1-533"><a href="#cb1-533"></a>gene_list_almi_cont <span class="ot">&lt;-</span> <span class="fu">pull_sigs</span>(almi_cont_results, <span class="st">"almi_continuous"</span>)</span>
<span id="cb1-534"><a href="#cb1-534"></a>gene_list_gait_cont <span class="ot">&lt;-</span> <span class="fu">pull_sigs</span>(gait_cont_results, <span class="st">"gait_continuous"</span>)</span>
<span id="cb1-535"><a href="#cb1-535"></a>gene_list_grip_cont <span class="ot">&lt;-</span> <span class="fu">pull_sigs</span>(grip_cont_results, <span class="st">"grip_continuous"</span>)</span>
<span id="cb1-536"><a href="#cb1-536"></a>gene_list_almi_quart <span class="ot">&lt;-</span> <span class="fu">pull_sigs</span>(almi_quart_results, <span class="st">"almi_quartiles"</span>)</span>
<span id="cb1-537"><a href="#cb1-537"></a>gene_list_gait_quart <span class="ot">&lt;-</span> <span class="fu">pull_sigs</span>(gait_quart_results, <span class="st">"gait_quartiles"</span>)</span>
<span id="cb1-538"><a href="#cb1-538"></a>gene_list_grip_quart <span class="ot">&lt;-</span> <span class="fu">pull_sigs</span>(grip_quart_results, <span class="st">"grip_quartiles"</span>)</span>
<span id="cb1-539"><a href="#cb1-539"></a>gene_list_status_binary <span class="ot">&lt;-</span> <span class="fu">pull_sigs</span>(status_binary_results, <span class="st">"status_binary"</span>)</span>
<span id="cb1-540"><a href="#cb1-540"></a></span>
<span id="cb1-541"><a href="#cb1-541"></a>final_gene_results_list <span class="ot">&lt;-</span> </span>
<span id="cb1-542"><a href="#cb1-542"></a><span class="fu">bind_rows</span>(</span>
<span id="cb1-543"><a href="#cb1-543"></a>gene_list_almi_cont,</span>
<span id="cb1-544"><a href="#cb1-544"></a>gene_list_gait_cont, </span>
<span id="cb1-545"><a href="#cb1-545"></a>gene_list_grip_cont, </span>
<span id="cb1-546"><a href="#cb1-546"></a>gene_list_almi_quart, </span>
<span id="cb1-547"><a href="#cb1-547"></a>gene_list_gait_quart, </span>
<span id="cb1-548"><a href="#cb1-548"></a>gene_list_grip_quart, </span>
<span id="cb1-549"><a href="#cb1-549"></a>gene_list_status_binary</span>
<span id="cb1-550"><a href="#cb1-550"></a>)</span>
<span id="cb1-551"><a href="#cb1-551"></a></span>
<span id="cb1-552"><a href="#cb1-552"></a>clipr<span class="sc">::</span><span class="fu">write_clip</span>(final_gene_results_list)</span>
<span id="cb1-553"><a href="#cb1-553"></a></span>
<span id="cb1-554"><a href="#cb1-554"></a></span>
<span id="cb1-555"><a href="#cb1-555"></a></span>
<span id="cb1-556"><a href="#cb1-556"></a><span class="co"># function(sce, voi){</span></span>
<span id="cb1-557"><a href="#cb1-557"></a><span class="co">#   countsToUse &lt;- counts[[sce]]</span></span>
<span id="cb1-558"><a href="#cb1-558"></a><span class="co">#   colnames(countsToUse) &lt;- colData(sce)[[ID]]</span></span>
<span id="cb1-559"><a href="#cb1-559"></a><span class="co">#   y &lt;- DGEList(countsToUse, samples=colData(sce))</span></span>
<span id="cb1-560"><a href="#cb1-560"></a><span class="co">#   </span></span>
<span id="cb1-561"><a href="#cb1-561"></a><span class="co">#   lose &lt;- is.na(y$samples[[xwspdms]])</span></span>
<span id="cb1-562"><a href="#cb1-562"></a><span class="co">#   y$samples &lt;- y$samples[!lose, ]</span></span>
<span id="cb1-563"><a href="#cb1-563"></a><span class="co">#   y$counts &lt;- y$counts[,colnames(y$counts) %in% rownames(y$samples)]</span></span>
<span id="cb1-564"><a href="#cb1-564"></a><span class="co">#   </span></span>
<span id="cb1-565"><a href="#cb1-565"></a><span class="co">#   lose &lt;- is.na(y$samples[[Sex]])</span></span>
<span id="cb1-566"><a href="#cb1-566"></a><span class="co">#   y$samples &lt;- y$samples[!lose, ]</span></span>
<span id="cb1-567"><a href="#cb1-567"></a><span class="co">#   y$counts &lt;- y$counts[,colnames(y$counts) %in% rownames(y$samples)]</span></span>
<span id="cb1-568"><a href="#cb1-568"></a><span class="co">#   </span></span>
<span id="cb1-569"><a href="#cb1-569"></a><span class="co">#   lose &lt;- is.na(y$samples[[age]])</span></span>
<span id="cb1-570"><a href="#cb1-570"></a><span class="co">#   y$samples &lt;- y$samples[!lose, ]</span></span>
<span id="cb1-571"><a href="#cb1-571"></a><span class="co">#   y$counts &lt;- y$counts[,colnames(y$counts) %in% rownames(y$samples)]</span></span>
<span id="cb1-572"><a href="#cb1-572"></a><span class="co">#   </span></span>
<span id="cb1-573"><a href="#cb1-573"></a><span class="co">#   # design &lt;- model.matrix(~xwspdms+Sex+age, y$samples)</span></span>
<span id="cb1-574"><a href="#cb1-574"></a><span class="co">#   # design</span></span>
<span id="cb1-575"><a href="#cb1-575"></a><span class="co">#   </span></span>
<span id="cb1-576"><a href="#cb1-576"></a><span class="co">#   discarded &lt;- y$samples$ncells &lt; 20</span></span>
<span id="cb1-577"><a href="#cb1-577"></a><span class="co">#   y &lt;- y[,!discarded]</span></span>
<span id="cb1-578"><a href="#cb1-578"></a><span class="co">#   summary(discarded)</span></span>
<span id="cb1-579"><a href="#cb1-579"></a><span class="co">#   </span></span>
<span id="cb1-580"><a href="#cb1-580"></a><span class="co">#   #keep &lt;- filterByExpr(y, design=design)    # 11420 removed</span></span>
<span id="cb1-581"><a href="#cb1-581"></a><span class="co">#   keep &lt;- filterByExpr(y, group=y$samples[[xwspdms]])    # 11420 removed</span></span>
<span id="cb1-582"><a href="#cb1-582"></a><span class="co">#   y &lt;- y[keep, , keep.lib.sizes=FALSE] </span></span>
<span id="cb1-583"><a href="#cb1-583"></a><span class="co">#   summary(keep)</span></span>
<span id="cb1-584"><a href="#cb1-584"></a><span class="co">#   </span></span>
<span id="cb1-585"><a href="#cb1-585"></a><span class="co">#   # y &lt;- calcNormFactors(y)</span></span>
<span id="cb1-586"><a href="#cb1-586"></a><span class="co">#   # y$samples</span></span>
<span id="cb1-587"><a href="#cb1-587"></a><span class="co">#   </span></span>
<span id="cb1-588"><a href="#cb1-588"></a><span class="co">#   # par(mfrow=c(2,4))</span></span>
<span id="cb1-589"><a href="#cb1-589"></a><span class="co">#   # for (i in seq_len(ncol(y))) {</span></span>
<span id="cb1-590"><a href="#cb1-590"></a><span class="co">#   #   plotMD(y, column=i)</span></span>
<span id="cb1-591"><a href="#cb1-591"></a><span class="co">#   # }</span></span>
<span id="cb1-592"><a href="#cb1-592"></a><span class="co">#   </span></span>
<span id="cb1-593"><a href="#cb1-593"></a><span class="co">#   # y$samples$xwspdms &lt;- factor(y$samples$xwspdms)</span></span>
<span id="cb1-594"><a href="#cb1-594"></a><span class="co">#   limma::plotMDS(cpm(y, log=TRUE), col = as.numeric(y$samples[[xwspdms]]))</span></span>
<span id="cb1-595"><a href="#cb1-595"></a><span class="co">#   </span></span>
<span id="cb1-596"><a href="#cb1-596"></a><span class="co">#   # y &lt;- estimateDisp(y, design)</span></span>
<span id="cb1-597"><a href="#cb1-597"></a><span class="co">#   # summary(y$trended.dispersion)</span></span>
<span id="cb1-598"><a href="#cb1-598"></a><span class="co">#   # </span></span>
<span id="cb1-599"><a href="#cb1-599"></a><span class="co">#   # plotBCV(y)</span></span>
<span id="cb1-600"><a href="#cb1-600"></a><span class="co">#   </span></span>
<span id="cb1-601"><a href="#cb1-601"></a><span class="co">#   # GLM dispersions ---------------------------------------------------------</span></span>
<span id="cb1-602"><a href="#cb1-602"></a><span class="co">#   </span></span>
<span id="cb1-603"><a href="#cb1-603"></a><span class="co">#   # d2 &lt;- estimateGLMCommonDisp(y_subbed,design)</span></span>
<span id="cb1-604"><a href="#cb1-604"></a><span class="co">#   # d2 &lt;- estimateGLMTrendedDisp(d2,design, method="auto")</span></span>
<span id="cb1-605"><a href="#cb1-605"></a><span class="co">#   # # You can change method to "auto", "bin.spline", "power", "spline", "bin.loess".</span></span>
<span id="cb1-606"><a href="#cb1-606"></a><span class="co">#   # # The default is "auto" which chooses "bin.spline" when &gt; 200 tags and "power" otherwise.</span></span>
<span id="cb1-607"><a href="#cb1-607"></a><span class="co">#   # d2 &lt;- estimateGLMTagwiseDisp(d2,design)</span></span>
<span id="cb1-608"><a href="#cb1-608"></a><span class="co">#   # plotBCV(d2)</span></span>
<span id="cb1-609"><a href="#cb1-609"></a><span class="co">#   </span></span>
<span id="cb1-610"><a href="#cb1-610"></a><span class="co">#   # Used if intercept removed -----------------------------------------------</span></span>
<span id="cb1-611"><a href="#cb1-611"></a><span class="co">#   </span></span>
<span id="cb1-612"><a href="#cb1-612"></a><span class="co">#   # fit &lt;- glmQLFit(y_subbed, design, robust=TRUE)</span></span>
<span id="cb1-613"><a href="#cb1-613"></a><span class="co">#   # cont &lt;- limma::makeContrasts(xwspdmsYes-xwspdmsNo, levels=design)</span></span>
<span id="cb1-614"><a href="#cb1-614"></a><span class="co">#   # res &lt;- glmQLFTest(fit, contrast=cont)</span></span>
<span id="cb1-615"><a href="#cb1-615"></a><span class="co">#   # topTags(res)$table</span></span>
<span id="cb1-616"><a href="#cb1-616"></a><span class="co">#   # is.de &lt;- decideTestsDGE(res)</span></span>
<span id="cb1-617"><a href="#cb1-617"></a><span class="co">#   # baps &lt;- summary(is.de)</span></span>
<span id="cb1-618"><a href="#cb1-618"></a><span class="co">#   </span></span>
<span id="cb1-619"><a href="#cb1-619"></a><span class="co">#   # fit &lt;- glmFit(d2, design, robust=TRUE)</span></span>
<span id="cb1-620"><a href="#cb1-620"></a><span class="co">#   # cont &lt;- limma::makeContrasts(xwspdmsYes-xwspdmsNo, levels=design)</span></span>
<span id="cb1-621"><a href="#cb1-621"></a><span class="co">#   # #res &lt;- glmLRT(fit, contrast=c(1,-1,0,0))</span></span>
<span id="cb1-622"><a href="#cb1-622"></a><span class="co">#   # res &lt;- glmLRT(fit, contrast=cont)</span></span>
<span id="cb1-623"><a href="#cb1-623"></a><span class="co">#   # topTags(res)$table</span></span>
<span id="cb1-624"><a href="#cb1-624"></a><span class="co">#   # is.de &lt;- decideTestsDGE(res)</span></span>
<span id="cb1-625"><a href="#cb1-625"></a><span class="co">#   # baps &lt;- summary(is.de)</span></span>
<span id="cb1-626"><a href="#cb1-626"></a><span class="co">#   </span></span>
<span id="cb1-627"><a href="#cb1-627"></a><span class="co">#   </span></span>
<span id="cb1-628"><a href="#cb1-628"></a><span class="co">#   # Use if intercept in place -----------------------------------------------</span></span>
<span id="cb1-629"><a href="#cb1-629"></a><span class="co">#   </span></span>
<span id="cb1-630"><a href="#cb1-630"></a><span class="co">#   # fit &lt;- glmQLFit(y, design, robust=TRUE)</span></span>
<span id="cb1-631"><a href="#cb1-631"></a><span class="co">#   # </span></span>
<span id="cb1-632"><a href="#cb1-632"></a><span class="co">#   # summary(fit$var.prior)</span></span>
<span id="cb1-633"><a href="#cb1-633"></a><span class="co">#   # summary(fit$df.prior)</span></span>
<span id="cb1-634"><a href="#cb1-634"></a><span class="co">#   # </span></span>
<span id="cb1-635"><a href="#cb1-635"></a><span class="co">#   # plotQLDisp(fit)</span></span>
<span id="cb1-636"><a href="#cb1-636"></a><span class="co">#   # </span></span>
<span id="cb1-637"><a href="#cb1-637"></a><span class="co">#   # #cont &lt;- limma::makeContrasts(xwspdmsYes-xwspdmsNo, levels=design)</span></span>
<span id="cb1-638"><a href="#cb1-638"></a><span class="co">#   # res &lt;- glmQLFTest(fit, coef=2)</span></span>
<span id="cb1-639"><a href="#cb1-639"></a><span class="co">#   # topTags(res)$table</span></span>
<span id="cb1-640"><a href="#cb1-640"></a><span class="co">#   # is.de &lt;- decideTestsDGE(res)</span></span>
<span id="cb1-641"><a href="#cb1-641"></a><span class="co">#   # baps_2 &lt;- summary(is.de)</span></span>
<span id="cb1-642"><a href="#cb1-642"></a><span class="co">#   </span></span>
<span id="cb1-643"><a href="#cb1-643"></a><span class="co">#   </span></span>
<span id="cb1-644"><a href="#cb1-644"></a><span class="co">#   # DESeq2 ------------------------------------------------------------------</span></span>
<span id="cb1-645"><a href="#cb1-645"></a><span class="co">#   </span></span>
<span id="cb1-646"><a href="#cb1-646"></a><span class="co">#   dds &lt;- DESeqDataSetFromMatrix(y$counts, </span></span>
<span id="cb1-647"><a href="#cb1-647"></a><span class="co">#                                 colData = y$samples, </span></span>
<span id="cb1-648"><a href="#cb1-648"></a><span class="co">#                                 design = ~Sex+age_scaled+xwspdms)</span></span>
<span id="cb1-649"><a href="#cb1-649"></a><span class="co">#   </span></span>
<span id="cb1-650"><a href="#cb1-650"></a><span class="co">#   # Transform counts for data visualization</span></span>
<span id="cb1-651"><a href="#cb1-651"></a><span class="co">#   rld &lt;- vst(dds, blind=TRUE)</span></span>
<span id="cb1-652"><a href="#cb1-652"></a><span class="co">#   </span></span>
<span id="cb1-653"><a href="#cb1-653"></a><span class="co">#   # Plot PCA</span></span>
<span id="cb1-654"><a href="#cb1-654"></a><span class="co">#   DESeq2::plotPCA(rld, intgroup = "xwspdms")</span></span>
<span id="cb1-655"><a href="#cb1-655"></a><span class="co">#   </span></span>
<span id="cb1-656"><a href="#cb1-656"></a><span class="co">#   # Extract the rlog matrix from the object and compute pairwise correlation values</span></span>
<span id="cb1-657"><a href="#cb1-657"></a><span class="co">#   rld_mat &lt;- assay(rld)</span></span>
<span id="cb1-658"><a href="#cb1-658"></a><span class="co">#   rld_cor &lt;- cor(rld_mat)</span></span>
<span id="cb1-659"><a href="#cb1-659"></a><span class="co">#   </span></span>
<span id="cb1-660"><a href="#cb1-660"></a><span class="co">#   # Plot heatmap</span></span>
<span id="cb1-661"><a href="#cb1-661"></a><span class="co">#   pheatmap::pheatmap(rld_cor, annotation = y$samples[, c("xwspdms"), drop=F])</span></span>
<span id="cb1-662"><a href="#cb1-662"></a><span class="co">#   </span></span>
<span id="cb1-663"><a href="#cb1-663"></a><span class="co">#   # Run DESeq2 differential expression analysis</span></span>
<span id="cb1-664"><a href="#cb1-664"></a><span class="co">#   dds &lt;- DESeq(dds)</span></span>
<span id="cb1-665"><a href="#cb1-665"></a><span class="co">#   </span></span>
<span id="cb1-666"><a href="#cb1-666"></a><span class="co">#   # Plot dispersion estimates</span></span>
<span id="cb1-667"><a href="#cb1-667"></a><span class="co">#   plotDispEsts(dds)</span></span>
<span id="cb1-668"><a href="#cb1-668"></a><span class="co">#   </span></span>
<span id="cb1-669"><a href="#cb1-669"></a><span class="co">#   # Check the coefficients for the comparison</span></span>
<span id="cb1-670"><a href="#cb1-670"></a><span class="co">#   resultsNames(dds)</span></span>
<span id="cb1-671"><a href="#cb1-671"></a><span class="co"># }</span></span>
<span id="cb1-672"><a href="#cb1-672"></a><span class="co"># </span></span>
<span id="cb1-673"><a href="#cb1-673"></a><span class="co"># # Generate results object</span></span>
<span id="cb1-674"><a href="#cb1-674"></a><span class="co"># </span></span>
<span id="cb1-675"><a href="#cb1-675"></a><span class="co"># # res &lt;- results(dds,</span></span>
<span id="cb1-676"><a href="#cb1-676"></a><span class="co"># #                contrast = c("xwspdms", "3", "1"),   # For catagorical variables</span></span>
<span id="cb1-677"><a href="#cb1-677"></a><span class="co"># #                alpha = 0.05)</span></span>
<span id="cb1-678"><a href="#cb1-678"></a><span class="co"># # </span></span>
<span id="cb1-679"><a href="#cb1-679"></a><span class="co"># # res &lt;- lfcShrink(dds,</span></span>
<span id="cb1-680"><a href="#cb1-680"></a><span class="co"># #                  contrast =  c("xwspdms", "3", "1"),</span></span>
<span id="cb1-681"><a href="#cb1-681"></a><span class="co"># #                  res=res,</span></span>
<span id="cb1-682"><a href="#cb1-682"></a><span class="co"># #                  type = "ashr")</span></span>
<span id="cb1-683"><a href="#cb1-683"></a><span class="co"># </span></span>
<span id="cb1-684"><a href="#cb1-684"></a><span class="co"># res &lt;- results(dds,</span></span>
<span id="cb1-685"><a href="#cb1-685"></a><span class="co">#                name = "xwspdms",</span></span>
<span id="cb1-686"><a href="#cb1-686"></a><span class="co">#                alpha = 0.05)</span></span>
<span id="cb1-687"><a href="#cb1-687"></a><span class="co"># </span></span>
<span id="cb1-688"><a href="#cb1-688"></a><span class="co"># # Shrink the log2 fold changes to be more appropriate using the apeglm method - </span></span>
<span id="cb1-689"><a href="#cb1-689"></a><span class="co"># # should cite [paper]() when using this method</span></span>
<span id="cb1-690"><a href="#cb1-690"></a><span class="co"># res &lt;- lfcShrink(dds,</span></span>
<span id="cb1-691"><a href="#cb1-691"></a><span class="co">#                  coef = "xwspdms",</span></span>
<span id="cb1-692"><a href="#cb1-692"></a><span class="co">#                  res=res,</span></span>
<span id="cb1-693"><a href="#cb1-693"></a><span class="co">#                  type = "apeglm")</span></span>
<span id="cb1-694"><a href="#cb1-694"></a><span class="co"># </span></span>
<span id="cb1-695"><a href="#cb1-695"></a><span class="co"># </span></span>
<span id="cb1-696"><a href="#cb1-696"></a><span class="co"># </span></span>
<span id="cb1-697"><a href="#cb1-697"></a><span class="co"># hist(res$pvalue, xlab="p value", main="Histogram of nominal p values")</span></span>
<span id="cb1-698"><a href="#cb1-698"></a><span class="co"># </span></span>
<span id="cb1-699"><a href="#cb1-699"></a><span class="co"># # Turn the results object into a tibble for use with tidyverse functions</span></span>
<span id="cb1-700"><a href="#cb1-700"></a><span class="co"># res_tbl &lt;- </span></span>
<span id="cb1-701"><a href="#cb1-701"></a><span class="co">#   res %&gt;%</span></span>
<span id="cb1-702"><a href="#cb1-702"></a><span class="co">#   data.frame() %&gt;%</span></span>
<span id="cb1-703"><a href="#cb1-703"></a><span class="co">#   rownames_to_column(var="gene") %&gt;%</span></span>
<span id="cb1-704"><a href="#cb1-704"></a><span class="co">#   as_tibble() %&gt;%</span></span>
<span id="cb1-705"><a href="#cb1-705"></a><span class="co">#   arrange(padj, pvalue)</span></span>
<span id="cb1-706"><a href="#cb1-706"></a><span class="co"># </span></span>
<span id="cb1-707"><a href="#cb1-707"></a><span class="co"># # Check results output</span></span>
<span id="cb1-708"><a href="#cb1-708"></a><span class="co"># tart &lt;- res_tbl </span></span>
<span id="cb1-709"><a href="#cb1-709"></a><span class="co"># </span></span>
<span id="cb1-710"><a href="#cb1-710"></a><span class="co"># # Set thresholds</span></span>
<span id="cb1-711"><a href="#cb1-711"></a><span class="co"># padj_cutoff &lt;- 0.05</span></span>
<span id="cb1-712"><a href="#cb1-712"></a><span class="co"># </span></span>
<span id="cb1-713"><a href="#cb1-713"></a><span class="co"># # Subset the significant results</span></span>
<span id="cb1-714"><a href="#cb1-714"></a><span class="co"># sig_res &lt;- dplyr::filter(res_tbl, padj &lt; padj_cutoff) %&gt;%</span></span>
<span id="cb1-715"><a href="#cb1-715"></a><span class="co">#   dplyr::arrange(padj)</span></span>
<span id="cb1-716"><a href="#cb1-716"></a><span class="co"># </span></span>
<span id="cb1-717"><a href="#cb1-717"></a><span class="co"># # Check significant genes output</span></span>
<span id="cb1-718"><a href="#cb1-718"></a><span class="co"># sig_res</span></span>
<span id="cb1-719"><a href="#cb1-719"></a><span class="co"># </span></span>
<span id="cb1-720"><a href="#cb1-720"></a><span class="co"># # Scatterplot</span></span>
<span id="cb1-721"><a href="#cb1-721"></a><span class="co"># ## ggplot of top genes</span></span>
<span id="cb1-722"><a href="#cb1-722"></a><span class="co"># normalized_counts &lt;- counts(dds, </span></span>
<span id="cb1-723"><a href="#cb1-723"></a><span class="co">#                             normalized = TRUE)</span></span>
<span id="cb1-724"><a href="#cb1-724"></a><span class="co"># </span></span>
<span id="cb1-725"><a href="#cb1-725"></a><span class="co"># # Extract normalized counts for only the significant genes</span></span>
<span id="cb1-726"><a href="#cb1-726"></a><span class="co"># sig_norm &lt;- data.frame(normalized_counts) %&gt;%</span></span>
<span id="cb1-727"><a href="#cb1-727"></a><span class="co">#   rownames_to_column(var = "gene") %&gt;%</span></span>
<span id="cb1-728"><a href="#cb1-728"></a><span class="co">#   dplyr::filter(gene %in% sig_res$gene)</span></span>
<span id="cb1-729"><a href="#cb1-729"></a><span class="co"># </span></span>
<span id="cb1-730"><a href="#cb1-730"></a><span class="co"># # # Set a color palette</span></span>
<span id="cb1-731"><a href="#cb1-731"></a><span class="co"># # heat_colors &lt;- RColorBrewer::brewer.pal(6, "YlOrRd")</span></span>
<span id="cb1-732"><a href="#cb1-732"></a><span class="co"># </span></span>
<span id="cb1-733"><a href="#cb1-733"></a><span class="co"># # # Run pheatmap using the metadata data frame for the annotation</span></span>
<span id="cb1-734"><a href="#cb1-734"></a><span class="co"># # pheatmap::pheatmap(sig_norm[ , 2:length(colnames(sig_norm))], </span></span>
<span id="cb1-735"><a href="#cb1-735"></a><span class="co"># #          color = heat_colors, </span></span>
<span id="cb1-736"><a href="#cb1-736"></a><span class="co"># #          cluster_rows = T, </span></span>
<span id="cb1-737"><a href="#cb1-737"></a><span class="co"># #          show_rownames = F,</span></span>
<span id="cb1-738"><a href="#cb1-738"></a><span class="co"># #          annotation = y_subbed$samples[, c("xwspdms", "cluster_id")], </span></span>
<span id="cb1-739"><a href="#cb1-739"></a><span class="co"># #          border_color = NA, </span></span>
<span id="cb1-740"><a href="#cb1-740"></a><span class="co"># #          fontsize = 10, </span></span>
<span id="cb1-741"><a href="#cb1-741"></a><span class="co"># #          scale = "row", </span></span>
<span id="cb1-742"><a href="#cb1-742"></a><span class="co"># #          fontsize_row = 10, </span></span>
<span id="cb1-743"><a href="#cb1-743"></a><span class="co"># #          height = 20)    </span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>