<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Matt work summary - Pseudotime</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title">Pseudotime</h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Matt work summary</a> 
    </div>
      </div>
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">Home</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./whole_muscle_scrnaseq.html" class="sidebar-item-text sidebar-link">Whole muscle single nuclear</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./myob_single_nuc.html" class="sidebar-item-text sidebar-link">Myoblast single nuclear</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./atac_seq.html" class="sidebar-item-text sidebar-link">ATACseq</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./conda_envs.html" class="sidebar-item-text sidebar-link">Conda environments</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./extra_tips.html" class="sidebar-item-text sidebar-link">Extra tips</a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">Single cell</a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./single_cell.html" class="sidebar-item-text sidebar-link">Overview</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pseudotime.html" class="sidebar-item-text sidebar-link active">Pseudotime</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pseudobulk.html" class="sidebar-item-text sidebar-link">Pseudobulk</a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#summary" id="toc-summary" class="nav-link active" data-scroll-target="#summary">Summary</a></li>
  <li><a href="#setup" id="toc-setup" class="nav-link" data-scroll-target="#setup">Setup</a></li>
  <li><a href="#processing-script" id="toc-processing-script" class="nav-link" data-scroll-target="#processing-script">Processing script</a>
  <ul class="collapse">
  <li><a href="#pseudotime-and-fitting-a-gam-to-allow-differential-expression-analysis" id="toc-pseudotime-and-fitting-a-gam-to-allow-differential-expression-analysis" class="nav-link" data-scroll-target="#pseudotime-and-fitting-a-gam-to-allow-differential-expression-analysis">Pseudotime and fitting a GAM (to allow differential expression analysis)</a></li>
  <li><a href="#differential-expression-stacked-bars-heat-maps-and-chi-sq" id="toc-differential-expression-stacked-bars-heat-maps-and-chi-sq" class="nav-link" data-scroll-target="#differential-expression-stacked-bars-heat-maps-and-chi-sq">Differential expression, stacked bars, heat maps and chi-sq</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title d-none d-lg-block">Pseudotime</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="summary" class="level1">
<h1>Summary</h1>
<p>Pseudotime conducted on the single cell dataset.</p>
</section>
<section id="setup" class="level1">
<h1>Setup</h1>
<ul>
<li><p>The analysis was conducted in two sections, one on Iridis to allow the fitting of a general additive model and subsequent differential methylation analysis which is highly computationally intensive. The iridis project folder and below script is <code>/scratch/moh1u21/22-10-28-pseudotime/src/pseudotime_01_counts.R</code> .</p></li>
<li><p>The other half of the analysis is at <code>/Users/fluentin44/Library/CloudStorage/OneDrive-UniversityofSouthampton/23-01-09-pseudotime</code> - this folder also contains all the objects used for the analysis.</p></li>
<li><p>The ‘raw data’ for the analysis is the single cell seurat object.</p></li>
<li><p>I used both the <a href="http://bioconductor.org/books/3.16/OSCA.advanced/trajectory-analysis.html">OCSA book</a> and the <a href="https://statomics.github.io/tradeSeq/articles/tradeSeq.html">tradeseq vignette</a> as guidance.</p></li>
</ul>
</section>
<section id="processing-script" class="level1">
<h1>Processing script</h1>
<section id="pseudotime-and-fitting-a-gam-to-allow-differential-expression-analysis" class="level2">
<h2 class="anchored" data-anchor-id="pseudotime-and-fitting-a-gam-to-allow-differential-expression-analysis">Pseudotime and fitting a GAM (to allow differential expression analysis)</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(ggplot2))</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(dplyr))</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(Seurat))</span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(scater))</span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="co">#suppressMessages(library(monocle3))</span></span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="co"># suppressMessages(library(DropletUtils))</span></span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="co"># suppressMessages(library(stringr))</span></span>
<span id="cb1-8"><a href="#cb1-8"></a><span class="co"># suppressMessages(library(celda))</span></span>
<span id="cb1-9"><a href="#cb1-9"></a><span class="co"># suppressMessages(library(slalom))</span></span>
<span id="cb1-10"><a href="#cb1-10"></a><span class="co"># suppressMessages(library(data.table))</span></span>
<span id="cb1-11"><a href="#cb1-11"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(TSCAN))</span>
<span id="cb1-12"><a href="#cb1-12"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(slingshot))</span>
<span id="cb1-13"><a href="#cb1-13"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(tradeSeq))</span>
<span id="cb1-14"><a href="#cb1-14"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(RColorBrewer))</span>
<span id="cb1-15"><a href="#cb1-15"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(BiocParallel))</span>
<span id="cb1-16"><a href="#cb1-16"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(scry))</span>
<span id="cb1-17"><a href="#cb1-17"></a></span>
<span id="cb1-18"><a href="#cb1-18"></a>BPPARAM <span class="ot">&lt;-</span> BiocParallel<span class="sc">::</span><span class="fu">MulticoreParam</span>(<span class="at">workers=</span><span class="dv">30</span>) </span>
<span id="cb1-19"><a href="#cb1-19"></a>BPPARAM</span>
<span id="cb1-20"><a href="#cb1-20"></a></span>
<span id="cb1-21"><a href="#cb1-21"></a>master <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"data/22-10-24-master_updated.rds"</span>)</span>
<span id="cb1-22"><a href="#cb1-22"></a><span class="fu">DefaultAssay</span>(master) <span class="ot">&lt;-</span> <span class="st">"RNA"</span></span>
<span id="cb1-23"><a href="#cb1-23"></a></span>
<span id="cb1-24"><a href="#cb1-24"></a>counts <span class="ot">&lt;-</span> <span class="fu">GetAssayData</span>(<span class="at">object =</span> master, <span class="at">slot =</span> <span class="st">"counts"</span>, <span class="at">assay=</span><span class="st">"RNA"</span>)</span>
<span id="cb1-25"><a href="#cb1-25"></a></span>
<span id="cb1-26"><a href="#cb1-26"></a><span class="co"># Filter based on expression </span></span>
<span id="cb1-27"><a href="#cb1-27"></a><span class="fu">nrow</span>(counts)</span>
<span id="cb1-28"><a href="#cb1-28"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(counts) <span class="sc">&gt;=</span> <span class="dv">1</span></span>
<span id="cb1-29"><a href="#cb1-29"></a><span class="fu">table</span>(keep)</span>
<span id="cb1-30"><a href="#cb1-30"></a>counts <span class="ot">&lt;-</span> counts[keep,]</span>
<span id="cb1-31"><a href="#cb1-31"></a><span class="fu">nrow</span>(counts)</span>
<span id="cb1-32"><a href="#cb1-32"></a></span>
<span id="cb1-33"><a href="#cb1-33"></a>U <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span><span class="dv">0</span> <span class="sc">+</span></span>
<span id="cb1-34"><a href="#cb1-34"></a>                   Encapulation <span class="sc">+</span></span>
<span id="cb1-35"><a href="#cb1-35"></a>                  <span class="co">#  Culture_days + </span></span>
<span id="cb1-36"><a href="#cb1-36"></a>                   Sex <span class="sc">+</span> </span>
<span id="cb1-37"><a href="#cb1-37"></a>                   CC.Difference <span class="sc">+</span> </span>
<span id="cb1-38"><a href="#cb1-38"></a>                   nFeature_RNA <span class="sc">+</span></span>
<span id="cb1-39"><a href="#cb1-39"></a>                   nCount_RNA <span class="sc">+</span></span>
<span id="cb1-40"><a href="#cb1-40"></a>                   percent.mt, </span>
<span id="cb1-41"><a href="#cb1-41"></a>                   <span class="at">data =</span> <span class="fu">as.data.frame</span>(master[[]]))</span>
<span id="cb1-42"><a href="#cb1-42"></a></span>
<span id="cb1-43"><a href="#cb1-43"></a>counts <span class="ot">&lt;-</span> counts[, <span class="fu">colnames</span>(counts) <span class="sc">%in%</span> <span class="fu">rownames</span>(U)]</span>
<span id="cb1-44"><a href="#cb1-44"></a></span>
<span id="cb1-45"><a href="#cb1-45"></a><span class="fu">all</span>(<span class="fu">colnames</span>(counts) <span class="sc">%in%</span> <span class="fu">rownames</span>(U))</span>
<span id="cb1-46"><a href="#cb1-46"></a></span>
<span id="cb1-47"><a href="#cb1-47"></a>boop <span class="ot">&lt;-</span> <span class="fu">rownames</span>(U)</span>
<span id="cb1-48"><a href="#cb1-48"></a>nerts <span class="ot">&lt;-</span> <span class="fu">subset</span>(master, <span class="at">cells =</span> boop)</span>
<span id="cb1-49"><a href="#cb1-49"></a></span>
<span id="cb1-50"><a href="#cb1-50"></a><span class="co"># use 1500 most variable genes for the analysis</span></span>
<span id="cb1-51"><a href="#cb1-51"></a>var_genes <span class="ot">&lt;-</span> <span class="fu">VariableFeatures</span>(nerts)</span>
<span id="cb1-52"><a href="#cb1-52"></a>var_genes <span class="ot">&lt;-</span> var_genes[var_genes <span class="sc">%in%</span> <span class="fu">rownames</span>(counts)]</span>
<span id="cb1-53"><a href="#cb1-53"></a>var_genes <span class="ot">&lt;-</span> var_genes[<span class="dv">1</span><span class="sc">:</span><span class="dv">1500</span>]</span>
<span id="cb1-54"><a href="#cb1-54"></a></span>
<span id="cb1-55"><a href="#cb1-55"></a><span class="co"># var_gene_counts &lt;- GetAssayData(nerts)[var_genes,]</span></span>
<span id="cb1-56"><a href="#cb1-56"></a><span class="co"># nrow(counts)</span></span>
<span id="cb1-57"><a href="#cb1-57"></a><span class="co"># keep &lt;- rowSums(counts) &gt; 1 # rows of the DESeqDataSet that have no counts, or only a single count across all samples</span></span>
<span id="cb1-58"><a href="#cb1-58"></a><span class="co"># counts &lt;- counts[keep,]</span></span>
<span id="cb1-59"><a href="#cb1-59"></a><span class="co"># nrow(counts)</span></span>
<span id="cb1-60"><a href="#cb1-60"></a></span>
<span id="cb1-61"><a href="#cb1-61"></a>merp <span class="ot">&lt;-</span> <span class="fu">as.SingleCellExperiment</span>(nerts)</span>
<span id="cb1-62"><a href="#cb1-62"></a><span class="fu">colLabels</span>(merp) <span class="ot">&lt;-</span> <span class="fu">colData</span>(merp)<span class="sc">$</span>cluster_id</span>
<span id="cb1-63"><a href="#cb1-63"></a></span>
<span id="cb1-64"><a href="#cb1-64"></a><span class="co"># The below is used to try and determine the 'root' node - i.e. the most undifferentiated. This can be done by looking at transcriptional entropy (disorder), the idea being that the mature cells will have a transcriptiome of low entropy as their transcriptional expression will be 'focused', tailored towards a specific function, whereas immature cells will have a high transcriptional entropy as their function/activity is not as well defined yet. </span></span>
<span id="cb1-65"><a href="#cb1-65"></a>entropy <span class="ot">&lt;-</span> <span class="fu">perCellEntropy</span>(merp)</span>
<span id="cb1-66"><a href="#cb1-66"></a>ent.data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">cluster=</span><span class="fu">colLabels</span>(merp), <span class="at">entropy=</span>entropy)</span>
<span id="cb1-67"><a href="#cb1-67"></a><span class="fu">ggplot</span>(ent.data, <span class="fu">aes</span>(<span class="at">x=</span>cluster, <span class="at">y=</span>entropy)) <span class="sc">+</span> </span>
<span id="cb1-68"><a href="#cb1-68"></a>  <span class="fu">geom_violin</span>() <span class="sc">+</span></span>
<span id="cb1-69"><a href="#cb1-69"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">7</span>, <span class="cn">NA</span>)) <span class="sc">+</span></span>
<span id="cb1-70"><a href="#cb1-70"></a>  <span class="fu">stat_summary</span>(<span class="at">fun=</span>median, <span class="at">geom=</span><span class="st">"point"</span>)</span>
<span id="cb1-71"><a href="#cb1-71"></a></span>
<span id="cb1-72"><a href="#cb1-72"></a><span class="co"># load("./data/SCTransformed_PCadjusted_clustered_data4.RData")</span></span>
<span id="cb1-73"><a href="#cb1-73"></a></span>
<span id="cb1-74"><a href="#cb1-74"></a><span class="co"># data4_sce &lt;- as.SingleCellExperiment(data4)</span></span>
<span id="cb1-75"><a href="#cb1-75"></a></span>
<span id="cb1-76"><a href="#cb1-76"></a><span class="co"># Save the objects as separate matrices for input in slingshot</span></span>
<span id="cb1-77"><a href="#cb1-77"></a><span class="fu">reducedDims</span>(merp) <span class="ot">&lt;-</span> <span class="fu">SimpleList</span>(<span class="at">PCA =</span> nerts<span class="sc">@</span>reductions<span class="sc">$</span>pca<span class="sc">@</span>cell.embeddings, </span>
<span id="cb1-78"><a href="#cb1-78"></a>                                <span class="at">UMAP =</span> nerts<span class="sc">@</span>reductions<span class="sc">$</span>umap<span class="sc">@</span>cell.embeddings)</span>
<span id="cb1-79"><a href="#cb1-79"></a></span>
<span id="cb1-80"><a href="#cb1-80"></a><span class="co"># pseudo.all &lt;- quickPseudotime(merp, use.dimred="UMAP")</span></span>
<span id="cb1-81"><a href="#cb1-81"></a><span class="co"># head(pseudo.all$ordering)</span></span>
<span id="cb1-82"><a href="#cb1-82"></a><span class="co"># mnn.pseudo &lt;- averagePseudotime(pseudo.all$ordering)</span></span>
<span id="cb1-83"><a href="#cb1-83"></a><span class="co"># plotUMAP(merp, colour_by=I(mnn.pseudo), text_by="label", text_colour="red") +</span></span>
<span id="cb1-84"><a href="#cb1-84"></a><span class="co">#   geom_line(data=pseudo.all$connected$UMAP, mapping=aes(x=UMAP_1, y=UMAP_2, group=edge))</span></span>
<span id="cb1-85"><a href="#cb1-85"></a></span>
<span id="cb1-86"><a href="#cb1-86"></a><span class="co"># data4_sce &lt;- as.SingleCellExperiment(data4)</span></span>
<span id="cb1-87"><a href="#cb1-87"></a><span class="co"># reducedDims(data4_sce) &lt;- SimpleList(PCA = data4@reductions$pca@cell.embeddings, UMAP = data4@reductions$umap@cell.embeddings)</span></span>
<span id="cb1-88"><a href="#cb1-88"></a>pto <span class="ot">&lt;-</span> <span class="fu">slingshot</span>(<span class="at">data=</span>merp, </span>
<span id="cb1-89"><a href="#cb1-89"></a>                 <span class="at">clusterLabels =</span> <span class="st">"cluster_id"</span>, </span>
<span id="cb1-90"><a href="#cb1-90"></a>                 <span class="at">reducedDim=</span><span class="st">"UMAP"</span>, </span>
<span id="cb1-91"><a href="#cb1-91"></a>                 <span class="at">start.clus=</span><span class="dv">6</span><span class="co">#,</span></span>
<span id="cb1-92"><a href="#cb1-92"></a>                 <span class="co"># omega=T</span></span>
<span id="cb1-93"><a href="#cb1-93"></a>                 )</span>
<span id="cb1-94"><a href="#cb1-94"></a>sds <span class="ot">&lt;-</span> <span class="fu">as.SlingshotDataSet</span>(pto)</span>
<span id="cb1-95"><a href="#cb1-95"></a><span class="co"># Cairo::CairoX11()</span></span>
<span id="cb1-96"><a href="#cb1-96"></a><span class="co"># plot.new()</span></span>
<span id="cb1-97"><a href="#cb1-97"></a>colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"firebrick2"</span>, <span class="st">"deepskyblue"</span>, <span class="st">"coral"</span>, <span class="st">"springgreen4"</span>, <span class="st">"yellow1"</span>, <span class="st">"limegreen"</span>, <span class="st">"blue1"</span>, <span class="st">"turquoise1"</span>, <span class="st">"darkorange"</span>, <span class="st">"darkmagenta"</span>, <span class="st">"deeppink1"</span>, <span class="st">"brown4"</span>, <span class="st">"green3"</span>)</span>
<span id="cb1-98"><a href="#cb1-98"></a><span class="fu">plot</span>(<span class="fu">reducedDims</span>(pto)<span class="sc">$</span>UMAP, <span class="at">col =</span> colors[pto<span class="sc">$</span>cluster_id], <span class="at">pch=</span><span class="dv">16</span>, <span class="at">asp =</span> <span class="dv">1</span>, <span class="at">cex=</span><span class="fl">0.5</span>)</span>
<span id="cb1-99"><a href="#cb1-99"></a><span class="fu">lines</span>(<span class="fu">SlingshotDataSet</span>(pto), <span class="at">lwd=</span><span class="dv">2</span>, <span class="at">type =</span> <span class="st">'lineages'</span>, <span class="at">col =</span> <span class="st">'black'</span>)</span>
<span id="cb1-100"><a href="#cb1-100"></a></span>
<span id="cb1-101"><a href="#cb1-101"></a><span class="co"># plot(reducedDims(pto)$UMAP, col = colors[pto$cluster_id], pch=16, asp = 1, cex=0.5)</span></span>
<span id="cb1-102"><a href="#cb1-102"></a><span class="co"># lines(sds, lwd = 3, col = "black")</span></span>
<span id="cb1-103"><a href="#cb1-103"></a></span>
<span id="cb1-104"><a href="#cb1-104"></a><span class="co"># # Reduced for testing</span></span>
<span id="cb1-105"><a href="#cb1-105"></a><span class="co"># filt_counts &lt;- counts[rowSums(counts &gt; 5) &gt; ncol(counts)/100, ]</span></span>
<span id="cb1-106"><a href="#cb1-106"></a><span class="co"># dim(filt_counts)</span></span>
<span id="cb1-107"><a href="#cb1-107"></a></span>
<span id="cb1-108"><a href="#cb1-108"></a>pseudo.paths <span class="ot">&lt;-</span> <span class="fu">slingPseudotime</span>(pto)</span>
<span id="cb1-109"><a href="#cb1-109"></a><span class="fu">head</span>(pseudo.paths)</span>
<span id="cb1-110"><a href="#cb1-110"></a>shared.pseudo <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(pseudo.paths, <span class="at">na.rm=</span><span class="cn">TRUE</span>)</span>
<span id="cb1-111"><a href="#cb1-111"></a></span>
<span id="cb1-112"><a href="#cb1-112"></a><span class="co"># Need to loop over the paths and add each one separately.</span></span>
<span id="cb1-113"><a href="#cb1-113"></a>gg <span class="ot">&lt;-</span> scater<span class="sc">::</span><span class="fu">plotUMAP</span>(pto, <span class="at">colour_by=</span><span class="fu">I</span>(shared.pseudo))</span>
<span id="cb1-114"><a href="#cb1-114"></a>embedded <span class="ot">&lt;-</span> <span class="fu">embedCurves</span>(pto, <span class="st">"UMAP"</span>)</span>
<span id="cb1-115"><a href="#cb1-115"></a>embedded <span class="ot">&lt;-</span> <span class="fu">slingCurves</span>(embedded)</span>
<span id="cb1-116"><a href="#cb1-116"></a><span class="cf">for</span> (path <span class="cf">in</span> embedded) {</span>
<span id="cb1-117"><a href="#cb1-117"></a>  embedded <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(path<span class="sc">$</span>s[path<span class="sc">$</span>ord,])</span>
<span id="cb1-118"><a href="#cb1-118"></a>gg <span class="ot">&lt;-</span> gg <span class="sc">+</span> <span class="fu">geom_path</span>(<span class="at">data=</span>embedded, <span class="fu">aes</span>(<span class="at">x=</span>UMAP_1, <span class="at">y=</span>UMAP_2), <span class="at">size=</span><span class="fl">1.2</span>)</span>
<span id="cb1-119"><a href="#cb1-119"></a>}</span>
<span id="cb1-120"><a href="#cb1-120"></a></span>
<span id="cb1-121"><a href="#cb1-121"></a>gg</span>
<span id="cb1-122"><a href="#cb1-122"></a></span>
<span id="cb1-123"><a href="#cb1-123"></a><span class="fu">head</span>(embedded)</span>
<span id="cb1-124"><a href="#cb1-124"></a></span>
<span id="cb1-125"><a href="#cb1-125"></a><span class="co"># set.seed(91823)</span></span>
<span id="cb1-126"><a href="#cb1-126"></a><span class="co"># icMat &lt;- evaluateK(counts = counts, </span></span>
<span id="cb1-127"><a href="#cb1-127"></a><span class="co">#                    sds = sds, </span></span>
<span id="cb1-128"><a href="#cb1-128"></a><span class="co">#                    k = 3:10, </span></span>
<span id="cb1-129"><a href="#cb1-129"></a><span class="co">#                    nGenes = 200, </span></span>
<span id="cb1-130"><a href="#cb1-130"></a><span class="co">#                    verbose = T, </span></span>
<span id="cb1-131"><a href="#cb1-131"></a><span class="co">#                    plot = TRUE)</span></span>
<span id="cb1-132"><a href="#cb1-132"></a><span class="co"># tradeSeq::plot_evalutateK_results(icMat)</span></span>
<span id="cb1-133"><a href="#cb1-133"></a><span class="co"># print(icMat[1:2, ])</span></span>
<span id="cb1-134"><a href="#cb1-134"></a></span>
<span id="cb1-135"><a href="#cb1-135"></a><span class="co"># saveRDS(icMat, "results/knots.rds")</span></span>
<span id="cb1-136"><a href="#cb1-136"></a></span>
<span id="cb1-137"><a href="#cb1-137"></a></span>
<span id="cb1-138"><a href="#cb1-138"></a><span class="co"># BPPARAM &lt;- BiocParallel::bpparam()</span></span>
<span id="cb1-139"><a href="#cb1-139"></a><span class="co"># BPPARAM # lists current options</span></span>
<span id="cb1-140"><a href="#cb1-140"></a><span class="co"># BPPARAM$workers &lt;- 30 </span></span>
<span id="cb1-141"><a href="#cb1-141"></a><span class="co"># BiocParallel::register(BiocParallel::SerialParam())</span></span>
<span id="cb1-142"><a href="#cb1-142"></a></span>
<span id="cb1-143"><a href="#cb1-143"></a><span class="co"># # fit negative binomial GAM</span></span>
<span id="cb1-144"><a href="#cb1-144"></a><span class="co"># U &lt;- model.matrix(~Culture +</span></span>
<span id="cb1-145"><a href="#cb1-145"></a><span class="co">#                    Culture_days + </span></span>
<span id="cb1-146"><a href="#cb1-146"></a><span class="co">#                    Sex + </span></span>
<span id="cb1-147"><a href="#cb1-147"></a><span class="co">#                    CC.Difference + </span></span>
<span id="cb1-148"><a href="#cb1-148"></a><span class="co">#                    nFeature_RNA +</span></span>
<span id="cb1-149"><a href="#cb1-149"></a><span class="co">#                    nCount_RNA +</span></span>
<span id="cb1-150"><a href="#cb1-150"></a><span class="co">#                    percent.mt, </span></span>
<span id="cb1-151"><a href="#cb1-151"></a><span class="co">#                    data = as.data.frame(colData(merp)))</span></span>
<span id="cb1-152"><a href="#cb1-152"></a></span>
<span id="cb1-153"><a href="#cb1-153"></a></span>
<span id="cb1-154"><a href="#cb1-154"></a></span>
<span id="cb1-155"><a href="#cb1-155"></a><span class="co"># var_genes &lt;- VariableFeatures(master)</span></span>
<span id="cb1-156"><a href="#cb1-156"></a><span class="co"># variable_genes &lt;- GetAssayData(master)[var_genes,]</span></span>
<span id="cb1-157"><a href="#cb1-157"></a></span>
<span id="cb1-158"><a href="#cb1-158"></a><span class="co"># sce &lt;- scry::devianceFeatureSelection(counts)</span></span>
<span id="cb1-159"><a href="#cb1-159"></a></span>
<span id="cb1-160"><a href="#cb1-160"></a><span class="co"># # Filter for highly variable</span></span>
<span id="cb1-161"><a href="#cb1-161"></a><span class="fu">all</span>(<span class="fu">colnames</span>(counts) <span class="sc">%in%</span> <span class="fu">rownames</span>(U))</span>
<span id="cb1-162"><a href="#cb1-162"></a><span class="co"># blips &lt;- counts[, colnames(counts) %in% rownames(U)]</span></span>
<span id="cb1-163"><a href="#cb1-163"></a><span class="co"># all(colnames(blips) %in% rownames(U))</span></span>
<span id="cb1-164"><a href="#cb1-164"></a></span>
<span id="cb1-165"><a href="#cb1-165"></a><span class="co"># length(var_genes)</span></span>
<span id="cb1-166"><a href="#cb1-166"></a><span class="co"># var_genes &lt;- var_genes[var_genes %in% rownames(blips)]</span></span>
<span id="cb1-167"><a href="#cb1-167"></a><span class="co"># length(var_genes)</span></span>
<span id="cb1-168"><a href="#cb1-168"></a></span>
<span id="cb1-169"><a href="#cb1-169"></a><span class="co"># final_counts &lt;- as.matrix(blips)</span></span>
<span id="cb1-170"><a href="#cb1-170"></a></span>
<span id="cb1-171"><a href="#cb1-171"></a><span class="co"># system.time(</span></span>
<span id="cb1-172"><a href="#cb1-172"></a><span class="co"># nerfs &lt;- fitGAM(</span></span>
<span id="cb1-173"><a href="#cb1-173"></a><span class="co">#                 counts = final_counts, </span></span>
<span id="cb1-174"><a href="#cb1-174"></a><span class="co">#                 genes = var_genes, </span></span>
<span id="cb1-175"><a href="#cb1-175"></a><span class="co">#                 # genes = 1:2, </span></span>
<span id="cb1-176"><a href="#cb1-176"></a><span class="co">#                 sds = sds, </span></span>
<span id="cb1-177"><a href="#cb1-177"></a><span class="co">#                 U = U,</span></span>
<span id="cb1-178"><a href="#cb1-178"></a><span class="co">#                 verbose = T, </span></span>
<span id="cb1-179"><a href="#cb1-179"></a><span class="co">#                 parallel = T, </span></span>
<span id="cb1-180"><a href="#cb1-180"></a><span class="co">#                 BPPARAM = BPPARAM,</span></span>
<span id="cb1-181"><a href="#cb1-181"></a><span class="co">#                 nknots = 6,</span></span>
<span id="cb1-182"><a href="#cb1-182"></a><span class="co">#                 sce = T)</span></span>
<span id="cb1-183"><a href="#cb1-183"></a><span class="co"># )</span></span>
<span id="cb1-184"><a href="#cb1-184"></a></span>
<span id="cb1-185"><a href="#cb1-185"></a><span class="co"># Fitting a model to allow for differential expression analyses</span></span>
<span id="cb1-186"><a href="#cb1-186"></a></span>
<span id="cb1-187"><a href="#cb1-187"></a>nerfs <span class="ot">&lt;-</span> <span class="fu">fitGAM</span>(</span>
<span id="cb1-188"><a href="#cb1-188"></a>                <span class="at">counts =</span> counts, </span>
<span id="cb1-189"><a href="#cb1-189"></a>                <span class="at">genes =</span> var_genes, </span>
<span id="cb1-190"><a href="#cb1-190"></a>                <span class="co"># genes = 1:2, </span></span>
<span id="cb1-191"><a href="#cb1-191"></a>                <span class="at">sds =</span> sds, </span>
<span id="cb1-192"><a href="#cb1-192"></a>                <span class="at">U =</span> U,</span>
<span id="cb1-193"><a href="#cb1-193"></a>                <span class="at">verbose =</span> T, </span>
<span id="cb1-194"><a href="#cb1-194"></a>                <span class="at">parallel =</span> T, </span>
<span id="cb1-195"><a href="#cb1-195"></a>                <span class="at">BPPARAM =</span> BPPARAM,</span>
<span id="cb1-196"><a href="#cb1-196"></a>                <span class="at">nknots =</span> <span class="dv">6</span>,</span>
<span id="cb1-197"><a href="#cb1-197"></a>                <span class="at">sce =</span> T)</span>
<span id="cb1-198"><a href="#cb1-198"></a></span>
<span id="cb1-199"><a href="#cb1-199"></a><span class="co"># saveRDS(nerfs, "results/gam_model_20_counts.rds")</span></span>
<span id="cb1-200"><a href="#cb1-200"></a>gam <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"results/gam_model_20_counts.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="differential-expression-stacked-bars-heat-maps-and-chi-sq" class="level2">
<h2 class="anchored" data-anchor-id="differential-expression-stacked-bars-heat-maps-and-chi-sq">Differential expression, stacked bars, heat maps and chi-sq</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="co"># Packages ----------------------------------------------------------------</span></span>
<span id="cb2-2"><a href="#cb2-2"></a></span>
<span id="cb2-3"><a href="#cb2-3"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(slingshot))</span>
<span id="cb2-4"><a href="#cb2-4"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(tradeSeq))</span>
<span id="cb2-5"><a href="#cb2-5"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(tidyverse))</span>
<span id="cb2-6"><a href="#cb2-6"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(org.Hs.eg.db))</span>
<span id="cb2-7"><a href="#cb2-7"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(clusterProfiler))</span>
<span id="cb2-8"><a href="#cb2-8"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(openxlsx))</span>
<span id="cb2-9"><a href="#cb2-9"></a></span>
<span id="cb2-10"><a href="#cb2-10"></a><span class="co"># Functions ---------------------------------------------------------------</span></span>
<span id="cb2-11"><a href="#cb2-11"></a></span>
<span id="cb2-12"><a href="#cb2-12"></a>format_results <span class="ot">&lt;-</span> <span class="cf">function</span>(dframe){</span>
<span id="cb2-13"><a href="#cb2-13"></a>  dframe<span class="sc">$</span>refined_p <span class="ot">&lt;-</span> stats<span class="sc">::</span><span class="fu">pchisq</span>(dframe<span class="sc">$</span>waldStat, </span>
<span id="cb2-14"><a href="#cb2-14"></a>                                    <span class="at">df =</span> dframe<span class="sc">$</span>df, </span>
<span id="cb2-15"><a href="#cb2-15"></a>                                    <span class="at">lower.tail =</span> <span class="cn">FALSE</span>, </span>
<span id="cb2-16"><a href="#cb2-16"></a>                                    <span class="at">log.p =</span> F)</span>
<span id="cb2-17"><a href="#cb2-17"></a>  <span class="co"># dframe$refined_p &lt;- 1 - stats::pchisq(dframe$waldStat, df = dframe$df)</span></span>
<span id="cb2-18"><a href="#cb2-18"></a>  <span class="co"># dframe$pvalue &lt;- NULL</span></span>
<span id="cb2-19"><a href="#cb2-19"></a>  dframe<span class="sc">$</span>fdr <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(dframe<span class="sc">$</span>refined_p, <span class="at">method=</span><span class="st">"BH"</span>, <span class="at">n=</span><span class="fu">length</span>(dframe<span class="sc">$</span>refined_p))</span>
<span id="cb2-20"><a href="#cb2-20"></a>  dframe <span class="ot">&lt;-</span> dframe[<span class="fu">order</span>(dframe<span class="sc">$</span>refined_p), ]</span>
<span id="cb2-21"><a href="#cb2-21"></a>  dframe<span class="sc">$</span>feature_id <span class="ot">&lt;-</span> <span class="fu">rownames</span>(dframe)</span>
<span id="cb2-22"><a href="#cb2-22"></a>  dframe <span class="ot">&lt;-</span> </span>
<span id="cb2-23"><a href="#cb2-23"></a>    dframe <span class="sc">%&gt;%</span> </span>
<span id="cb2-24"><a href="#cb2-24"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(fdr<span class="sc">&lt;</span><span class="fl">0.05</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb2-25"><a href="#cb2-25"></a>    <span class="co"># dplyr::filter(logFC&gt;1.5) %&gt;%. -will need an ifelse if want to include</span></span>
<span id="cb2-26"><a href="#cb2-26"></a>    <span class="fu">arrange</span>(<span class="fu">desc</span>(waldStat))</span>
<span id="cb2-27"><a href="#cb2-27"></a>  dframe<span class="sc">$</span>pvalue <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb2-28"><a href="#cb2-28"></a>  <span class="fu">return</span>(dframe)</span>
<span id="cb2-29"><a href="#cb2-29"></a>}</span>
<span id="cb2-30"><a href="#cb2-30"></a>enrichment_startvsend <span class="ot">&lt;-</span> <span class="cf">function</span>(df) {</span>
<span id="cb2-31"><a href="#cb2-31"></a>    <span class="fu">map</span>(df, <span class="sc">~</span></span>
<span id="cb2-32"><a href="#cb2-32"></a>          .x <span class="sc">%&gt;%</span></span>
<span id="cb2-33"><a href="#cb2-33"></a>          dplyr<span class="sc">::</span><span class="fu">filter</span>(fdr <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-34"><a href="#cb2-34"></a>          <span class="fu">pull</span>(feature_id)) <span class="sc">%&gt;%</span></span>
<span id="cb2-35"><a href="#cb2-35"></a>    <span class="fu">map</span>(</span>
<span id="cb2-36"><a href="#cb2-36"></a>      <span class="sc">~</span> <span class="fu">enrichGO</span>(</span>
<span id="cb2-37"><a href="#cb2-37"></a>        <span class="at">gene =</span> .x,</span>
<span id="cb2-38"><a href="#cb2-38"></a>        <span class="at">OrgDb =</span> org.Hs.eg.db,</span>
<span id="cb2-39"><a href="#cb2-39"></a>        <span class="at">keyType =</span> <span class="st">"SYMBOL"</span>,</span>
<span id="cb2-40"><a href="#cb2-40"></a>        <span class="at">ont =</span> <span class="st">"ALL"</span>,</span>
<span id="cb2-41"><a href="#cb2-41"></a>        <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb2-42"><a href="#cb2-42"></a>        <span class="at">readable =</span> <span class="cn">TRUE</span></span>
<span id="cb2-43"><a href="#cb2-43"></a>      )</span>
<span id="cb2-44"><a href="#cb2-44"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb2-45"><a href="#cb2-45"></a>    <span class="fu">map</span>(as_tibble) <span class="sc">%&gt;%</span></span>
<span id="cb2-46"><a href="#cb2-46"></a>    <span class="fu">map</span>(<span class="sc">~</span> <span class="fu">filter</span>(.x, Count <span class="sc">&gt;=</span> <span class="dv">3</span>))</span>
<span id="cb2-47"><a href="#cb2-47"></a>}</span>
<span id="cb2-48"><a href="#cb2-48"></a>arts <span class="ot">&lt;-</span> <span class="cf">function</span>(number_used, results_tab){</span>
<span id="cb2-49"><a href="#cb2-49"></a>  list_whole <span class="ot">&lt;-</span> results_tab[,<span class="fu">grep</span>(number_used, <span class="fu">colnames</span>(results_tab))]</span>
<span id="cb2-50"><a href="#cb2-50"></a>  <span class="fu">return</span>(list_whole)</span>
<span id="cb2-51"><a href="#cb2-51"></a>}</span>
<span id="cb2-52"><a href="#cb2-52"></a>sort_list_by_colname <span class="ot">&lt;-</span> <span class="cf">function</span>(df, matching_pattern, dec){</span>
<span id="cb2-53"><a href="#cb2-53"></a>  darble <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df)[<span class="fu">grep</span>(matching_pattern, <span class="fu">colnames</span>(df), <span class="at">ignore.case=</span>T)]</span>
<span id="cb2-54"><a href="#cb2-54"></a>  <span class="cf">if</span>(dec){</span>
<span id="cb2-55"><a href="#cb2-55"></a>    df <span class="ot">&lt;-</span> <span class="fu">arrange</span>(df, <span class="fu">desc</span>(<span class="sc">!!</span><span class="fu">sym</span>(darble)))</span>
<span id="cb2-56"><a href="#cb2-56"></a>  } <span class="cf">else</span>{</span>
<span id="cb2-57"><a href="#cb2-57"></a>    df <span class="ot">&lt;-</span> <span class="fu">arrange</span>(df, <span class="sc">!!</span><span class="fu">sym</span>(darble))</span>
<span id="cb2-58"><a href="#cb2-58"></a>  }</span>
<span id="cb2-59"><a href="#cb2-59"></a>  <span class="fu">return</span>(df)</span>
<span id="cb2-60"><a href="#cb2-60"></a>}</span>
<span id="cb2-61"><a href="#cb2-61"></a>filter_list_by_colname <span class="ot">&lt;-</span> <span class="cf">function</span>(df, matching_pattern, sig_level){</span>
<span id="cb2-62"><a href="#cb2-62"></a>  darble <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df)[<span class="fu">grep</span>(matching_pattern, <span class="fu">colnames</span>(df), <span class="at">ignore.case=</span>T)]</span>
<span id="cb2-63"><a href="#cb2-63"></a>  df <span class="ot">&lt;-</span> <span class="fu">filter</span>(df, <span class="sc">!!</span><span class="fu">sym</span>(darble)<span class="sc">&lt;</span>sig_level)</span>
<span id="cb2-64"><a href="#cb2-64"></a>  <span class="fu">return</span>(df)</span>
<span id="cb2-65"><a href="#cb2-65"></a>}</span>
<span id="cb2-66"><a href="#cb2-66"></a></span>
<span id="cb2-67"><a href="#cb2-67"></a><span class="co"># Import ------------------------------------------------------------------</span></span>
<span id="cb2-68"><a href="#cb2-68"></a></span>
<span id="cb2-69"><a href="#cb2-69"></a>pto <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"data/pto.rds"</span>)</span>
<span id="cb2-70"><a href="#cb2-70"></a>sds <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"data/sds.rds"</span>)</span>
<span id="cb2-71"><a href="#cb2-71"></a>res <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"data/assores_ind_lineages.rds"</span>)</span>
<span id="cb2-72"><a href="#cb2-72"></a>gam <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"data/gam_model_20_counts.rds"</span>)</span>
<span id="cb2-73"><a href="#cb2-73"></a>var_genes <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"data/var_genes.rds"</span>)</span>
<span id="cb2-74"><a href="#cb2-74"></a>seur_obj <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../21-11-18-single_cell_seq/data/processed/22-11-29-master_updated.rds"</span>)</span>
<span id="cb2-75"><a href="#cb2-75"></a></span>
<span id="cb2-76"><a href="#cb2-76"></a><span class="co"># Association test (genes that change with pseudotime) --------------------</span></span>
<span id="cb2-77"><a href="#cb2-77"></a></span>
<span id="cb2-78"><a href="#cb2-78"></a><span class="co"># results_list &lt;- list()</span></span>
<span id="cb2-79"><a href="#cb2-79"></a><span class="co"># results_list[["global"]] &lt;- select(res, waldStat, df, pvalue, meanLogFC)</span></span>
<span id="cb2-80"><a href="#cb2-80"></a><span class="co"># results_list[["lin_1"]] &lt;- select(res, contains("_1"), meanLogFC)</span></span>
<span id="cb2-81"><a href="#cb2-81"></a><span class="co"># results_list[["lin_2"]] &lt;- select(res, contains("_2"), meanLogFC)</span></span>
<span id="cb2-82"><a href="#cb2-82"></a><span class="co"># results_list[["lin_3"]] &lt;- select(res, contains("_3"), meanLogFC)</span></span>
<span id="cb2-83"><a href="#cb2-83"></a><span class="co"># results_list[["lin_4"]] &lt;- select(res, contains("_4"), meanLogFC)</span></span>
<span id="cb2-84"><a href="#cb2-84"></a><span class="co"># results_list[["lin_5"]] &lt;- select(res, contains("_5"), meanLogFC)</span></span>
<span id="cb2-85"><a href="#cb2-85"></a></span>
<span id="cb2-86"><a href="#cb2-86"></a><span class="co"># Create two vectors: one for list names and another for search patterns</span></span>
<span id="cb2-87"><a href="#cb2-87"></a>results_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb2-88"><a href="#cb2-88"></a>list_names <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"global"</span>, <span class="st">"lin_1"</span>, <span class="st">"lin_2"</span>, <span class="st">"lin_3"</span>, <span class="st">"lin_4"</span>, <span class="st">"lin_5"</span>)</span>
<span id="cb2-89"><a href="#cb2-89"></a>search_patterns <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">""</span>, <span class="st">"_1"</span>, <span class="st">"_2"</span>, <span class="st">"_3"</span>, <span class="st">"_4"</span>, <span class="st">"_5"</span>)</span>
<span id="cb2-90"><a href="#cb2-90"></a></span>
<span id="cb2-91"><a href="#cb2-91"></a><span class="co"># Use map2 to create the results_list</span></span>
<span id="cb2-92"><a href="#cb2-92"></a>results_list <span class="ot">&lt;-</span> <span class="fu">map2</span>(list_names, search_patterns, <span class="cf">function</span>(list_name, search_pattern) {</span>
<span id="cb2-93"><a href="#cb2-93"></a>  <span class="cf">if</span> (list_name <span class="sc">==</span> <span class="st">"global"</span>) {</span>
<span id="cb2-94"><a href="#cb2-94"></a>    selected_columns <span class="ot">&lt;-</span> <span class="fu">select</span>(res, waldStat, df, pvalue, meanLogFC)</span>
<span id="cb2-95"><a href="#cb2-95"></a>  } <span class="cf">else</span> {</span>
<span id="cb2-96"><a href="#cb2-96"></a>    selected_columns <span class="ot">&lt;-</span> <span class="fu">select</span>(res, <span class="fu">contains</span>(search_pattern), meanLogFC)</span>
<span id="cb2-97"><a href="#cb2-97"></a>  }</span>
<span id="cb2-98"><a href="#cb2-98"></a>  selected_columns</span>
<span id="cb2-99"><a href="#cb2-99"></a>})</span>
<span id="cb2-100"><a href="#cb2-100"></a></span>
<span id="cb2-101"><a href="#cb2-101"></a><span class="co"># Assign the list names to results_list</span></span>
<span id="cb2-102"><a href="#cb2-102"></a><span class="fu">names</span>(results_list) <span class="ot">&lt;-</span> list_names</span>
<span id="cb2-103"><a href="#cb2-103"></a></span>
<span id="cb2-104"><a href="#cb2-104"></a></span>
<span id="cb2-105"><a href="#cb2-105"></a><span class="co"># Take number out of colnames so all can be bound</span></span>
<span id="cb2-106"><a href="#cb2-106"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">names</span>(results_list)){</span>
<span id="cb2-107"><a href="#cb2-107"></a>  <span class="fu">colnames</span>(results_list[[i]]) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"_(</span><span class="sc">\\</span><span class="st">d{1})"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(results_list[[i]]))</span>
<span id="cb2-108"><a href="#cb2-108"></a>}</span>
<span id="cb2-109"><a href="#cb2-109"></a></span>
<span id="cb2-110"><a href="#cb2-110"></a>final_res <span class="ot">&lt;-</span> <span class="fu">map</span>(results_list, format_results)</span>
<span id="cb2-111"><a href="#cb2-111"></a></span>
<span id="cb2-112"><a href="#cb2-112"></a>wb <span class="ot">&lt;-</span> <span class="fu">createWorkbook</span>()</span>
<span id="cb2-113"><a href="#cb2-113"></a></span>
<span id="cb2-114"><a href="#cb2-114"></a>dirf <span class="ot">&lt;-</span> </span>
<span id="cb2-115"><a href="#cb2-115"></a>  final_res <span class="sc">%&gt;%</span> </span>
<span id="cb2-116"><a href="#cb2-116"></a>  <span class="fu">map</span>(<span class="sc">~</span><span class="fu">arrange</span>(.x, fdr)) <span class="sc">%&gt;%</span> </span>
<span id="cb2-117"><a href="#cb2-117"></a>  <span class="fu">map</span>(<span class="sc">~</span><span class="fu">rownames_to_column</span>(.x, <span class="at">var=</span><span class="st">"gene"</span>))</span>
<span id="cb2-118"><a href="#cb2-118"></a>tot <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(dirf, <span class="at">.id =</span> <span class="st">"lineage"</span>)</span>
<span id="cb2-119"><a href="#cb2-119"></a><span class="co"># write_csv(tot, "23-04-12-pseudo_DGE.csv")</span></span>
<span id="cb2-120"><a href="#cb2-120"></a></span>
<span id="cb2-121"><a href="#cb2-121"></a><span class="fu">addWorksheet</span>(wb, <span class="st">"assn_test"</span>)</span>
<span id="cb2-122"><a href="#cb2-122"></a><span class="fu">writeData</span>(wb, <span class="st">"assn_test"</span>, tot)</span>
<span id="cb2-123"><a href="#cb2-123"></a></span>
<span id="cb2-124"><a href="#cb2-124"></a><span class="co"># Start vs. end (genes that change between two points) --------------------</span></span>
<span id="cb2-125"><a href="#cb2-125"></a></span>
<span id="cb2-126"><a href="#cb2-126"></a><span class="co"># Also called discovering progenitor genes</span></span>
<span id="cb2-127"><a href="#cb2-127"></a></span>
<span id="cb2-128"><a href="#cb2-128"></a>startRes <span class="ot">&lt;-</span> <span class="fu">startVsEndTest</span>(gam, </span>
<span id="cb2-129"><a href="#cb2-129"></a>                           <span class="at">global=</span>T, </span>
<span id="cb2-130"><a href="#cb2-130"></a>                           <span class="at">lineages=</span>T <span class="co">#,</span></span>
<span id="cb2-131"><a href="#cb2-131"></a>                           <span class="co"># l2fc=1</span></span>
<span id="cb2-132"><a href="#cb2-132"></a>                           )</span>
<span id="cb2-133"><a href="#cb2-133"></a>start_res <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb2-134"><a href="#cb2-134"></a>start_res[[<span class="st">"global"</span>]] <span class="ot">&lt;-</span> <span class="fu">select</span>(startRes, waldStat, df, pvalue)</span>
<span id="cb2-135"><a href="#cb2-135"></a>start_res[[<span class="st">"lin_1"</span>]] <span class="ot">&lt;-</span> <span class="fu">select</span>(startRes, <span class="fu">contains</span>(<span class="st">"1"</span>))</span>
<span id="cb2-136"><a href="#cb2-136"></a>start_res[[<span class="st">"lin_2"</span>]] <span class="ot">&lt;-</span> <span class="fu">select</span>(startRes, <span class="fu">contains</span>(<span class="st">"2"</span>))</span>
<span id="cb2-137"><a href="#cb2-137"></a>start_res[[<span class="st">"lin_3"</span>]] <span class="ot">&lt;-</span> <span class="fu">select</span>(startRes, <span class="fu">contains</span>(<span class="st">"3"</span>))</span>
<span id="cb2-138"><a href="#cb2-138"></a>start_res[[<span class="st">"lin_4"</span>]] <span class="ot">&lt;-</span> <span class="fu">select</span>(startRes, <span class="fu">contains</span>(<span class="st">"4"</span>))</span>
<span id="cb2-139"><a href="#cb2-139"></a>start_res[[<span class="st">"lin_5"</span>]] <span class="ot">&lt;-</span> <span class="fu">select</span>(startRes, <span class="fu">contains</span>(<span class="st">"5"</span>))</span>
<span id="cb2-140"><a href="#cb2-140"></a></span>
<span id="cb2-141"><a href="#cb2-141"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">names</span>(start_res)){</span>
<span id="cb2-142"><a href="#cb2-142"></a>  <span class="fu">colnames</span>(start_res[[i]]) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"lineage(</span><span class="sc">\\</span><span class="st">d{1})|_lineage(</span><span class="sc">\\</span><span class="st">d{1})"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(start_res[[i]]))</span>
<span id="cb2-143"><a href="#cb2-143"></a>}</span>
<span id="cb2-144"><a href="#cb2-144"></a></span>
<span id="cb2-145"><a href="#cb2-145"></a>sorted_res <span class="ot">&lt;-</span> <span class="fu">map</span>(start_res, format_results)</span>
<span id="cb2-146"><a href="#cb2-146"></a></span>
<span id="cb2-147"><a href="#cb2-147"></a>dirf <span class="ot">&lt;-</span> </span>
<span id="cb2-148"><a href="#cb2-148"></a>  sorted_res <span class="sc">%&gt;%</span> </span>
<span id="cb2-149"><a href="#cb2-149"></a>  <span class="fu">map</span>(<span class="sc">~</span><span class="fu">arrange</span>(.x, fdr)) <span class="sc">%&gt;%</span> </span>
<span id="cb2-150"><a href="#cb2-150"></a>  <span class="fu">map</span>(<span class="sc">~</span><span class="fu">rownames_to_column</span>(.x, <span class="at">var=</span><span class="st">"gene"</span>))</span>
<span id="cb2-151"><a href="#cb2-151"></a>tot <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(dirf, <span class="at">.id =</span> <span class="st">"lineage"</span>)</span>
<span id="cb2-152"><a href="#cb2-152"></a><span class="co"># write_csv(tot, "23-04-12-pseudo_DGE.csv")</span></span>
<span id="cb2-153"><a href="#cb2-153"></a><span class="fu">addWorksheet</span>(wb, <span class="st">"start_vs_end"</span>)</span>
<span id="cb2-154"><a href="#cb2-154"></a><span class="fu">writeData</span>(wb, <span class="st">"start_vs_end"</span>, tot)</span>
<span id="cb2-155"><a href="#cb2-155"></a></span>
<span id="cb2-156"><a href="#cb2-156"></a><span class="fu">saveWorkbook</span>(wb, <span class="st">"23-04-12-pseudo_DGE.xlsx"</span>, <span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-157"><a href="#cb2-157"></a></span>
<span id="cb2-158"><a href="#cb2-158"></a>genes_sve <span class="ot">&lt;-</span><span class="fu">map</span>(sorted_res, <span class="sc">~</span>dplyr<span class="sc">::</span><span class="fu">filter</span>(.x, fdr <span class="sc">&lt;</span><span class="fl">0.05</span>)) </span>
<span id="cb2-159"><a href="#cb2-159"></a>clipr<span class="sc">::</span><span class="fu">write_clip</span>(genes_sve<span class="sc">$</span>global)</span>
<span id="cb2-160"><a href="#cb2-160"></a>clipr<span class="sc">::</span><span class="fu">write_clip</span>(genes_sve<span class="sc">$</span>lin_1)</span>
<span id="cb2-161"><a href="#cb2-161"></a>clipr<span class="sc">::</span><span class="fu">write_clip</span>(genes_sve<span class="sc">$</span>lin_2)</span>
<span id="cb2-162"><a href="#cb2-162"></a>clipr<span class="sc">::</span><span class="fu">write_clip</span>(genes_sve<span class="sc">$</span>lin_3)</span>
<span id="cb2-163"><a href="#cb2-163"></a>clipr<span class="sc">::</span><span class="fu">write_clip</span>(genes_sve<span class="sc">$</span>lin_4)</span>
<span id="cb2-164"><a href="#cb2-164"></a>clipr<span class="sc">::</span><span class="fu">write_clip</span>(genes_sve<span class="sc">$</span>lin_5)</span>
<span id="cb2-165"><a href="#cb2-165"></a></span>
<span id="cb2-166"><a href="#cb2-166"></a>enr_res <span class="ot">&lt;-</span> <span class="fu">enrichment_startvsend</span>(sorted_res)</span>
<span id="cb2-167"><a href="#cb2-167"></a></span>
<span id="cb2-168"><a href="#cb2-168"></a><span class="fu">map</span>(enr_res, <span class="sc">~</span></span>
<span id="cb2-169"><a href="#cb2-169"></a>      .x <span class="sc">%&gt;%</span> </span>
<span id="cb2-170"><a href="#cb2-170"></a>      dplyr<span class="sc">::</span><span class="fu">filter</span>(p.adjust <span class="sc">&lt;</span><span class="fl">0.05</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb2-171"><a href="#cb2-171"></a>  <span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"Lineage"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb2-172"><a href="#cb2-172"></a>  clipr<span class="sc">::</span><span class="fu">write_clip</span>()</span>
<span id="cb2-173"><a href="#cb2-173"></a></span>
<span id="cb2-174"><a href="#cb2-174"></a></span>
<span id="cb2-175"><a href="#cb2-175"></a><span class="fu">plot</span>(<span class="fu">reducedDims</span>(pto)<span class="sc">$</span>UMAP, <span class="at">col =</span> colors[pto<span class="sc">$</span>cluster_id], <span class="at">pch=</span><span class="dv">16</span>, <span class="at">asp =</span> <span class="dv">1</span>, <span class="at">cex=</span><span class="fl">0.5</span>)</span>
<span id="cb2-176"><a href="#cb2-176"></a><span class="co"># only plot first lineage</span></span>
<span id="cb2-177"><a href="#cb2-177"></a><span class="fu">lines</span>(<span class="fu">SlingshotDataSet</span>(pto), <span class="at">linInd=</span><span class="dv">1</span>)</span>
<span id="cb2-178"><a href="#cb2-178"></a><span class="co"># only plot second lineage</span></span>
<span id="cb2-179"><a href="#cb2-179"></a><span class="fu">lines</span>(<span class="fu">SlingshotDataSet</span>(pto), <span class="at">linInd=</span><span class="dv">2</span>)</span>
<span id="cb2-180"><a href="#cb2-180"></a><span class="co"># plot both lineages</span></span>
<span id="cb2-181"><a href="#cb2-181"></a><span class="fu">lines</span>(<span class="fu">SlingshotDataSet</span>(sce))</span>
<span id="cb2-182"><a href="#cb2-182"></a></span>
<span id="cb2-183"><a href="#cb2-183"></a>obj <span class="ot">&lt;-</span> </span>
<span id="cb2-184"><a href="#cb2-184"></a><span class="fu">plotGeneCount</span>(<span class="at">curve =</span> bbp, </span>
<span id="cb2-185"><a href="#cb2-185"></a>              <span class="at">clusters =</span> <span class="fu">apply</span>(<span class="fu">slingClusterLabels</span>(bbp), <span class="dv">1</span>, which.max),</span>
<span id="cb2-186"><a href="#cb2-186"></a>              <span class="at">models =</span> gam) </span>
<span id="cb2-187"><a href="#cb2-187"></a></span>
<span id="cb2-188"><a href="#cb2-188"></a></span>
<span id="cb2-189"><a href="#cb2-189"></a><span class="fu">plotGeneCount</span>(pto, counts, <span class="at">gene =</span> <span class="st">"NES"</span>)</span>
<span id="cb2-190"><a href="#cb2-190"></a><span class="fu">plotGeneCount</span>(pto, counts, <span class="at">gene =</span> <span class="st">"HIST1H4C"</span>)</span>
<span id="cb2-191"><a href="#cb2-191"></a><span class="fu">plotGeneCount</span>(pto, counts, <span class="at">gene =</span> <span class="st">"POSTN"</span>)</span>
<span id="cb2-192"><a href="#cb2-192"></a><span class="fu">plotSmoothers</span>(gam, counts, <span class="st">"NES"</span>)</span>
<span id="cb2-193"><a href="#cb2-193"></a><span class="fu">plotSmoothers</span>(gam, counts, <span class="st">"DES"</span>)</span>
<span id="cb2-194"><a href="#cb2-194"></a><span class="fu">plotSmoothers</span>(gam, counts, <span class="st">"H19"</span>)</span>
<span id="cb2-195"><a href="#cb2-195"></a><span class="fu">plotSmoothers</span>(gam, counts, <span class="st">"MYF5"</span>)</span>
<span id="cb2-196"><a href="#cb2-196"></a><span class="fu">plotSmoothers</span>(gam, counts, <span class="st">"POSTN"</span>)</span>
<span id="cb2-197"><a href="#cb2-197"></a><span class="fu">plotSmoothers</span>(gam, counts, <span class="st">"MYL1"</span>)</span>
<span id="cb2-198"><a href="#cb2-198"></a><span class="fu">plotSmoothers</span>(gam, counts, <span class="st">"MYOD1"</span>)</span>
<span id="cb2-199"><a href="#cb2-199"></a><span class="fu">plotSmoothers</span>(gam, counts, <span class="st">"TNNT1"</span>)</span>
<span id="cb2-200"><a href="#cb2-200"></a><span class="fu">plotSmoothers</span>(gam, counts, <span class="st">"BRCA1"</span>)</span>
<span id="cb2-201"><a href="#cb2-201"></a><span class="fu">plotSmoothers</span>(gam, counts, <span class="st">"HIST1H4C"</span>) <span class="sc">+</span></span>
<span id="cb2-202"><a href="#cb2-202"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>lineage)</span>
<span id="cb2-203"><a href="#cb2-203"></a></span>
<span id="cb2-204"><a href="#cb2-204"></a>curve.assignments <span class="ot">&lt;-</span> <span class="fu">slingBranchID</span>(pto)</span>
<span id="cb2-205"><a href="#cb2-205"></a><span class="fu">table</span>(curve.assignments)</span>
<span id="cb2-206"><a href="#cb2-206"></a></span>
<span id="cb2-207"><a href="#cb2-207"></a><span class="co"># Between lineages (endpoint) ---------------------------------------------</span></span>
<span id="cb2-208"><a href="#cb2-208"></a></span>
<span id="cb2-209"><a href="#cb2-209"></a><span class="co"># Use previous to pick two lineages then study difference?</span></span>
<span id="cb2-210"><a href="#cb2-210"></a></span>
<span id="cb2-211"><a href="#cb2-211"></a>wb <span class="ot">&lt;-</span> <span class="fu">createWorkbook</span>()</span>
<span id="cb2-212"><a href="#cb2-212"></a></span>
<span id="cb2-213"><a href="#cb2-213"></a>bls <span class="ot">&lt;-</span> <span class="fu">diffEndTest</span>(gam, <span class="at">global=</span>T, <span class="at">pairwise=</span>T)</span>
<span id="cb2-214"><a href="#cb2-214"></a></span>
<span id="cb2-215"><a href="#cb2-215"></a>two_vs_everything <span class="ot">&lt;-</span> <span class="fu">select</span>(bls, waldStat, df, pvalue, <span class="fu">contains</span>(<span class="st">"2"</span>))</span>
<span id="cb2-216"><a href="#cb2-216"></a></span>
<span id="cb2-217"><a href="#cb2-217"></a>obs <span class="ot">&lt;-</span> <span class="fu">map</span>(<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">5</span>), <span class="sc">~</span><span class="fu">arts</span>(.x,two_vs_everything)) </span>
<span id="cb2-218"><a href="#cb2-218"></a>obs <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">set_names</span>(obs, <span class="fu">c</span>(<span class="st">"lin1"</span>, <span class="st">"lin3"</span>, <span class="st">"lin4"</span>, <span class="st">"lin5"</span>)) </span>
<span id="cb2-219"><a href="#cb2-219"></a></span>
<span id="cb2-220"><a href="#cb2-220"></a>obs <span class="ot">&lt;-</span> </span>
<span id="cb2-221"><a href="#cb2-221"></a>  obs <span class="sc">%&gt;%</span> </span>
<span id="cb2-222"><a href="#cb2-222"></a>  <span class="fu">map</span>(<span class="sc">~</span><span class="fu">sort_list_by_colname</span>(.x, <span class="st">"wald"</span>, <span class="at">dec=</span>T)) <span class="sc">%&gt;%</span> </span>
<span id="cb2-223"><a href="#cb2-223"></a>  <span class="fu">map</span>(<span class="sc">~</span><span class="fu">filter_list_by_colname</span>(.x, <span class="st">"pvalue"</span>, <span class="at">sig_level=</span><span class="fl">0.05</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb2-224"><a href="#cb2-224"></a>  <span class="fu">map</span>(<span class="sc">~</span><span class="fu">rownames_to_column</span>(.x, <span class="at">var=</span><span class="st">"gene"</span>))</span>
<span id="cb2-225"><a href="#cb2-225"></a></span>
<span id="cb2-226"><a href="#cb2-226"></a></span>
<span id="cb2-227"><a href="#cb2-227"></a><span class="co"># Iterate through the list of data frames and add each as a sheet to the workbook</span></span>
<span id="cb2-228"><a href="#cb2-228"></a><span class="cf">for</span> (sheet_name <span class="cf">in</span> <span class="fu">names</span>(obs)) {</span>
<span id="cb2-229"><a href="#cb2-229"></a>  <span class="fu">addWorksheet</span>(wb, sheet_name)</span>
<span id="cb2-230"><a href="#cb2-230"></a>  <span class="fu">writeData</span>(wb, sheet_name, obs[[sheet_name]])</span>
<span id="cb2-231"><a href="#cb2-231"></a>}</span>
<span id="cb2-232"><a href="#cb2-232"></a></span>
<span id="cb2-233"><a href="#cb2-233"></a><span class="co"># Save the workbook to a file</span></span>
<span id="cb2-234"><a href="#cb2-234"></a><span class="fu">saveWorkbook</span>(wb, <span class="st">"results/23-04-12-pseudo_between_lineages_endpoint.xlsx"</span>, <span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-235"><a href="#cb2-235"></a></span>
<span id="cb2-236"><a href="#cb2-236"></a><span class="co"># Between lineage (body) --------------------------------------------------</span></span>
<span id="cb2-237"><a href="#cb2-237"></a></span>
<span id="cb2-238"><a href="#cb2-238"></a>patternRes <span class="ot">&lt;-</span> <span class="fu">patternTest</span>(gam, <span class="at">global=</span>T, <span class="at">pairwise=</span>T)</span>
<span id="cb2-239"><a href="#cb2-239"></a>two_vs_everything <span class="ot">&lt;-</span> <span class="fu">select</span>(patternRes, waldStat, df, pvalue, <span class="fu">contains</span>(<span class="st">"2"</span>))</span>
<span id="cb2-240"><a href="#cb2-240"></a></span>
<span id="cb2-241"><a href="#cb2-241"></a>obs <span class="ot">&lt;-</span> <span class="fu">map</span>(<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">5</span>), <span class="sc">~</span><span class="fu">arts</span>(.x,two_vs_everything)) </span>
<span id="cb2-242"><a href="#cb2-242"></a>obs <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">set_names</span>(obs, <span class="fu">c</span>(<span class="st">"lin1"</span>, <span class="st">"lin3"</span>, <span class="st">"lin4"</span>, <span class="st">"lin5"</span>)) </span>
<span id="cb2-243"><a href="#cb2-243"></a></span>
<span id="cb2-244"><a href="#cb2-244"></a>obs <span class="ot">&lt;-</span> </span>
<span id="cb2-245"><a href="#cb2-245"></a>  obs <span class="sc">%&gt;%</span> </span>
<span id="cb2-246"><a href="#cb2-246"></a>  <span class="fu">map</span>(<span class="sc">~</span><span class="fu">sort_list_by_colname</span>(.x, <span class="st">"wald"</span>, <span class="at">dec=</span>T)) <span class="sc">%&gt;%</span> </span>
<span id="cb2-247"><a href="#cb2-247"></a>  <span class="fu">map</span>(<span class="sc">~</span><span class="fu">filter_list_by_colname</span>(.x, <span class="st">"pvalue"</span>, <span class="at">sig_level=</span><span class="fl">0.05</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb2-248"><a href="#cb2-248"></a>  <span class="fu">map</span>(<span class="sc">~</span><span class="fu">rownames_to_column</span>(.x, <span class="at">var=</span><span class="st">"gene"</span>))</span>
<span id="cb2-249"><a href="#cb2-249"></a></span>
<span id="cb2-250"><a href="#cb2-250"></a>wb <span class="ot">&lt;-</span> <span class="fu">createWorkbook</span>()</span>
<span id="cb2-251"><a href="#cb2-251"></a></span>
<span id="cb2-252"><a href="#cb2-252"></a><span class="co"># Iterate through the list of data frames and add each as a sheet to the workbook</span></span>
<span id="cb2-253"><a href="#cb2-253"></a><span class="cf">for</span> (sheet_name <span class="cf">in</span> <span class="fu">names</span>(obs)) {</span>
<span id="cb2-254"><a href="#cb2-254"></a>  <span class="fu">addWorksheet</span>(wb, sheet_name)</span>
<span id="cb2-255"><a href="#cb2-255"></a>  <span class="fu">writeData</span>(wb, sheet_name, obs[[sheet_name]])</span>
<span id="cb2-256"><a href="#cb2-256"></a>}</span>
<span id="cb2-257"><a href="#cb2-257"></a></span>
<span id="cb2-258"><a href="#cb2-258"></a><span class="co"># Save the workbook to a file</span></span>
<span id="cb2-259"><a href="#cb2-259"></a><span class="fu">saveWorkbook</span>(wb, <span class="st">"results/23-04-12-pseudo_between_lineages_body.xlsx"</span>, <span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-260"><a href="#cb2-260"></a></span>
<span id="cb2-261"><a href="#cb2-261"></a></span>
<span id="cb2-262"><a href="#cb2-262"></a><span class="co"># Curve assignments -------------------------------------------------------</span></span>
<span id="cb2-263"><a href="#cb2-263"></a><span class="do">## Stack plots for cell / sample characteristics per lineage</span></span>
<span id="cb2-264"><a href="#cb2-264"></a></span>
<span id="cb2-265"><a href="#cb2-265"></a>branch_ids <span class="ot">&lt;-</span> <span class="fu">slingBranchID</span>(pto)</span>
<span id="cb2-266"><a href="#cb2-266"></a>branchids_df <span class="ot">&lt;-</span> <span class="fu">enframe</span>(branch_ids, <span class="at">name=</span><span class="st">"cell_id"</span>, <span class="at">value=</span><span class="st">"lineage"</span>)</span>
<span id="cb2-267"><a href="#cb2-267"></a>branchids_df2 <span class="ot">&lt;-</span> </span>
<span id="cb2-268"><a href="#cb2-268"></a>  branchids_df <span class="sc">%&gt;%</span> </span>
<span id="cb2-269"><a href="#cb2-269"></a>  <span class="fu">separate</span>(cell_id, </span>
<span id="cb2-270"><a href="#cb2-270"></a>           <span class="at">sep=</span><span class="st">"_"</span>, </span>
<span id="cb2-271"><a href="#cb2-271"></a>           <span class="at">into=</span><span class="fu">c</span>(<span class="st">"sample_id"</span>, <span class="st">"barcode"</span>), </span>
<span id="cb2-272"><a href="#cb2-272"></a>           <span class="at">convert=</span><span class="cn">TRUE</span>,</span>
<span id="cb2-273"><a href="#cb2-273"></a>           <span class="at">remove=</span><span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb2-274"><a href="#cb2-274"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(sample_id, as_factor))</span>
<span id="cb2-275"><a href="#cb2-275"></a></span>
<span id="cb2-276"><a href="#cb2-276"></a>ibs <span class="ot">&lt;-</span> <span class="fu">colData</span>(pto)</span>
<span id="cb2-277"><a href="#cb2-277"></a>ibs <span class="ot">&lt;-</span> </span>
<span id="cb2-278"><a href="#cb2-278"></a>  <span class="fu">tibble</span>(<span class="st">"id"</span> <span class="ot">=</span> <span class="fu">rownames</span>(ibs),</span>
<span id="cb2-279"><a href="#cb2-279"></a>         <span class="st">"broad_samples"</span> <span class="ot">=</span> ibs<span class="sc">$</span>ID,</span>
<span id="cb2-280"><a href="#cb2-280"></a>         <span class="st">"status_binary"</span> <span class="ot">=</span> ibs<span class="sc">$</span>status_binary,</span>
<span id="cb2-281"><a href="#cb2-281"></a>         <span class="st">"grip"</span> <span class="ot">=</span> ibs<span class="sc">$</span>xamaxgrip_quart,</span>
<span id="cb2-282"><a href="#cb2-282"></a>         <span class="st">"bmi"</span> <span class="ot">=</span> ibs<span class="sc">$</span>xabmi_quart,</span>
<span id="cb2-283"><a href="#cb2-283"></a>         <span class="st">"almi"</span> <span class="ot">=</span> ibs<span class="sc">$</span>almhsq_quart,</span>
<span id="cb2-284"><a href="#cb2-284"></a>         <span class="st">"gait"</span> <span class="ot">=</span> ibs<span class="sc">$</span>xwspdms_quart)</span>
<span id="cb2-285"><a href="#cb2-285"></a></span>
<span id="cb2-286"><a href="#cb2-286"></a>branchids_df2 <span class="ot">&lt;-</span> <span class="fu">left_join</span>(branchids_df2, ibs, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"cell_id"</span> <span class="ot">=</span> <span class="st">"id"</span>) )</span>
<span id="cb2-287"><a href="#cb2-287"></a></span>
<span id="cb2-288"><a href="#cb2-288"></a><span class="fu">ggplot</span>(branchids_df2, <span class="fu">aes</span>(<span class="at">x=</span>lineage, <span class="at">fill=</span>gait)) <span class="sc">+</span> </span>
<span id="cb2-289"><a href="#cb2-289"></a>  <span class="fu">geom_bar</span>() </span>
<span id="cb2-290"><a href="#cb2-290"></a></span>
<span id="cb2-291"><a href="#cb2-291"></a>stack_for_variable <span class="ot">&lt;-</span> <span class="cf">function</span>(variable){</span>
<span id="cb2-292"><a href="#cb2-292"></a>  variable <span class="ot">&lt;-</span> <span class="fu">enquo</span>(variable)</span>
<span id="cb2-293"><a href="#cb2-293"></a>tarp <span class="ot">&lt;-</span></span>
<span id="cb2-294"><a href="#cb2-294"></a>branchids_df2 <span class="sc">%&gt;%</span></span>
<span id="cb2-295"><a href="#cb2-295"></a>  <span class="fu">group_by</span>(lineage, <span class="sc">!!</span>variable) <span class="sc">%&gt;%</span></span>
<span id="cb2-296"><a href="#cb2-296"></a>  <span class="fu">tally</span>()</span>
<span id="cb2-297"><a href="#cb2-297"></a></span>
<span id="cb2-298"><a href="#cb2-298"></a>tarp <span class="ot">&lt;-</span></span>
<span id="cb2-299"><a href="#cb2-299"></a>  <span class="fu">filter</span>(tarp, lineage <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"1,2,3,4,5"</span>,</span>
<span id="cb2-300"><a href="#cb2-300"></a>                              <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>, <span class="st">"4"</span>, <span class="st">"5"</span>))</span>
<span id="cb2-301"><a href="#cb2-301"></a></span>
<span id="cb2-302"><a href="#cb2-302"></a>totals <span class="ot">&lt;-</span> </span>
<span id="cb2-303"><a href="#cb2-303"></a>  tarp <span class="sc">%&gt;%</span></span>
<span id="cb2-304"><a href="#cb2-304"></a>  <span class="fu">group_by</span>(lineage) <span class="sc">%&gt;%</span></span>
<span id="cb2-305"><a href="#cb2-305"></a>  <span class="fu">summarize</span>(<span class="at">total =</span> <span class="fu">sum</span>(n))</span>
<span id="cb2-306"><a href="#cb2-306"></a></span>
<span id="cb2-307"><a href="#cb2-307"></a>tarp<span class="sc">$</span>lineage <span class="ot">&lt;-</span></span>
<span id="cb2-308"><a href="#cb2-308"></a>  <span class="fu">factor</span>(tarp<span class="sc">$</span>lineage, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"1,2,3,4,5"</span>,</span>
<span id="cb2-309"><a href="#cb2-309"></a>                                    <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>, <span class="st">"4"</span>, <span class="st">"5"</span>))</span>
<span id="cb2-310"><a href="#cb2-310"></a></span>
<span id="cb2-311"><a href="#cb2-311"></a><span class="fu">ggplot</span>(tarp, <span class="fu">aes</span>(<span class="at">fill=</span><span class="sc">!!</span>variable, <span class="at">y=</span>n, <span class="at">x=</span>lineage)) <span class="sc">+</span> </span>
<span id="cb2-312"><a href="#cb2-312"></a>  <span class="fu">geom_bar</span>(<span class="at">position=</span><span class="st">"fill"</span>, <span class="at">stat=</span><span class="st">"identity"</span>) </span>
<span id="cb2-313"><a href="#cb2-313"></a>}</span>
<span id="cb2-314"><a href="#cb2-314"></a></span>
<span id="cb2-315"><a href="#cb2-315"></a><span class="fu">stack_for_variable</span>(gait)</span>
<span id="cb2-316"><a href="#cb2-316"></a><span class="fu">stack_for_variable</span>(almi)</span>
<span id="cb2-317"><a href="#cb2-317"></a><span class="fu">stack_for_variable</span>(grip)</span>
<span id="cb2-318"><a href="#cb2-318"></a></span>
<span id="cb2-319"><a href="#cb2-319"></a></span>
<span id="cb2-320"><a href="#cb2-320"></a><span class="co"># Chi-sq ------------------------------------------------------------------</span></span>
<span id="cb2-321"><a href="#cb2-321"></a><span class="do">## Chi-sq to confirm whether differences in proportions seen in previous stacked</span></span>
<span id="cb2-322"><a href="#cb2-322"></a><span class="do">## bars are significant</span></span>
<span id="cb2-323"><a href="#cb2-323"></a></span>
<span id="cb2-324"><a href="#cb2-324"></a><span class="do">### almi</span></span>
<span id="cb2-325"><a href="#cb2-325"></a>cs_test <span class="ot">&lt;-</span> </span>
<span id="cb2-326"><a href="#cb2-326"></a>  branchids_df2 <span class="sc">%&gt;%</span> </span>
<span id="cb2-327"><a href="#cb2-327"></a>  <span class="fu">group_by</span>(lineage, almi) <span class="sc">%&gt;%</span> </span>
<span id="cb2-328"><a href="#cb2-328"></a>  <span class="fu">tally</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb2-329"><a href="#cb2-329"></a>  <span class="fu">drop_na</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb2-330"><a href="#cb2-330"></a>  <span class="fu">mutate</span>(<span class="at">almi =</span> <span class="fu">paste0</span>(<span class="st">"quartile_"</span>, almi)) <span class="sc">%&gt;%</span> </span>
<span id="cb2-331"><a href="#cb2-331"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> almi, <span class="at">values_from =</span> n) <span class="sc">%&gt;%</span> </span>
<span id="cb2-332"><a href="#cb2-332"></a>  <span class="fu">filter</span>(lineage <span class="sc">%in%</span> <span class="fu">c</span>( <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>, <span class="st">"4"</span>, <span class="st">"5"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb2-333"><a href="#cb2-333"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb2-334"><a href="#cb2-334"></a></span>
<span id="cb2-335"><a href="#cb2-335"></a>x <span class="ot">&lt;-</span> <span class="fu">as.table</span>(<span class="fu">cbind</span>(cs_test<span class="sc">$</span>quartile_1, </span>
<span id="cb2-336"><a href="#cb2-336"></a>                    cs_test<span class="sc">$</span>quartile_2,</span>
<span id="cb2-337"><a href="#cb2-337"></a>                    cs_test<span class="sc">$</span>quartile_3,</span>
<span id="cb2-338"><a href="#cb2-338"></a>                    cs_test<span class="sc">$</span>quartile_4))</span>
<span id="cb2-339"><a href="#cb2-339"></a><span class="fu">dimnames</span>(x) <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">lineage =</span> <span class="fu">c</span>(<span class="fu">paste0</span>(<span class="st">"lineage_"</span>, <span class="fu">seq</span>(<span class="dv">1</span>,<span class="dv">5</span>,<span class="dv">1</span>))),</span>
<span id="cb2-340"><a href="#cb2-340"></a>                    <span class="at">quartile =</span> <span class="fu">c</span>(<span class="st">"quartile_1"</span>, </span>
<span id="cb2-341"><a href="#cb2-341"></a>                                 <span class="st">"quartile_2"</span>, </span>
<span id="cb2-342"><a href="#cb2-342"></a>                                 <span class="st">"quartile_3"</span>, </span>
<span id="cb2-343"><a href="#cb2-343"></a>                                 <span class="st">"quartile_4"</span>))</span>
<span id="cb2-344"><a href="#cb2-344"></a>nabs <span class="ot">&lt;-</span> <span class="fu">chisq.test</span>(x)</span>
<span id="cb2-345"><a href="#cb2-345"></a>corrplot<span class="sc">::</span><span class="fu">corrplot</span>(nabs<span class="sc">$</span>residuals, <span class="at">is.cor =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2-346"><a href="#cb2-346"></a>contrib <span class="ot">&lt;-</span> <span class="dv">100</span><span class="sc">*</span>nabs<span class="sc">$</span>residuals<span class="sc">^</span><span class="dv">2</span><span class="sc">/</span>nabs<span class="sc">$</span>statistic</span>
<span id="cb2-347"><a href="#cb2-347"></a><span class="fu">round</span>(contrib, <span class="dv">3</span>)</span>
<span id="cb2-348"><a href="#cb2-348"></a>corrplot<span class="sc">::</span><span class="fu">corrplot</span>(contrib, <span class="at">is.cor =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2-349"><a href="#cb2-349"></a></span>
<span id="cb2-350"><a href="#cb2-350"></a><span class="do">### Grip</span></span>
<span id="cb2-351"><a href="#cb2-351"></a>cs_test <span class="ot">&lt;-</span> </span>
<span id="cb2-352"><a href="#cb2-352"></a>  branchids_df2 <span class="sc">%&gt;%</span> </span>
<span id="cb2-353"><a href="#cb2-353"></a>  <span class="fu">group_by</span>(lineage, grip) <span class="sc">%&gt;%</span> </span>
<span id="cb2-354"><a href="#cb2-354"></a>  <span class="fu">tally</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb2-355"><a href="#cb2-355"></a>  <span class="fu">drop_na</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb2-356"><a href="#cb2-356"></a>  <span class="fu">mutate</span>(<span class="at">grip =</span> <span class="fu">paste0</span>(<span class="st">"quartile_"</span>, grip)) <span class="sc">%&gt;%</span> </span>
<span id="cb2-357"><a href="#cb2-357"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> grip, <span class="at">values_from =</span> n) <span class="sc">%&gt;%</span> </span>
<span id="cb2-358"><a href="#cb2-358"></a>  <span class="fu">filter</span>(lineage <span class="sc">%in%</span> <span class="fu">c</span>( <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>, <span class="st">"4"</span>, <span class="st">"5"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb2-359"><a href="#cb2-359"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb2-360"><a href="#cb2-360"></a></span>
<span id="cb2-361"><a href="#cb2-361"></a>x <span class="ot">&lt;-</span> <span class="fu">as.table</span>(<span class="fu">cbind</span>(cs_test<span class="sc">$</span>quartile_1, </span>
<span id="cb2-362"><a href="#cb2-362"></a>                    cs_test<span class="sc">$</span>quartile_2,</span>
<span id="cb2-363"><a href="#cb2-363"></a>                    cs_test<span class="sc">$</span>quartile_3,</span>
<span id="cb2-364"><a href="#cb2-364"></a>                    cs_test<span class="sc">$</span>quartile_4))</span>
<span id="cb2-365"><a href="#cb2-365"></a><span class="fu">dimnames</span>(x) <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">lineage =</span> <span class="fu">c</span>(<span class="fu">paste0</span>(<span class="st">"lineage_"</span>, <span class="fu">seq</span>(<span class="dv">1</span>,<span class="dv">5</span>,<span class="dv">1</span>))),</span>
<span id="cb2-366"><a href="#cb2-366"></a>                    <span class="at">quartile =</span> <span class="fu">c</span>(<span class="st">"quartile_1"</span>, </span>
<span id="cb2-367"><a href="#cb2-367"></a>                                 <span class="st">"quartile_2"</span>, </span>
<span id="cb2-368"><a href="#cb2-368"></a>                                 <span class="st">"quartile_3"</span>, </span>
<span id="cb2-369"><a href="#cb2-369"></a>                                 <span class="st">"quartile_4"</span>))</span>
<span id="cb2-370"><a href="#cb2-370"></a>nabs <span class="ot">&lt;-</span> <span class="fu">chisq.test</span>(x)</span>
<span id="cb2-371"><a href="#cb2-371"></a>corrplot<span class="sc">::</span><span class="fu">corrplot</span>(nabs<span class="sc">$</span>residuals, <span class="at">is.cor =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2-372"><a href="#cb2-372"></a>contrib <span class="ot">&lt;-</span> <span class="dv">100</span><span class="sc">*</span>nabs<span class="sc">$</span>residuals<span class="sc">^</span><span class="dv">2</span><span class="sc">/</span>nabs<span class="sc">$</span>statistic</span>
<span id="cb2-373"><a href="#cb2-373"></a><span class="fu">round</span>(contrib, <span class="dv">3</span>)</span>
<span id="cb2-374"><a href="#cb2-374"></a>corrplot<span class="sc">::</span><span class="fu">corrplot</span>(contrib, <span class="at">is.cor =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2-375"><a href="#cb2-375"></a></span>
<span id="cb2-376"><a href="#cb2-376"></a><span class="do">### gait</span></span>
<span id="cb2-377"><a href="#cb2-377"></a>cs_test <span class="ot">&lt;-</span> </span>
<span id="cb2-378"><a href="#cb2-378"></a>  branchids_df2 <span class="sc">%&gt;%</span> </span>
<span id="cb2-379"><a href="#cb2-379"></a>  <span class="fu">group_by</span>(lineage, gait) <span class="sc">%&gt;%</span> </span>
<span id="cb2-380"><a href="#cb2-380"></a>  <span class="fu">tally</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb2-381"><a href="#cb2-381"></a>  <span class="fu">drop_na</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb2-382"><a href="#cb2-382"></a>  <span class="fu">mutate</span>(<span class="at">gait =</span> <span class="fu">paste0</span>(<span class="st">"quartile_"</span>, gait)) <span class="sc">%&gt;%</span> </span>
<span id="cb2-383"><a href="#cb2-383"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> gait, <span class="at">values_from =</span> n) <span class="sc">%&gt;%</span> </span>
<span id="cb2-384"><a href="#cb2-384"></a>  <span class="fu">filter</span>(lineage <span class="sc">%in%</span> <span class="fu">c</span>( <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>, <span class="st">"4"</span>, <span class="st">"5"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb2-385"><a href="#cb2-385"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb2-386"><a href="#cb2-386"></a></span>
<span id="cb2-387"><a href="#cb2-387"></a>x <span class="ot">&lt;-</span> <span class="fu">as.table</span>(<span class="fu">cbind</span>(cs_test<span class="sc">$</span>quartile_1, </span>
<span id="cb2-388"><a href="#cb2-388"></a>                    cs_test<span class="sc">$</span>quartile_2,</span>
<span id="cb2-389"><a href="#cb2-389"></a>                    cs_test<span class="sc">$</span>quartile_3,</span>
<span id="cb2-390"><a href="#cb2-390"></a>                    cs_test<span class="sc">$</span>quartile_4))</span>
<span id="cb2-391"><a href="#cb2-391"></a><span class="fu">dimnames</span>(x) <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">lineage =</span> <span class="fu">c</span>(<span class="fu">paste0</span>(<span class="st">"lineage_"</span>, <span class="fu">seq</span>(<span class="dv">1</span>,<span class="dv">5</span>,<span class="dv">1</span>))),</span>
<span id="cb2-392"><a href="#cb2-392"></a>                    <span class="at">quartile =</span> <span class="fu">c</span>(<span class="st">"quartile_1"</span>, </span>
<span id="cb2-393"><a href="#cb2-393"></a>                                 <span class="st">"quartile_2"</span>, </span>
<span id="cb2-394"><a href="#cb2-394"></a>                                 <span class="st">"quartile_3"</span>, </span>
<span id="cb2-395"><a href="#cb2-395"></a>                                 <span class="st">"quartile_4"</span>))</span>
<span id="cb2-396"><a href="#cb2-396"></a>nabs <span class="ot">&lt;-</span> <span class="fu">chisq.test</span>(x)</span>
<span id="cb2-397"><a href="#cb2-397"></a>corrplot<span class="sc">::</span><span class="fu">corrplot</span>(nabs<span class="sc">$</span>residuals, <span class="at">is.cor =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2-398"><a href="#cb2-398"></a>contrib <span class="ot">&lt;-</span> <span class="dv">100</span><span class="sc">*</span>nabs<span class="sc">$</span>residuals<span class="sc">^</span><span class="dv">2</span><span class="sc">/</span>nabs<span class="sc">$</span>statistic</span>
<span id="cb2-399"><a href="#cb2-399"></a><span class="fu">round</span>(contrib, <span class="dv">3</span>)</span>
<span id="cb2-400"><a href="#cb2-400"></a>corrplot<span class="sc">::</span><span class="fu">corrplot</span>(contrib, <span class="at">is.cor =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2-401"><a href="#cb2-401"></a></span>
<span id="cb2-402"><a href="#cb2-402"></a><span class="do">### This is a version which includes the root branch, i.e. 1,2,3,4 and 5</span></span>
<span id="cb2-403"><a href="#cb2-403"></a>cs_test <span class="ot">&lt;-</span> </span>
<span id="cb2-404"><a href="#cb2-404"></a>  branchids_df2 <span class="sc">%&gt;%</span> </span>
<span id="cb2-405"><a href="#cb2-405"></a>  <span class="fu">group_by</span>(lineage, gait) <span class="sc">%&gt;%</span> </span>
<span id="cb2-406"><a href="#cb2-406"></a>  <span class="fu">tally</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb2-407"><a href="#cb2-407"></a>  <span class="fu">drop_na</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb2-408"><a href="#cb2-408"></a>  <span class="fu">mutate</span>(<span class="at">gait =</span> <span class="fu">paste0</span>(<span class="st">"quartile_"</span>, gait)) <span class="sc">%&gt;%</span> </span>
<span id="cb2-409"><a href="#cb2-409"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> gait, <span class="at">values_from =</span> n) <span class="sc">%&gt;%</span> </span>
<span id="cb2-410"><a href="#cb2-410"></a>  <span class="fu">filter</span>(lineage <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"1,2,3,4,5"</span>, </span>
<span id="cb2-411"><a href="#cb2-411"></a>    <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>, <span class="st">"4"</span>, <span class="st">"5"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb2-412"><a href="#cb2-412"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb2-413"><a href="#cb2-413"></a></span>
<span id="cb2-414"><a href="#cb2-414"></a>x <span class="ot">&lt;-</span> <span class="fu">as.table</span>(<span class="fu">cbind</span>(cs_test<span class="sc">$</span>quartile_1, </span>
<span id="cb2-415"><a href="#cb2-415"></a>                    cs_test<span class="sc">$</span>quartile_2,</span>
<span id="cb2-416"><a href="#cb2-416"></a>                    cs_test<span class="sc">$</span>quartile_3,</span>
<span id="cb2-417"><a href="#cb2-417"></a>                    cs_test<span class="sc">$</span>quartile_4))</span>
<span id="cb2-418"><a href="#cb2-418"></a><span class="fu">dimnames</span>(x) <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">lineage =</span> <span class="fu">c</span>(<span class="fu">c</span>(<span class="st">"lineage_1"</span>, <span class="st">"lineage_1,2,3,4,5"</span>), <span class="fu">c</span>(<span class="fu">paste0</span>(<span class="st">"lineage_"</span>, <span class="fu">seq</span>(<span class="dv">2</span>,<span class="dv">5</span>,<span class="dv">1</span>)))),</span>
<span id="cb2-419"><a href="#cb2-419"></a>                    <span class="at">quartile =</span> <span class="fu">c</span>(<span class="st">"quartile_1"</span>, </span>
<span id="cb2-420"><a href="#cb2-420"></a>                                 <span class="st">"quartile_2"</span>, </span>
<span id="cb2-421"><a href="#cb2-421"></a>                                 <span class="st">"quartile_3"</span>, </span>
<span id="cb2-422"><a href="#cb2-422"></a>                                 <span class="st">"quartile_4"</span>))</span>
<span id="cb2-423"><a href="#cb2-423"></a>nabs <span class="ot">&lt;-</span> <span class="fu">chisq.test</span>(x)</span>
<span id="cb2-424"><a href="#cb2-424"></a>corrplot<span class="sc">::</span><span class="fu">corrplot</span>(nabs<span class="sc">$</span>residuals, <span class="at">is.cor =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2-425"><a href="#cb2-425"></a>contrib <span class="ot">&lt;-</span> <span class="dv">100</span><span class="sc">*</span>nabs<span class="sc">$</span>residuals<span class="sc">^</span><span class="dv">2</span><span class="sc">/</span>nabs<span class="sc">$</span>statistic</span>
<span id="cb2-426"><a href="#cb2-426"></a><span class="fu">round</span>(contrib, <span class="dv">3</span>)</span>
<span id="cb2-427"><a href="#cb2-427"></a>corrplot<span class="sc">::</span><span class="fu">corrplot</span>(contrib, <span class="at">is.cor =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2-428"><a href="#cb2-428"></a></span>
<span id="cb2-429"><a href="#cb2-429"></a></span>
<span id="cb2-430"><a href="#cb2-430"></a><span class="do">### More direct way of doing the chi-sq plots</span></span>
<span id="cb2-431"><a href="#cb2-431"></a>c_sq_one <span class="ot">&lt;-</span> <span class="cf">function</span>(df, variable){</span>
<span id="cb2-432"><a href="#cb2-432"></a>  <span class="co"># variable &lt;- enquo(variable)</span></span>
<span id="cb2-433"><a href="#cb2-433"></a>  <span class="co"># One lineage</span></span>
<span id="cb2-434"><a href="#cb2-434"></a>  cs_test <span class="ot">&lt;-</span> </span>
<span id="cb2-435"><a href="#cb2-435"></a>    df <span class="sc">%&gt;%</span> </span>
<span id="cb2-436"><a href="#cb2-436"></a>    <span class="fu">group_by</span>(lineage, {{variable}}) <span class="sc">%&gt;%</span> </span>
<span id="cb2-437"><a href="#cb2-437"></a>    <span class="fu">tally</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb2-438"><a href="#cb2-438"></a>    <span class="fu">drop_na</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb2-439"><a href="#cb2-439"></a>    <span class="fu">mutate</span>({{variable}}<span class="sc">:</span><span class="er">=</span> <span class="fu">paste0</span>(<span class="st">"quartile_"</span>, {{variable}})) <span class="sc">%&gt;%</span> </span>
<span id="cb2-440"><a href="#cb2-440"></a>    <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> {{variable}}, <span class="at">values_from =</span> n) <span class="sc">%&gt;%</span> </span>
<span id="cb2-441"><a href="#cb2-441"></a>    <span class="fu">filter</span>(lineage <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>, <span class="st">"4"</span>, <span class="st">"5"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb2-442"><a href="#cb2-442"></a>    <span class="fu">as.data.frame</span>()</span>
<span id="cb2-443"><a href="#cb2-443"></a>  </span>
<span id="cb2-444"><a href="#cb2-444"></a>  x <span class="ot">&lt;-</span> <span class="fu">as.table</span>(<span class="fu">cbind</span>(cs_test<span class="sc">$</span>quartile_1,</span>
<span id="cb2-445"><a href="#cb2-445"></a>                      cs_test<span class="sc">$</span>quartile_2,</span>
<span id="cb2-446"><a href="#cb2-446"></a>                      cs_test<span class="sc">$</span>quartile_3,</span>
<span id="cb2-447"><a href="#cb2-447"></a>                      cs_test<span class="sc">$</span>quartile_4))</span>
<span id="cb2-448"><a href="#cb2-448"></a>  <span class="fu">dimnames</span>(x) <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">lineage =</span> <span class="fu">c</span>(<span class="fu">paste0</span>(<span class="st">"lineage_"</span>, <span class="fu">seq</span>(<span class="dv">1</span>,<span class="dv">5</span>,<span class="dv">1</span>))),</span>
<span id="cb2-449"><a href="#cb2-449"></a>                      <span class="at">quartile =</span> <span class="fu">c</span>(<span class="st">"quartile_1"</span>,</span>
<span id="cb2-450"><a href="#cb2-450"></a>                                   <span class="st">"quartile_2"</span>,</span>
<span id="cb2-451"><a href="#cb2-451"></a>                                   <span class="st">"quartile_3"</span>,</span>
<span id="cb2-452"><a href="#cb2-452"></a>                                   <span class="st">"quartile_4"</span>))</span>
<span id="cb2-453"><a href="#cb2-453"></a>  nabs <span class="ot">&lt;-</span> <span class="fu">chisq.test</span>(x)</span>
<span id="cb2-454"><a href="#cb2-454"></a>  corrplot<span class="sc">::</span><span class="fu">corrplot</span>(nabs<span class="sc">$</span>residuals, <span class="at">is.cor =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2-455"><a href="#cb2-455"></a>  <span class="co"># contrib &lt;- 100*nabs$residuals^2/nabs$statistic</span></span>
<span id="cb2-456"><a href="#cb2-456"></a>  <span class="co"># round(contrib, 3)</span></span>
<span id="cb2-457"><a href="#cb2-457"></a>  <span class="co"># corrplot(contrib, is.cor = FALSE)</span></span>
<span id="cb2-458"><a href="#cb2-458"></a>  <span class="fu">message</span>(<span class="fu">deparse</span>(<span class="fu">substitute</span>(variable)), <span class="st">" p_value ="</span>, <span class="fu">format</span>(nabs<span class="sc">$</span>p.value, <span class="at">digits=</span><span class="dv">3</span>))</span>
<span id="cb2-459"><a href="#cb2-459"></a>  <span class="fu">return</span>(nabs)</span>
<span id="cb2-460"><a href="#cb2-460"></a>}</span>
<span id="cb2-461"><a href="#cb2-461"></a></span>
<span id="cb2-462"><a href="#cb2-462"></a><span class="fu">c_sq_one</span>(branchids_df2, almi)</span>
<span id="cb2-463"><a href="#cb2-463"></a><span class="fu">c_sq_one</span>(branchids_df2, grip)</span>
<span id="cb2-464"><a href="#cb2-464"></a><span class="fu">c_sq_one</span>(branchids_df2, gait)</span>
<span id="cb2-465"><a href="#cb2-465"></a></span>
<span id="cb2-466"><a href="#cb2-466"></a><span class="do">### Includes all the lineages in the chi-sq plots</span></span>
<span id="cb2-467"><a href="#cb2-467"></a>c_sq_all <span class="ot">&lt;-</span> <span class="cf">function</span>(variable){</span>
<span id="cb2-468"><a href="#cb2-468"></a><span class="co"># All lineages</span></span>
<span id="cb2-469"><a href="#cb2-469"></a>cs_test <span class="ot">&lt;-</span> </span>
<span id="cb2-470"><a href="#cb2-470"></a>  branchids_df2 <span class="sc">%&gt;%</span> </span>
<span id="cb2-471"><a href="#cb2-471"></a>  <span class="fu">group_by</span>(lineage, {{variable}}) <span class="sc">%&gt;%</span> </span>
<span id="cb2-472"><a href="#cb2-472"></a>  <span class="fu">tally</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb2-473"><a href="#cb2-473"></a>  <span class="fu">drop_na</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb2-474"><a href="#cb2-474"></a>  <span class="fu">mutate</span>({{variable}} <span class="sc">:</span><span class="er">=</span> <span class="fu">paste0</span>(<span class="st">"quartile_"</span>, {{variable}})) <span class="sc">%&gt;%</span> </span>
<span id="cb2-475"><a href="#cb2-475"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> {{variable}}, <span class="at">values_from =</span> n) <span class="sc">%&gt;%</span> </span>
<span id="cb2-476"><a href="#cb2-476"></a>  <span class="fu">replace_na</span>(<span class="fu">list</span>(<span class="at">quartile_1=</span><span class="dv">0</span>,</span>
<span id="cb2-477"><a href="#cb2-477"></a>                  <span class="at">quartile_2=</span><span class="dv">0</span>,</span>
<span id="cb2-478"><a href="#cb2-478"></a>                  <span class="at">quartile_3=</span><span class="dv">0</span>,</span>
<span id="cb2-479"><a href="#cb2-479"></a>                  <span class="at">quartile_4=</span><span class="dv">0</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb2-480"><a href="#cb2-480"></a>  <span class="co"># filter(lineage %in% c("1,2,3,4,5",</span></span>
<span id="cb2-481"><a href="#cb2-481"></a>  <span class="co">#                             "1", "2", "3", "4", "5")) %&gt;% </span></span>
<span id="cb2-482"><a href="#cb2-482"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb2-483"><a href="#cb2-483"></a></span>
<span id="cb2-484"><a href="#cb2-484"></a>x <span class="ot">&lt;-</span> <span class="fu">as.table</span>(<span class="fu">cbind</span>(cs_test<span class="sc">$</span>quartile_1, </span>
<span id="cb2-485"><a href="#cb2-485"></a>                    cs_test<span class="sc">$</span>quartile_2,</span>
<span id="cb2-486"><a href="#cb2-486"></a>                    cs_test<span class="sc">$</span>quartile_3,</span>
<span id="cb2-487"><a href="#cb2-487"></a>                    cs_test<span class="sc">$</span>quartile_4))</span>
<span id="cb2-488"><a href="#cb2-488"></a><span class="fu">dimnames</span>(x) <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">lineage =</span> <span class="fu">c</span>(<span class="fu">as.character</span>(cs_test<span class="sc">$</span>lineage)),</span>
<span id="cb2-489"><a href="#cb2-489"></a>                    <span class="at">quartile =</span> <span class="fu">c</span>(<span class="st">"quartile_1"</span>, </span>
<span id="cb2-490"><a href="#cb2-490"></a>                                 <span class="st">"quartile_2"</span>, </span>
<span id="cb2-491"><a href="#cb2-491"></a>                                 <span class="st">"quartile_3"</span>, </span>
<span id="cb2-492"><a href="#cb2-492"></a>                                 <span class="st">"quartile_4"</span>))</span>
<span id="cb2-493"><a href="#cb2-493"></a>nabs <span class="ot">&lt;-</span> <span class="fu">chisq.test</span>(x)</span>
<span id="cb2-494"><a href="#cb2-494"></a>corrplot<span class="sc">::</span><span class="fu">corrplot</span>(nabs<span class="sc">$</span>residuals, <span class="at">is.cor =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2-495"><a href="#cb2-495"></a><span class="co"># contrib &lt;- 100*nabs$residuals^2/nabs$statistic</span></span>
<span id="cb2-496"><a href="#cb2-496"></a><span class="co"># round(contrib, 3)</span></span>
<span id="cb2-497"><a href="#cb2-497"></a><span class="co"># corrplot(contrib, is.cor = FALSE)</span></span>
<span id="cb2-498"><a href="#cb2-498"></a>}</span>
<span id="cb2-499"><a href="#cb2-499"></a></span>
<span id="cb2-500"><a href="#cb2-500"></a><span class="fu">c_sq_all</span>(almi)</span>
<span id="cb2-501"><a href="#cb2-501"></a><span class="fu">c_sq_all</span>(grip)</span>
<span id="cb2-502"><a href="#cb2-502"></a><span class="fu">c_sq_all</span>(gait)</span>
<span id="cb2-503"><a href="#cb2-503"></a></span>
<span id="cb2-504"><a href="#cb2-504"></a><span class="co"># Clustering --------------------------------------------------------------</span></span>
<span id="cb2-505"><a href="#cb2-505"></a></span>
<span id="cb2-506"><a href="#cb2-506"></a><span class="co"># library(clusterExperiment)</span></span>
<span id="cb2-507"><a href="#cb2-507"></a><span class="co"># nPointsClus &lt;- 20</span></span>
<span id="cb2-508"><a href="#cb2-508"></a><span class="co"># clusPat &lt;- clusterExpressionPatterns(models = gam, </span></span>
<span id="cb2-509"><a href="#cb2-509"></a><span class="co">#                                      nPoints = nPointsClus,</span></span>
<span id="cb2-510"><a href="#cb2-510"></a><span class="co">#                                      genes = var_genes)</span></span>
<span id="cb2-511"><a href="#cb2-511"></a><span class="co"># </span></span>
<span id="cb2-512"><a href="#cb2-512"></a><span class="co"># # saveRDS(clusPat, "data/clusPat.rds")</span></span>
<span id="cb2-513"><a href="#cb2-513"></a><span class="co"># clusPat &lt;- readRDS("data/clusPat.rds")</span></span>
<span id="cb2-514"><a href="#cb2-514"></a><span class="co"># clusterLabels &lt;- primaryCluster(clusPat$rsec)</span></span>
<span id="cb2-515"><a href="#cb2-515"></a><span class="co"># primaryClusterNamed(clusPat$rsec)</span></span>
<span id="cb2-516"><a href="#cb2-516"></a><span class="co"># </span></span>
<span id="cb2-517"><a href="#cb2-517"></a><span class="co"># tableClusters(clusPat$rsec)</span></span>
<span id="cb2-518"><a href="#cb2-518"></a><span class="co"># head(clusterMatrix(clusPat$rsec)[,1:4])</span></span>
<span id="cb2-519"><a href="#cb2-519"></a><span class="co"># </span></span>
<span id="cb2-520"><a href="#cb2-520"></a><span class="co"># nClusterings(clusPat$rsec)</span></span>
<span id="cb2-521"><a href="#cb2-521"></a><span class="co"># nSamples(clusPat$rsec)</span></span>
<span id="cb2-522"><a href="#cb2-522"></a><span class="co"># nFeatures(clusPat$rsec)</span></span>
<span id="cb2-523"><a href="#cb2-523"></a><span class="co"># </span></span>
<span id="cb2-524"><a href="#cb2-524"></a><span class="co"># plotReducedDims(clusPat$rsec,whichDims=c(1:10))</span></span>
<span id="cb2-525"><a href="#cb2-525"></a><span class="co"># plotReducedDims(clusPat$rsec)</span></span>
<span id="cb2-526"><a href="#cb2-526"></a><span class="co"># </span></span>
<span id="cb2-527"><a href="#cb2-527"></a><span class="co"># cUniq &lt;- unique(clusterLabels)</span></span>
<span id="cb2-528"><a href="#cb2-528"></a><span class="co"># cUniq &lt;- cUniq[!cUniq == -1] # remove unclustered genes</span></span>
<span id="cb2-529"><a href="#cb2-529"></a><span class="co"># unique(cUniq)</span></span>
<span id="cb2-530"><a href="#cb2-530"></a><span class="co"># </span></span>
<span id="cb2-531"><a href="#cb2-531"></a><span class="co"># for (xx in cUniq[1:10]) {</span></span>
<span id="cb2-532"><a href="#cb2-532"></a><span class="co">#   cId &lt;- which(clusterLabels == xx)</span></span>
<span id="cb2-533"><a href="#cb2-533"></a><span class="co">#   p &lt;- ggplot(data = data.frame(x = 1:nPointsClus,</span></span>
<span id="cb2-534"><a href="#cb2-534"></a><span class="co">#                                 y = rep(range(clusPat$yhatScaled[cId, ]),</span></span>
<span id="cb2-535"><a href="#cb2-535"></a><span class="co">#                                         nPointsClus / 2)),</span></span>
<span id="cb2-536"><a href="#cb2-536"></a><span class="co">#               aes(x = x, y = y)) +</span></span>
<span id="cb2-537"><a href="#cb2-537"></a><span class="co">#     geom_point(alpha = 0) +</span></span>
<span id="cb2-538"><a href="#cb2-538"></a><span class="co">#     labs(title = paste0("Cluster ", xx),  x = "Pseudotime", y = "Normalized expression") +</span></span>
<span id="cb2-539"><a href="#cb2-539"></a><span class="co">#     theme_classic() +</span></span>
<span id="cb2-540"><a href="#cb2-540"></a><span class="co">#     theme(plot.title = element_text(hjust = 0.5))</span></span>
<span id="cb2-541"><a href="#cb2-541"></a><span class="co">#   for (ii in 1:length(cId)) {</span></span>
<span id="cb2-542"><a href="#cb2-542"></a><span class="co">#     geneId &lt;- rownames(clusPat$yhatScaled)[cId[ii]]</span></span>
<span id="cb2-543"><a href="#cb2-543"></a><span class="co">#     p &lt;- p +</span></span>
<span id="cb2-544"><a href="#cb2-544"></a><span class="co">#       geom_line(data = data.frame(x = rep(1:nPointsClus, 5),</span></span>
<span id="cb2-545"><a href="#cb2-545"></a><span class="co">#                                   y = clusPat$yhatScaled[geneId, ],</span></span>
<span id="cb2-546"><a href="#cb2-546"></a><span class="co">#                                   lineage = rep(0:4, each = nPointsClus)),</span></span>
<span id="cb2-547"><a href="#cb2-547"></a><span class="co">#                 aes(col = as.character(lineage), group = lineage), lwd = 1.5)</span></span>
<span id="cb2-548"><a href="#cb2-548"></a><span class="co">#   }</span></span>
<span id="cb2-549"><a href="#cb2-549"></a><span class="co">#   p &lt;- p + guides(color = FALSE) +</span></span>
<span id="cb2-550"><a href="#cb2-550"></a><span class="co">#     scale_color_manual(values = c("orange", "darkseagreen3", "steelblue4", "indianred3", "mediumpurple3"),</span></span>
<span id="cb2-551"><a href="#cb2-551"></a><span class="co">#                        breaks = c("0", "1", "2", "3", "4"))  </span></span>
<span id="cb2-552"><a href="#cb2-552"></a><span class="co">#   print(p)</span></span>
<span id="cb2-553"><a href="#cb2-553"></a><span class="co"># }</span></span>
<span id="cb2-554"><a href="#cb2-554"></a></span>
<span id="cb2-555"><a href="#cb2-555"></a></span>
<span id="cb2-556"><a href="#cb2-556"></a><span class="co"># Heatmaps ----------------------------------------------------------------</span></span>
<span id="cb2-557"><a href="#cb2-557"></a></span>
<span id="cb2-558"><a href="#cb2-558"></a><span class="do">## This below line can be used to search whether genes are available in the </span></span>
<span id="cb2-559"><a href="#cb2-559"></a><span class="do">## gam object</span></span>
<span id="cb2-560"><a href="#cb2-560"></a></span>
<span id="cb2-561"><a href="#cb2-561"></a><span class="fu">rownames</span>(gam)[<span class="fu">grep</span>(<span class="st">"^col"</span>, <span class="fu">rownames</span>(gam), <span class="at">ignore.case=</span>T)]</span>
<span id="cb2-562"><a href="#cb2-562"></a></span>
<span id="cb2-563"><a href="#cb2-563"></a><span class="do">### based on mean smoother</span></span>
<span id="cb2-564"><a href="#cb2-564"></a><span class="co"># yhatSmooth &lt;- predictSmooth(gam, gene = var_genes[1:50], nPoints = 50, tidy = FALSE)</span></span>
<span id="cb2-565"><a href="#cb2-565"></a><span class="co"># yhatSmooth &lt;- yhatSmooth[order(apply(yhatSmooth,1,which.max)), ]</span></span>
<span id="cb2-566"><a href="#cb2-566"></a></span>
<span id="cb2-567"><a href="#cb2-567"></a><span class="do">## Put genes of interest for heatmaps in this object</span></span>
<span id="cb2-568"><a href="#cb2-568"></a>yhatSmooth <span class="ot">&lt;-</span> <span class="fu">predictSmooth</span>(gam, <span class="at">gene =</span> <span class="fu">c</span>(<span class="st">"H19"</span>,</span>
<span id="cb2-569"><a href="#cb2-569"></a>                                          <span class="st">"MKI67"</span>,</span>
<span id="cb2-570"><a href="#cb2-570"></a>                                          <span class="st">"TOP2A"</span>,</span>
<span id="cb2-571"><a href="#cb2-571"></a>                                          <span class="st">"PCNA"</span>,</span>
<span id="cb2-572"><a href="#cb2-572"></a>                                          <span class="st">"CCNE2"</span>,</span>
<span id="cb2-573"><a href="#cb2-573"></a>                                          <span class="st">"CCND1"</span>,</span>
<span id="cb2-574"><a href="#cb2-574"></a>                                          <span class="st">"CCND2"</span>,</span>
<span id="cb2-575"><a href="#cb2-575"></a>                                          <span class="st">"CENPK"</span>,</span>
<span id="cb2-576"><a href="#cb2-576"></a>                                          <span class="st">"CDC6"</span>,</span>
<span id="cb2-577"><a href="#cb2-577"></a>                                          <span class="st">"TNNT2"</span>,</span>
<span id="cb2-578"><a href="#cb2-578"></a>                                          <span class="st">"MYH3"</span>,</span>
<span id="cb2-579"><a href="#cb2-579"></a>                                          <span class="st">"MYOG"</span>,</span>
<span id="cb2-580"><a href="#cb2-580"></a>                                          <span class="st">"SQSTM1"</span>,</span>
<span id="cb2-581"><a href="#cb2-581"></a>                                          <span class="st">"GSTP1"</span>,</span>
<span id="cb2-582"><a href="#cb2-582"></a>                                          <span class="st">"NQO1"</span>,</span>
<span id="cb2-583"><a href="#cb2-583"></a>                                          <span class="st">"POSTN"</span>,</span>
<span id="cb2-584"><a href="#cb2-584"></a>                                          <span class="st">"FBN1"</span>,</span>
<span id="cb2-585"><a href="#cb2-585"></a>                                          <span class="st">"TTN"</span>,</span>
<span id="cb2-586"><a href="#cb2-586"></a>                                          <span class="st">"HES1"</span>,</span>
<span id="cb2-587"><a href="#cb2-587"></a>                                          <span class="st">"BRCA1"</span>,</span>
<span id="cb2-588"><a href="#cb2-588"></a>                                          <span class="st">"BRCA2"</span>,</span>
<span id="cb2-589"><a href="#cb2-589"></a>                                          <span class="st">"COX5B"</span>,</span>
<span id="cb2-590"><a href="#cb2-590"></a>                                          <span class="st">"ACTA2"</span>,</span>
<span id="cb2-591"><a href="#cb2-591"></a>                                          <span class="st">"ACTA1"</span>,</span>
<span id="cb2-592"><a href="#cb2-592"></a>                                          <span class="st">"RAD21"</span>,</span>
<span id="cb2-593"><a href="#cb2-593"></a>                                          <span class="st">"RAD51AP1"</span>,</span>
<span id="cb2-594"><a href="#cb2-594"></a>                                          <span class="st">"NES"</span>,</span>
<span id="cb2-595"><a href="#cb2-595"></a>                                          <span class="st">"MEF2C"</span>,</span>
<span id="cb2-596"><a href="#cb2-596"></a>                                          <span class="st">"DES"</span>,</span>
<span id="cb2-597"><a href="#cb2-597"></a>                                          <span class="st">"MYF5"</span>,</span>
<span id="cb2-598"><a href="#cb2-598"></a>                                          <span class="st">"CDKN1A"</span>,</span>
<span id="cb2-599"><a href="#cb2-599"></a>                                          <span class="st">"CDKN2A"</span>,</span>
<span id="cb2-600"><a href="#cb2-600"></a>                                          <span class="st">"FTH1"</span>,</span>
<span id="cb2-601"><a href="#cb2-601"></a>                                          <span class="st">"FTL"</span>,</span>
<span id="cb2-602"><a href="#cb2-602"></a>                                          <span class="st">"MYOD1"</span>), <span class="at">nPoints =</span> <span class="dv">50</span>, <span class="at">tidy =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-603"><a href="#cb2-603"></a></span>
<span id="cb2-604"><a href="#cb2-604"></a><span class="co"># Most highly expressed per given timepoint (orders it nicely)</span></span>
<span id="cb2-605"><a href="#cb2-605"></a>nums <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">5</span>)</span>
<span id="cb2-606"><a href="#cb2-606"></a>lins <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb2-607"><a href="#cb2-607"></a><span class="cf">for</span>(i <span class="cf">in</span> nums){</span>
<span id="cb2-608"><a href="#cb2-608"></a>  iff <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"lin"</span>, i)</span>
<span id="cb2-609"><a href="#cb2-609"></a>  <span class="co"># lins[[paste0(iff)]] &lt;- </span></span>
<span id="cb2-610"><a href="#cb2-610"></a>  <span class="co"># arts(paste0("lineage", i), yhatSmooth)</span></span>
<span id="cb2-611"><a href="#cb2-611"></a>  bib <span class="ot">&lt;-</span> <span class="fu">arts</span>(<span class="fu">paste0</span>(<span class="st">"lineage"</span>, i), yhatSmooth)</span>
<span id="cb2-612"><a href="#cb2-612"></a>  lins[[<span class="fu">paste0</span>(iff)]] <span class="ot">&lt;-</span> bib[<span class="fu">order</span>(<span class="fu">apply</span>(bib,<span class="dv">1</span>,which.max)), ]</span>
<span id="cb2-613"><a href="#cb2-613"></a>  <span class="co">#lins[[paste0(iff)]] &lt;- bib</span></span>
<span id="cb2-614"><a href="#cb2-614"></a>  <span class="co"># print(i)</span></span>
<span id="cb2-615"><a href="#cb2-615"></a>}</span>
<span id="cb2-616"><a href="#cb2-616"></a></span>
<span id="cb2-617"><a href="#cb2-617"></a><span class="do">## The below will plot the heatmaps individually </span></span>
<span id="cb2-618"><a href="#cb2-618"></a>heatm <span class="ot">&lt;-</span> </span>
<span id="cb2-619"><a href="#cb2-619"></a>pheatmap<span class="sc">::</span><span class="fu">pheatmap</span>(<span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(lins<span class="sc">$</span>lin1[, <span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>]))),</span>
<span id="cb2-620"><a href="#cb2-620"></a>                   <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>,</span>
<span id="cb2-621"><a href="#cb2-621"></a>                   <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>,</span>
<span id="cb2-622"><a href="#cb2-622"></a>                   <span class="at">show_rownames =</span> <span class="cn">TRUE</span>, </span>
<span id="cb2-623"><a href="#cb2-623"></a>                   <span class="at">show_colnames =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-624"><a href="#cb2-624"></a></span>
<span id="cb2-625"><a href="#cb2-625"></a>pheatmap<span class="sc">::</span><span class="fu">pheatmap</span>(<span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(lins<span class="sc">$</span>lin2[, <span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>]))),</span>
<span id="cb2-626"><a href="#cb2-626"></a>                   <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>,</span>
<span id="cb2-627"><a href="#cb2-627"></a>                   <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>,</span>
<span id="cb2-628"><a href="#cb2-628"></a>                   <span class="at">show_rownames =</span> <span class="cn">TRUE</span>, </span>
<span id="cb2-629"><a href="#cb2-629"></a>                   <span class="at">show_colnames =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-630"><a href="#cb2-630"></a></span>
<span id="cb2-631"><a href="#cb2-631"></a>pheatmap<span class="sc">::</span><span class="fu">pheatmap</span>(<span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(lins<span class="sc">$</span>lin3[, <span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>]))),</span>
<span id="cb2-632"><a href="#cb2-632"></a>                   <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>,</span>
<span id="cb2-633"><a href="#cb2-633"></a>                   <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>,</span>
<span id="cb2-634"><a href="#cb2-634"></a>                   <span class="at">show_rownames =</span> <span class="cn">TRUE</span>, </span>
<span id="cb2-635"><a href="#cb2-635"></a>                   <span class="at">show_colnames =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-636"><a href="#cb2-636"></a></span>
<span id="cb2-637"><a href="#cb2-637"></a>pheatmap<span class="sc">::</span><span class="fu">pheatmap</span>(<span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(lins<span class="sc">$</span>lin4[, <span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>]))),</span>
<span id="cb2-638"><a href="#cb2-638"></a>                   <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>,</span>
<span id="cb2-639"><a href="#cb2-639"></a>                   <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>,</span>
<span id="cb2-640"><a href="#cb2-640"></a>                   <span class="at">show_rownames =</span> <span class="cn">TRUE</span>, </span>
<span id="cb2-641"><a href="#cb2-641"></a>                   <span class="at">show_colnames =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-642"><a href="#cb2-642"></a></span>
<span id="cb2-643"><a href="#cb2-643"></a>heatSmooth <span class="ot">&lt;-</span> pheatmap<span class="sc">::</span><span class="fu">pheatmap</span>(<span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(lins<span class="sc">$</span>lin5[, <span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>]))),</span>
<span id="cb2-644"><a href="#cb2-644"></a>                   <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>,</span>
<span id="cb2-645"><a href="#cb2-645"></a>                   <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>,</span>
<span id="cb2-646"><a href="#cb2-646"></a>                   <span class="at">show_rownames =</span> <span class="cn">TRUE</span>, </span>
<span id="cb2-647"><a href="#cb2-647"></a>                   <span class="at">show_colnames =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-648"><a href="#cb2-648"></a></span>
<span id="cb2-649"><a href="#cb2-649"></a><span class="co"># earlyDERes &lt;- earlyDETest(gam, knots = c(2, 3))</span></span>
<span id="cb2-650"><a href="#cb2-650"></a><span class="co"># oEarly &lt;- order(earlyDERes$waldStat, decreasing = TRUE)</span></span>
<span id="cb2-651"><a href="#cb2-651"></a><span class="co"># head(rownames(earlyDERes)[oEarly])</span></span>
<span id="cb2-652"><a href="#cb2-652"></a><span class="co"># plotGeneCount(pto, counts, gene = "NES")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>