<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Matt work summary - Whole muscle single nuclear</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title">Whole muscle single nuclear</h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Matt work summary</a> 
    </div>
      </div>
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">Home</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./whole_muscle_scrnaseq.html" class="sidebar-item-text sidebar-link active">Whole muscle single nuclear</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./myob_single_nuc.html" class="sidebar-item-text sidebar-link">Myoblast single nuclear</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./atac_seq.html" class="sidebar-item-text sidebar-link">ATACseq</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./conda_envs.html" class="sidebar-item-text sidebar-link">Conda environments</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./extra_tips.html" class="sidebar-item-text sidebar-link">Extra tips</a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">Single cell</a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./single_cell.html" class="sidebar-item-text sidebar-link">Overview</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pseudotime.html" class="sidebar-item-text sidebar-link">Pseudotime</a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#summary" id="toc-summary" class="nav-link active" data-scroll-target="#summary">Summary</a></li>
  <li><a href="#setup" id="toc-setup" class="nav-link" data-scroll-target="#setup">Setup</a></li>
  <li><a href="#bash-bit" id="toc-bash-bit" class="nav-link" data-scroll-target="#bash-bit">‘Bash bit’</a></li>
  <li><a href="#r-bit" id="toc-r-bit" class="nav-link" data-scroll-target="#r-bit">‘R bit’</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title d-none d-lg-block">Whole muscle single nuclear</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="summary" class="level1">
<h1>Summary</h1>
</section>
<section id="setup" class="level1">
<h1>Setup</h1>
<ul>
<li><p>The project folder is <code>/scratch/moh1u21/23-01-30-whole_muscle_snrnaseq</code></p></li>
<li><p>The raw data is at <code>/scratch/moh1u21/23-01-30-whole_muscle_snrnaseq/data/230126_NB501007_0335_AH3K2FAFX5</code></p></li>
<li><p>All work for the ‘bash bit’ was potentially done in the qc_env conda environment, or the qc_and_alignment.sif apptainer and the ‘R bit’ was done in the sc_qc conda environment.</p></li>
<li><p>If the bash bit doesn’t work - remove any commented lines, can get a bit funny with those sometimes.</p></li>
<li><p>This pipeline deviates from the code used for the single cell alignment, as the previous single cell alignment pipeline based on the pipeline proposed by the McCarrol lab did not account for intronic reads and was inflexible with regards to multi-mapping reads.</p></li>
</ul>
</section>
<section id="bash-bit" class="level1">
<h1>‘Bash bit’</h1>
<ul>
<li>The file for the bash bit is located at <code>/scratch/moh1u21/23-01-30-whole_muscle_snrnaseq/src/solo_alignment_trimR2only.sh</code>.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1"></a> </span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="bu">cd</span> /scratch/moh1u21/23-01-05-single_nuclear</span>
<span id="cb1-3"><a href="#cb1-3"></a></span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="bu">echo</span> <span class="va">$PATH</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="st">'s/:/\n/g'</span> <span class="kw">|</span> <span class="fu">sort</span> <span class="kw">|</span> <span class="fu">uniq</span> <span class="at">-c</span></span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="va">today</span><span class="op">=</span><span class="va">$(</span><span class="fu">date</span> +<span class="st">"%Y-%m-%d"</span><span class="va">)</span></span>
<span id="cb1-6"><a href="#cb1-6"></a></span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="co"># Set up project folders and environmental variables</span></span>
<span id="cb1-8"><a href="#cb1-8"></a></span>
<span id="cb1-9"><a href="#cb1-9"></a><span class="va">project_folder</span><span class="op">=</span>/scratch/moh1u21/23-01-05-single_nuclear</span>
<span id="cb1-10"><a href="#cb1-10"></a><span class="co"># genomedir=/Users/fluentin44/Documents/22-05-15-index_genome/genome_index_hg19 </span></span>
<span id="cb1-11"><a href="#cb1-11"></a><span class="va">genomedir</span><span class="op">=</span>/scratch/moh1u21/22-05-15-index_genome/hg19_ensembl</span>
<span id="cb1-12"><a href="#cb1-12"></a></span>
<span id="cb1-13"><a href="#cb1-13"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="st">"</span><span class="va">$project_folder</span><span class="st">"</span>/results/logs</span>
<span id="cb1-14"><a href="#cb1-14"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="st">"</span><span class="va">$project_folder</span><span class="st">"</span>/results/raw/solo_output</span>
<span id="cb1-15"><a href="#cb1-15"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="st">"</span><span class="va">$project_folder</span><span class="st">"</span>/results/raw/fastqs</span>
<span id="cb1-16"><a href="#cb1-16"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="st">"</span><span class="va">$project_folder</span><span class="st">"</span>/results/raw/fastqc</span>
<span id="cb1-17"><a href="#cb1-17"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="st">"</span><span class="va">$project_folder</span><span class="st">"</span>/results/raw/multiqc</span>
<span id="cb1-18"><a href="#cb1-18"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="st">"</span><span class="va">$project_folder</span><span class="st">"</span>/results/raw/trimmed</span>
<span id="cb1-19"><a href="#cb1-19"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="st">"</span><span class="va">$project_folder</span><span class="st">"</span>/results/raw/final_fqs</span>
<span id="cb1-20"><a href="#cb1-20"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="st">"</span><span class="va">$project_folder</span><span class="st">"</span>/results/raw/fastp_reports</span>
<span id="cb1-21"><a href="#cb1-21"></a></span>
<span id="cb1-22"><a href="#cb1-22"></a><span class="va">fastp_logfile</span><span class="op">=</span>results/logs/fastp_log.txt</span>
<span id="cb1-23"><a href="#cb1-23"></a><span class="va">star_logfile</span><span class="op">=</span>results/logs/star_log.txt</span>
<span id="cb1-24"><a href="#cb1-24"></a></span>
<span id="cb1-25"><a href="#cb1-25"></a><span class="co"># tmpdir="$project_folder"/tmp_dir</span></span>
<span id="cb1-26"><a href="#cb1-26"></a><span class="va">solo_folder</span><span class="op">=</span><span class="st">"</span><span class="va">$project_folder</span><span class="st">"</span>/results/raw/solo_output</span>
<span id="cb1-27"><a href="#cb1-27"></a></span>
<span id="cb1-28"><a href="#cb1-28"></a><span class="co"># fastqc data/full_reads/*_2.fastq \</span></span>
<span id="cb1-29"><a href="#cb1-29"></a><span class="co"># -o results/solo_folder/fastqcs/pre-trim/fastqc_reports/</span></span>
<span id="cb1-30"><a href="#cb1-30"></a></span>
<span id="cb1-31"><a href="#cb1-31"></a><span class="co"># Do an initial QC of the reads</span></span>
<span id="cb1-32"><a href="#cb1-32"></a></span>
<span id="cb1-33"><a href="#cb1-33"></a><span class="ex">fastqc</span> /scratch/moh1u21/single_nuclear/bsg-ftp.well.ox.ac.uk/220427_A00711_0538_BH7VMTDMXY/fastqs/<span class="pp">*</span>.fastq.gz <span class="dt">\</span></span>
<span id="cb1-34"><a href="#cb1-34"></a>-o results/raw/fastqc/ <span class="dt">\</span></span>
<span id="cb1-35"><a href="#cb1-35"></a>-t 60 <span class="dt">\</span></span>
<span id="cb1-36"><a href="#cb1-36"></a>--contaminants /scratch/moh1u21/ATAC_seq_test/data/adapters.txt</span>
<span id="cb1-37"><a href="#cb1-37"></a></span>
<span id="cb1-38"><a href="#cb1-38"></a><span class="ex">multiqc</span> results/raw/fastqc/ <span class="dt">\</span></span>
<span id="cb1-39"><a href="#cb1-39"></a>-o results/raw/multiqc/ <span class="dt">\</span></span>
<span id="cb1-40"><a href="#cb1-40"></a>-n multiqc_pre_trim</span>
<span id="cb1-41"><a href="#cb1-41"></a></span>
<span id="cb1-42"><a href="#cb1-42"></a><span class="co"># Cycle though the files to do a trim of read 2 (read one is index etc, not aligned and would be detrimental. to lose any info)</span></span>
<span id="cb1-43"><a href="#cb1-43"></a><span class="va">id</span><span class="op">=</span><span class="va">$(</span><span class="fu">ls</span> /scratch/moh1u21/single_nuclear/bsg-ftp.well.ox.ac.uk/220427_A00711_0538_BH7VMTDMXY/fastqs<span class="va">)</span></span>
<span id="cb1-44"><a href="#cb1-44"></a></span>
<span id="cb1-45"><a href="#cb1-45"></a><span class="va">arr</span><span class="op">=</span><span class="va">($id)</span></span>
<span id="cb1-46"><a href="#cb1-46"></a><span class="co"># Get central bit of filename</span></span>
<span id="cb1-47"><a href="#cb1-47"></a><span class="va">arr_names</span><span class="op">=</span><span class="va">()</span></span>
<span id="cb1-48"><a href="#cb1-48"></a><span class="cf">for</span> i <span class="kw">in</span> <span class="st">"</span><span class="va">${arr</span><span class="op">[@]</span><span class="va">}</span><span class="st">"</span></span>
<span id="cb1-49"><a href="#cb1-49"></a><span class="cf">do</span></span>
<span id="cb1-50"><a href="#cb1-50"></a>    <span class="va">arr_names</span><span class="op">+=</span><span class="va">(</span><span class="st">"</span><span class="va">$(</span><span class="bu">echo</span> <span class="va">$i</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-d</span><span class="st">'_'</span> <span class="at">-f</span> 1-3<span class="va">)</span><span class="st">"</span><span class="va">)</span></span>
<span id="cb1-51"><a href="#cb1-51"></a><span class="cf">done</span></span>
<span id="cb1-52"><a href="#cb1-52"></a><span class="bu">declare</span> <span class="at">-p</span> <span class="va">arr_names</span></span>
<span id="cb1-53"><a href="#cb1-53"></a></span>
<span id="cb1-54"><a href="#cb1-54"></a><span class="co"># Get only the unique filenames</span></span>
<span id="cb1-55"><a href="#cb1-55"></a><span class="va">newArr</span><span class="op">=</span><span class="va">()</span><span class="kw">;</span> </span>
<span id="cb1-56"><a href="#cb1-56"></a><span class="cf">while</span> <span class="va">IFS</span><span class="op">=</span> <span class="bu">read</span> <span class="at">-r</span> <span class="at">-d</span> <span class="st">''</span> <span class="va">x</span><span class="kw">;</span> </span>
<span id="cb1-57"><a href="#cb1-57"></a><span class="cf">do</span> <span class="va">newArr</span><span class="op">+=</span><span class="va">(</span><span class="st">"</span><span class="va">$x</span><span class="st">"</span><span class="va">)</span><span class="kw">;</span> </span>
<span id="cb1-58"><a href="#cb1-58"></a><span class="cf">done</span> <span class="op">&lt;</span> <span class="op">&lt;(</span><span class="bu">printf</span> <span class="st">"%s\0"</span> <span class="st">"</span><span class="va">${arr_names</span><span class="op">[@]</span><span class="va">}</span><span class="st">"</span> <span class="kw">|</span> <span class="fu">sort</span> <span class="at">-uz</span><span class="op">)</span></span>
<span id="cb1-59"><a href="#cb1-59"></a><span class="bu">declare</span> <span class="at">-p</span> <span class="va">newArr</span></span>
<span id="cb1-60"><a href="#cb1-60"></a></span>
<span id="cb1-61"><a href="#cb1-61"></a><span class="cf">for</span> i <span class="kw">in</span> <span class="va">${newArr</span><span class="op">[@]</span><span class="va">}</span></span>
<span id="cb1-62"><a href="#cb1-62"></a><span class="cf">do</span></span>
<span id="cb1-63"><a href="#cb1-63"></a></span>
<span id="cb1-64"><a href="#cb1-64"></a><span class="ex">fastp</span> <span class="dt">\</span></span>
<span id="cb1-65"><a href="#cb1-65"></a>-w 16 <span class="dt">\</span></span>
<span id="cb1-66"><a href="#cb1-66"></a>-p <span class="dt">\</span></span>
<span id="cb1-67"><a href="#cb1-67"></a>-Q <span class="dt">\</span></span>
<span id="cb1-68"><a href="#cb1-68"></a>-l 50 <span class="dt">\</span></span>
<span id="cb1-69"><a href="#cb1-69"></a>-i /scratch/moh1u21/single_nuclear/bsg-ftp.well.ox.ac.uk/220427_A00711_0538_BH7VMTDMXY/fastqs/<span class="va">${i}</span>_2.fastq.gz <span class="dt">\</span></span>
<span id="cb1-70"><a href="#cb1-70"></a>-o results/raw/trimmed/trimmed_<span class="va">${i}</span>_2.fastq.gz <span class="dt">\</span></span>
<span id="cb1-71"><a href="#cb1-71"></a>--adapter_sequence=AGCAGTGGTATCAACGCAGAGTGAATGGG <span class="dt">\</span></span>
<span id="cb1-72"><a href="#cb1-72"></a>--adapter_fasta /scratch/moh1u21/23-01-30-whole_muscle_snrnaseq/data/adapters.fasta <span class="dt">\</span></span>
<span id="cb1-73"><a href="#cb1-73"></a>--trim_poly_g <span class="dt">\</span></span>
<span id="cb1-74"><a href="#cb1-74"></a>--trim_poly_x <span class="dt">\</span></span>
<span id="cb1-75"><a href="#cb1-75"></a>--cut_tail <span class="dt">\</span></span>
<span id="cb1-76"><a href="#cb1-76"></a>--cut_mean_quality 30 <span class="dt">\</span></span>
<span id="cb1-77"><a href="#cb1-77"></a>--html results/raw/fastp_reports/trimmed_<span class="va">${i}</span>_fastp_report.html <span class="dt">\</span></span>
<span id="cb1-78"><a href="#cb1-78"></a>--json results/raw/fastp_reports/trimmed_<span class="va">${i}</span>_fastp_report.json <span class="op">&gt;&gt;</span> <span class="va">$fastp_logfile</span> <span class="dv">2</span><span class="op">&gt;&amp;</span><span class="dv">1</span></span>
<span id="cb1-79"><a href="#cb1-79"></a></span>
<span id="cb1-80"><a href="#cb1-80"></a><span class="co"># fastP will remove and reads in the R2 file with too low a quality, which will leave the partner read in R1 </span></span>
<span id="cb1-81"><a href="#cb1-81"></a><span class="ex">seqkit</span> pair <span class="dt">\</span></span>
<span id="cb1-82"><a href="#cb1-82"></a>--read1 /scratch/moh1u21/single_nuclear/bsg-ftp.well.ox.ac.uk/220427_A00711_0538_BH7VMTDMXY/fastqs/<span class="va">${i}</span>_1.fastq.gz <span class="dt">\</span></span>
<span id="cb1-83"><a href="#cb1-83"></a>--read2 <span class="st">"</span><span class="va">$project_folder</span><span class="st">"</span>/results/raw/trimmed/trimmed_<span class="va">${i}</span>_2.fastq.gz <span class="dt">\</span></span>
<span id="cb1-84"><a href="#cb1-84"></a>--out-dir <span class="st">"</span><span class="va">$project_folder</span><span class="st">"</span>/results/raw/final_fqs</span>
<span id="cb1-85"><a href="#cb1-85"></a></span>
<span id="cb1-86"><a href="#cb1-86"></a><span class="cf">done</span> </span>
<span id="cb1-87"><a href="#cb1-87"></a></span>
<span id="cb1-88"><a href="#cb1-88"></a><span class="ex">fastqc</span> results/raw/final_fqs/<span class="pp">*</span>.fastq.gz <span class="dt">\</span></span>
<span id="cb1-89"><a href="#cb1-89"></a>-o results/raw/fastqc/ <span class="dt">\</span></span>
<span id="cb1-90"><a href="#cb1-90"></a>-t 60 <span class="dt">\</span></span>
<span id="cb1-91"><a href="#cb1-91"></a>--contaminants /scratch/moh1u21/ATAC_seq_test/data/adapters.txt</span>
<span id="cb1-92"><a href="#cb1-92"></a></span>
<span id="cb1-93"><a href="#cb1-93"></a><span class="ex">multiqc</span> results/raw/fastqc/ <span class="dt">\</span></span>
<span id="cb1-94"><a href="#cb1-94"></a>-o results/raw/multiqc/ <span class="dt">\</span></span>
<span id="cb1-95"><a href="#cb1-95"></a>-n multiqc_post_trim</span>
<span id="cb1-96"><a href="#cb1-96"></a></span>
<span id="cb1-97"><a href="#cb1-97"></a></span>
<span id="cb1-98"><a href="#cb1-98"></a><span class="co"># id=$(ls results/raw/trimmed)</span></span>
<span id="cb1-99"><a href="#cb1-99"></a></span>
<span id="cb1-100"><a href="#cb1-100"></a><span class="co"># arr=($id)</span></span>
<span id="cb1-101"><a href="#cb1-101"></a><span class="co"># # Get central bit of filename</span></span>
<span id="cb1-102"><a href="#cb1-102"></a><span class="co"># arr_names=()</span></span>
<span id="cb1-103"><a href="#cb1-103"></a><span class="co"># for i in "${arr[@]}"</span></span>
<span id="cb1-104"><a href="#cb1-104"></a><span class="co"># do</span></span>
<span id="cb1-105"><a href="#cb1-105"></a><span class="co">#   arr_names+=("$(echo $i | cut -d'_' -f 1-4)")</span></span>
<span id="cb1-106"><a href="#cb1-106"></a><span class="co"># done</span></span>
<span id="cb1-107"><a href="#cb1-107"></a><span class="co"># declare -p arr_names</span></span>
<span id="cb1-108"><a href="#cb1-108"></a></span>
<span id="cb1-109"><a href="#cb1-109"></a><span class="co"># # Get only the unique filenames</span></span>
<span id="cb1-110"><a href="#cb1-110"></a><span class="co"># newArr=(); </span></span>
<span id="cb1-111"><a href="#cb1-111"></a><span class="co"># while IFS= read -r -d '' x; </span></span>
<span id="cb1-112"><a href="#cb1-112"></a><span class="co"># do newArr+=("$x"); </span></span>
<span id="cb1-113"><a href="#cb1-113"></a><span class="co"># done &lt; &lt;(printf "%s\0" "${arr_names[@]}" | sort -uz)</span></span>
<span id="cb1-114"><a href="#cb1-114"></a><span class="co"># declare -p newArr</span></span>
<span id="cb1-115"><a href="#cb1-115"></a></span>
<span id="cb1-116"><a href="#cb1-116"></a><span class="co"># mkdir -p results/solo_folder/output/bams</span></span>
<span id="cb1-117"><a href="#cb1-117"></a><span class="co"># mkdir -p results/solo_folder/output/final_logs</span></span>
<span id="cb1-118"><a href="#cb1-118"></a><span class="co"># mkdir -p results/solo_folder/output/other_logs</span></span>
<span id="cb1-119"><a href="#cb1-119"></a></span>
<span id="cb1-120"><a href="#cb1-120"></a><span class="cf">for</span> i <span class="kw">in</span> <span class="va">${newArr</span><span class="op">[@]</span><span class="va">}</span></span>
<span id="cb1-121"><a href="#cb1-121"></a><span class="cf">do</span></span>
<span id="cb1-122"><a href="#cb1-122"></a></span>
<span id="cb1-123"><a href="#cb1-123"></a><span class="va">fastq_1</span><span class="op">=</span>results/raw/final_fqs/<span class="va">${i}</span>_1.fastq.gz</span>
<span id="cb1-124"><a href="#cb1-124"></a><span class="va">fastq_2</span><span class="op">=</span>results/raw/final_fqs/trimmed_<span class="va">${i}</span>_2.fastq.gz</span>
<span id="cb1-125"><a href="#cb1-125"></a><span class="bu">echo</span> <span class="va">${fastq_1}</span></span>
<span id="cb1-126"><a href="#cb1-126"></a><span class="bu">echo</span> <span class="va">${fastq_2}</span></span>
<span id="cb1-127"><a href="#cb1-127"></a></span>
<span id="cb1-128"><a href="#cb1-128"></a><span class="ex">STAR</span> <span class="dt">\</span></span>
<span id="cb1-129"><a href="#cb1-129"></a>--runThreadN 60 <span class="dt">\</span></span>
<span id="cb1-130"><a href="#cb1-130"></a>--genomeDir <span class="va">${genomedir}</span> <span class="dt">\</span></span>
<span id="cb1-131"><a href="#cb1-131"></a>--outFileNamePrefix <span class="va">${solo_folder}</span>/sample_<span class="va">${i}</span>_star. <span class="dt">\</span></span>
<span id="cb1-132"><a href="#cb1-132"></a>--readFilesIn <span class="va">${fastq_2}</span> <span class="va">${fastq_1}</span> <span class="dt">\</span></span>
<span id="cb1-133"><a href="#cb1-133"></a>--soloType CB_UMI_Simple <span class="dt">\</span></span>
<span id="cb1-134"><a href="#cb1-134"></a>--readFilesCommand zcat <span class="dt">\</span></span>
<span id="cb1-135"><a href="#cb1-135"></a>--soloCBwhitelist None <span class="dt">\</span></span>
<span id="cb1-136"><a href="#cb1-136"></a>--soloCBstart 1 <span class="dt">\</span></span>
<span id="cb1-137"><a href="#cb1-137"></a>--soloCBlen 12 <span class="dt">\</span></span>
<span id="cb1-138"><a href="#cb1-138"></a>--soloUMIstart 13 <span class="dt">\</span></span>
<span id="cb1-139"><a href="#cb1-139"></a>--soloUMIlen 8 <span class="dt">\</span></span>
<span id="cb1-140"><a href="#cb1-140"></a>--soloBarcodeReadLength 0 <span class="dt">\</span></span>
<span id="cb1-141"><a href="#cb1-141"></a>--soloStrand Unstranded <span class="dt">\</span></span>
<span id="cb1-142"><a href="#cb1-142"></a>--soloCellFilter EmptyDrops_CR 1000 0.99 10 45000 90000 1000 0.01 20000 0.01 10000 <span class="dt">\</span></span>
<span id="cb1-143"><a href="#cb1-143"></a>--outSAMattributes All CB UB sM sS sQ uT RG <span class="dt">\</span></span>
<span id="cb1-144"><a href="#cb1-144"></a>--outSAMattrRGline ID:<span class="va">${i}</span> SM:<span class="va">${i}</span> <span class="dt">\</span></span>
<span id="cb1-145"><a href="#cb1-145"></a>--outSAMtype BAM SortedByCoordinate <span class="dt">\</span></span>
<span id="cb1-146"><a href="#cb1-146"></a>--outMultimapperOrder Random <span class="dt">\</span></span>
<span id="cb1-147"><a href="#cb1-147"></a>--soloCellReadStats Standard <span class="dt">\</span></span>
<span id="cb1-148"><a href="#cb1-148"></a>--soloFeatures Gene GeneFull Velocyto <span class="dt">\</span></span>
<span id="cb1-149"><a href="#cb1-149"></a>--soloMultiMappers EM <span class="dt">\</span></span>
<span id="cb1-150"><a href="#cb1-150"></a>--outFilterMultimapNmax 10 <span class="op">&gt;&gt;</span> <span class="va">$star_logfile</span> <span class="dv">2</span><span class="op">&gt;&amp;</span><span class="dv">1</span></span>
<span id="cb1-151"><a href="#cb1-151"></a></span>
<span id="cb1-152"><a href="#cb1-152"></a><span class="co"># STAR \</span></span>
<span id="cb1-153"><a href="#cb1-153"></a><span class="co"># --runThreadN 60 \</span></span>
<span id="cb1-154"><a href="#cb1-154"></a><span class="co"># --readFilesCommand zcat \</span></span>
<span id="cb1-155"><a href="#cb1-155"></a><span class="co"># --genomeDir ${genomedir} \</span></span>
<span id="cb1-156"><a href="#cb1-156"></a><span class="co"># --outFileNamePrefix ${solo_folder}/sample_${i}_star. \</span></span>
<span id="cb1-157"><a href="#cb1-157"></a><span class="co"># --readFilesIn $fastq_2 $fastq_1 \</span></span>
<span id="cb1-158"><a href="#cb1-158"></a><span class="co"># --soloType CB_UMI_Simple \</span></span>
<span id="cb1-159"><a href="#cb1-159"></a><span class="co"># --soloCBstart 1 \</span></span>
<span id="cb1-160"><a href="#cb1-160"></a><span class="co"># --soloCBlen 12 \</span></span>
<span id="cb1-161"><a href="#cb1-161"></a><span class="co"># --soloUMIstart 13 \</span></span>
<span id="cb1-162"><a href="#cb1-162"></a><span class="co"># --soloUMIlen 8 \</span></span>
<span id="cb1-163"><a href="#cb1-163"></a><span class="co"># --soloCBwhitelist None \</span></span>
<span id="cb1-164"><a href="#cb1-164"></a><span class="co"># --clipAdapterType CellRanger4 \</span></span>
<span id="cb1-165"><a href="#cb1-165"></a><span class="co"># --soloBarcodeReadLength  0 \</span></span>
<span id="cb1-166"><a href="#cb1-166"></a><span class="co"># --soloStrand Unstranded \</span></span>
<span id="cb1-167"><a href="#cb1-167"></a><span class="co"># --outSAMattributes NH HI nM AS CR UR CB UB sS sQ RG sM \</span></span>
<span id="cb1-168"><a href="#cb1-168"></a><span class="co"># --outSAMattrRGline ID:${i} SM:${i} \</span></span>
<span id="cb1-169"><a href="#cb1-169"></a><span class="co"># --outSAMtype BAM SortedByCoordinate \</span></span>
<span id="cb1-170"><a href="#cb1-170"></a><span class="co"># --outMultimapperOrder Random \</span></span>
<span id="cb1-171"><a href="#cb1-171"></a><span class="co"># --soloCellFilter  EmptyDrops_CR \</span></span>
<span id="cb1-172"><a href="#cb1-172"></a><span class="co"># --soloCellReadStats Standard \</span></span>
<span id="cb1-173"><a href="#cb1-173"></a><span class="co"># --soloFeatures Gene GeneFull Velocyto \</span></span>
<span id="cb1-174"><a href="#cb1-174"></a><span class="co"># --soloMultiMappers EM \</span></span>
<span id="cb1-175"><a href="#cb1-175"></a><span class="co"># --outFilterMultimapNmax 10</span></span>
<span id="cb1-176"><a href="#cb1-176"></a></span>
<span id="cb1-177"><a href="#cb1-177"></a><span class="co"># mv ${solo_folder}/*.bam ${solo_folder}/output/bams</span></span>
<span id="cb1-178"><a href="#cb1-178"></a><span class="co"># mv ${solo_folder}/*.Log.final.out ${solo_folder}/output/final_logs</span></span>
<span id="cb1-179"><a href="#cb1-179"></a><span class="co"># mv ${solo_folder}/*.Log.out ${solo_folder}/*.Log.progress.out ${solo_folder}/*.SJ.out.tab ${solo_folder}/output/other_logs</span></span>
<span id="cb1-180"><a href="#cb1-180"></a></span>
<span id="cb1-181"><a href="#cb1-181"></a><span class="cf">done</span></span>
<span id="cb1-182"><a href="#cb1-182"></a></span>
<span id="cb1-183"><a href="#cb1-183"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="st">"</span><span class="va">$project_folder</span><span class="st">"</span>/results/raw/solo_output/bams</span>
<span id="cb1-184"><a href="#cb1-184"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="st">"</span><span class="va">$project_folder</span><span class="st">"</span>/results/raw/solo_output/final_logs</span>
<span id="cb1-185"><a href="#cb1-185"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="st">"</span><span class="va">$project_folder</span><span class="st">"</span>/results/raw/solo_output/progress_logs</span>
<span id="cb1-186"><a href="#cb1-186"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="st">"</span><span class="va">$project_folder</span><span class="st">"</span>/results/raw/solo_output/sj_outs</span>
<span id="cb1-187"><a href="#cb1-187"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="st">"</span><span class="va">$project_folder</span><span class="st">"</span>/results/raw/solo_output/process_logs</span>
<span id="cb1-188"><a href="#cb1-188"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="st">"</span><span class="va">$project_folder</span><span class="st">"</span>/results/raw/solo_output/results_data</span>
<span id="cb1-189"><a href="#cb1-189"></a></span>
<span id="cb1-190"><a href="#cb1-190"></a><span class="fu">mv</span>  <span class="st">"</span><span class="va">$project_folder</span><span class="st">"</span>/results/raw/solo_output/<span class="pp">*</span>.bam <span class="dt">\</span></span>
<span id="cb1-191"><a href="#cb1-191"></a>    <span class="st">"</span><span class="va">$project_folder</span><span class="st">"</span>/results/raw/solo_output/bams</span>
<span id="cb1-192"><a href="#cb1-192"></a><span class="fu">mv</span>  <span class="st">"</span><span class="va">$project_folder</span><span class="st">"</span>/results/raw/solo_output/<span class="pp">*</span>.Log.final.out <span class="dt">\</span></span>
<span id="cb1-193"><a href="#cb1-193"></a>    <span class="st">"</span><span class="va">$project_folder</span><span class="st">"</span>/results/raw/solo_output/final_logs</span>
<span id="cb1-194"><a href="#cb1-194"></a><span class="fu">mv</span>  <span class="st">"</span><span class="va">$project_folder</span><span class="st">"</span>/results/raw/solo_output/<span class="pp">*</span>.Log.progress.out <span class="dt">\</span></span>
<span id="cb1-195"><a href="#cb1-195"></a>    <span class="st">"</span><span class="va">$project_folder</span><span class="st">"</span>/results/raw/solo_output/progress_logs</span>
<span id="cb1-196"><a href="#cb1-196"></a><span class="fu">mv</span>  <span class="st">"</span><span class="va">$project_folder</span><span class="st">"</span>/results/raw/solo_output/<span class="pp">*</span>.SJ.out.tab <span class="dt">\</span></span>
<span id="cb1-197"><a href="#cb1-197"></a>    <span class="st">"</span><span class="va">$project_folder</span><span class="st">"</span>/results/raw/solo_output/sj_outs</span>
<span id="cb1-198"><a href="#cb1-198"></a><span class="fu">mv</span>  <span class="st">"</span><span class="va">$project_folder</span><span class="st">"</span>/results/raw/solo_output/<span class="pp">*</span>star.Log.out <span class="dt">\</span></span>
<span id="cb1-199"><a href="#cb1-199"></a>    <span class="st">"</span><span class="va">$project_folder</span><span class="st">"</span>/results/raw/solo_output/process_logs</span>
<span id="cb1-200"><a href="#cb1-200"></a><span class="fu">mv</span>  <span class="st">"</span><span class="va">$project_folder</span><span class="st">"</span>/results/raw/solo_output/<span class="pp">*</span>star.Solo.out <span class="dt">\</span></span>
<span id="cb1-201"><a href="#cb1-201"></a>    <span class="st">"</span><span class="va">$project_folder</span><span class="st">"</span>/results/raw/solo_output/results_data</span>
<span id="cb1-202"><a href="#cb1-202"></a></span>
<span id="cb1-203"><a href="#cb1-203"></a><span class="co"># --soloCellFilter EmptyDrops_CR 1000 0.99 10 45000 90000 1000 0.01 20000 0.01 10000 \</span></span>
<span id="cb1-204"><a href="#cb1-204"></a></span>
<span id="cb1-205"><a href="#cb1-205"></a><span class="co"># fastq_1=data/full_reads/WTCHG_928874_CU001_1.fastq</span></span>
<span id="cb1-206"><a href="#cb1-206"></a><span class="co"># fastq_2=data/full_reads/WTCHG_928874_CU001_2.fastq</span></span>
<span id="cb1-207"><a href="#cb1-207"></a><span class="co"># echo ${fastq_1}</span></span>
<span id="cb1-208"><a href="#cb1-208"></a><span class="co"># echo ${fastq_2}</span></span>
<span id="cb1-209"><a href="#cb1-209"></a></span>
<span id="cb1-210"><a href="#cb1-210"></a></span>
<span id="cb1-211"><a href="#cb1-211"></a></span>
<span id="cb1-212"><a href="#cb1-212"></a><span class="co"># STAR \</span></span>
<span id="cb1-213"><a href="#cb1-213"></a><span class="co"># --runThreadN 60 \</span></span>
<span id="cb1-214"><a href="#cb1-214"></a><span class="co"># --genomeDir ${genomedir} \</span></span>
<span id="cb1-215"><a href="#cb1-215"></a><span class="co"># --outFileNamePrefix ${solo_folder}/sample_CU001_star. \</span></span>
<span id="cb1-216"><a href="#cb1-216"></a><span class="co"># --readFilesIn ${fastq_2} ${fastq_1} \</span></span>
<span id="cb1-217"><a href="#cb1-217"></a><span class="co"># --soloType CB_UMI_Simple \</span></span>
<span id="cb1-218"><a href="#cb1-218"></a><span class="co"># --soloCBwhitelist None \</span></span>
<span id="cb1-219"><a href="#cb1-219"></a><span class="co"># --soloCBstart 1 \</span></span>
<span id="cb1-220"><a href="#cb1-220"></a><span class="co"># --soloCBlen 12 \</span></span>
<span id="cb1-221"><a href="#cb1-221"></a><span class="co"># --soloUMIstart 13 \</span></span>
<span id="cb1-222"><a href="#cb1-222"></a><span class="co"># --soloUMIlen 8 \</span></span>
<span id="cb1-223"><a href="#cb1-223"></a><span class="co"># --soloBarcodeReadLength 0 \</span></span>
<span id="cb1-224"><a href="#cb1-224"></a><span class="co"># --soloStrand Unstranded \</span></span>
<span id="cb1-225"><a href="#cb1-225"></a><span class="co"># --clipAdapterType CellRanger4 \</span></span>
<span id="cb1-226"><a href="#cb1-226"></a><span class="co"># --soloCellFilter EmptyDrops_CR \</span></span>
<span id="cb1-227"><a href="#cb1-227"></a><span class="co"># --outSAMattributes All RG \</span></span>
<span id="cb1-228"><a href="#cb1-228"></a><span class="co"># --outSAMattrRGline ID:CU001 SM:CU001 \</span></span>
<span id="cb1-229"><a href="#cb1-229"></a><span class="co"># --outSAMtype BAM SortedByCoordinate \</span></span>
<span id="cb1-230"><a href="#cb1-230"></a><span class="co"># --outMultimapperOrder Random \</span></span>
<span id="cb1-231"><a href="#cb1-231"></a><span class="co"># --soloCellReadStats Standard \</span></span>
<span id="cb1-232"><a href="#cb1-232"></a><span class="co"># --soloFeatures Gene GeneFull Velocyto \</span></span>
<span id="cb1-233"><a href="#cb1-233"></a><span class="co"># --soloMultiMappers EM \</span></span>
<span id="cb1-234"><a href="#cb1-234"></a><span class="co"># --outFilterMultimapNmax 10</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="r-bit" class="level1">
<h1>‘R bit’</h1>
<ul>
<li><p>The below script is at <code>/scratch/moh1u21/23-01-30-whole_muscle_snrnaseq/src/whole_muscle_dsn_qc.R</code></p></li>
<li><p>Ive saved a number of objects along the way, so you shouldnt have to run the whole thing again and again, and i’d advise you not to as it may change slightly each time, or over time.</p></li>
<li><p>Note - the final Seurat object does not exclude doublets (of which there were not many - have been documented in csv files produced along the way) and background contamination (again few cells with significant contamination) - ideally would have done so but ran out of time. This diverges from what may be typically expected and the next person to take on this code may wish to carry on in this vein.</p></li>
<li><p>Note - a threshold of 4.5 was chosen for single nuclear as graphs showed samples with high mitochondrial gene expression also with high read count, indicating should not necessarily be removed.</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(DropletUtils))</span>
<span id="cb2-2"><a href="#cb2-2"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(reticulate))</span>
<span id="cb2-3"><a href="#cb2-3"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(singleCellTK))</span>
<span id="cb2-4"><a href="#cb2-4"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(AnnotationHub))</span>
<span id="cb2-5"><a href="#cb2-5"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(SingleCellExperiment))</span>
<span id="cb2-6"><a href="#cb2-6"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(scater))</span>
<span id="cb2-7"><a href="#cb2-7"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(ensembldb))</span>
<span id="cb2-8"><a href="#cb2-8"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(BiocParallel))</span>
<span id="cb2-9"><a href="#cb2-9"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(tidyverse))</span>
<span id="cb2-10"><a href="#cb2-10"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(patchwork))</span>
<span id="cb2-11"><a href="#cb2-11"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(ggvenn))</span>
<span id="cb2-12"><a href="#cb2-12"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(magrittr))</span>
<span id="cb2-13"><a href="#cb2-13"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(Seurat))</span>
<span id="cb2-14"><a href="#cb2-14"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(clustree))</span>
<span id="cb2-15"><a href="#cb2-15"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(scran))</span>
<span id="cb2-16"><a href="#cb2-16"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(bluster))</span>
<span id="cb2-17"><a href="#cb2-17"></a></span>
<span id="cb2-18"><a href="#cb2-18"></a><span class="fu">set.seed</span>(<span class="dv">9816</span>)</span>
<span id="cb2-19"><a href="#cb2-19"></a></span>
<span id="cb2-20"><a href="#cb2-20"></a>bp.params <span class="ot">&lt;-</span> <span class="fu">MulticoreParam</span>(<span class="at">workers =</span> <span class="dv">40</span>)</span>
<span id="cb2-21"><a href="#cb2-21"></a></span>
<span id="cb2-22"><a href="#cb2-22"></a>starsolodirs <span class="ot">&lt;-</span> <span class="fu">dir</span>(<span class="st">"/scratch/moh1u21/23-01-30-whole_muscle_snrnaseq/results/raw/solo_output/results_data"</span>, <span class="at">full.names=</span>T)</span>
<span id="cb2-23"><a href="#cb2-23"></a></span>
<span id="cb2-24"><a href="#cb2-24"></a>samples <span class="ot">&lt;-</span></span>
<span id="cb2-25"><a href="#cb2-25"></a>    <span class="fu">c</span>(<span class="st">"sample_a_178K400"</span>,</span>
<span id="cb2-26"><a href="#cb2-26"></a>    <span class="st">"sample_b_178K300"</span>,</span>
<span id="cb2-27"><a href="#cb2-27"></a>    <span class="st">"sample_c_178K200"</span>,</span>
<span id="cb2-28"><a href="#cb2-28"></a>    <span class="st">"sample_d_259K400"</span>,</span>
<span id="cb2-29"><a href="#cb2-29"></a>    <span class="st">"sample_e_259K300"</span>,</span>
<span id="cb2-30"><a href="#cb2-30"></a>    <span class="st">"sample_h_12K112"</span>)</span>
<span id="cb2-31"><a href="#cb2-31"></a></span>
<span id="cb2-32"><a href="#cb2-32"></a><span class="co"># saveRDS(data/bibs.rds)</span></span>
<span id="cb2-33"><a href="#cb2-33"></a><span class="co"># bibs &lt;- readRDS("data/bibs.rds")</span></span>
<span id="cb2-34"><a href="#cb2-34"></a></span>
<span id="cb2-35"><a href="#cb2-35"></a>bibs <span class="ot">&lt;-</span> </span>
<span id="cb2-36"><a href="#cb2-36"></a><span class="fu">importSTARsolo</span>(</span>
<span id="cb2-37"><a href="#cb2-37"></a>  starsolodirs,</span>
<span id="cb2-38"><a href="#cb2-38"></a>  samples,</span>
<span id="cb2-39"><a href="#cb2-39"></a>  <span class="at">STARsoloOuts =</span> <span class="st">"GeneFull/raw"</span>,</span>
<span id="cb2-40"><a href="#cb2-40"></a>  <span class="at">matrixFileNames =</span> <span class="st">"UniqueAndMult-EM.mtx"</span>,</span>
<span id="cb2-41"><a href="#cb2-41"></a>  <span class="at">featuresFileNames =</span> <span class="st">"features.tsv"</span>,</span>
<span id="cb2-42"><a href="#cb2-42"></a>  <span class="at">barcodesFileNames =</span> <span class="st">"barcodes.tsv"</span>,</span>
<span id="cb2-43"><a href="#cb2-43"></a>  <span class="at">gzipped =</span> <span class="st">"auto"</span>,</span>
<span id="cb2-44"><a href="#cb2-44"></a>  <span class="at">class =</span> <span class="fu">c</span>(<span class="st">"Matrix"</span>, <span class="st">"matrix"</span>),</span>
<span id="cb2-45"><a href="#cb2-45"></a>  <span class="at">delayedArray =</span> <span class="cn">FALSE</span>,</span>
<span id="cb2-46"><a href="#cb2-46"></a>  <span class="at">rowNamesDedup =</span> <span class="cn">TRUE</span></span>
<span id="cb2-47"><a href="#cb2-47"></a>)</span>
<span id="cb2-48"><a href="#cb2-48"></a></span>
<span id="cb2-49"><a href="#cb2-49"></a>bibs_qcd <span class="ot">&lt;-</span> </span>
<span id="cb2-50"><a href="#cb2-50"></a>  <span class="fu">runBarcodeRankDrops</span>(</span>
<span id="cb2-51"><a href="#cb2-51"></a>    bibs,</span>
<span id="cb2-52"><a href="#cb2-52"></a>    <span class="at">sample =</span> <span class="fu">colData</span>(bibs)<span class="sc">$</span>sample,</span>
<span id="cb2-53"><a href="#cb2-53"></a>    <span class="at">useAssay =</span> <span class="st">"counts"</span>,</span>
<span id="cb2-54"><a href="#cb2-54"></a>    <span class="at">lower =</span> <span class="dv">500</span>,</span>
<span id="cb2-55"><a href="#cb2-55"></a>    <span class="at">fitBounds =</span> <span class="cn">NULL</span>,</span>
<span id="cb2-56"><a href="#cb2-56"></a>    <span class="at">df =</span> <span class="dv">20</span></span>
<span id="cb2-57"><a href="#cb2-57"></a>  )</span>
<span id="cb2-58"><a href="#cb2-58"></a></span>
<span id="cb2-59"><a href="#cb2-59"></a>plots <span class="ot">&lt;-</span> </span>
<span id="cb2-60"><a href="#cb2-60"></a>  <span class="fu">plotBarcodeRankScatter</span>(</span>
<span id="cb2-61"><a href="#cb2-61"></a>    bibs_qcd,</span>
<span id="cb2-62"><a href="#cb2-62"></a>    <span class="at">sample =</span> <span class="fu">colData</span>(bibs_qcd)<span class="sc">$</span>sample,</span>
<span id="cb2-63"><a href="#cb2-63"></a>    <span class="at">defaultTheme =</span> <span class="cn">FALSE</span>,</span>
<span id="cb2-64"><a href="#cb2-64"></a>    <span class="at">dotSize =</span> <span class="fl">0.5</span>,</span>
<span id="cb2-65"><a href="#cb2-65"></a>    <span class="at">title =</span> <span class="cn">NULL</span>,</span>
<span id="cb2-66"><a href="#cb2-66"></a>    <span class="at">titleSize =</span> <span class="dv">12</span>,</span>
<span id="cb2-67"><a href="#cb2-67"></a>    <span class="at">xlab =</span> <span class="cn">NULL</span>,</span>
<span id="cb2-68"><a href="#cb2-68"></a>    <span class="at">ylab =</span> <span class="cn">NULL</span>,</span>
<span id="cb2-69"><a href="#cb2-69"></a>    <span class="at">axisSize =</span> <span class="dv">12</span>,</span>
<span id="cb2-70"><a href="#cb2-70"></a>    <span class="at">axisLabelSize =</span> <span class="dv">15</span>,</span>
<span id="cb2-71"><a href="#cb2-71"></a>    <span class="at">legendSize =</span> <span class="dv">10</span>,</span>
<span id="cb2-72"><a href="#cb2-72"></a>    <span class="at">combinePlot =</span> <span class="st">"all"</span>,</span>
<span id="cb2-73"><a href="#cb2-73"></a>    <span class="at">sampleRelHeights =</span> <span class="dv">1</span>,</span>
<span id="cb2-74"><a href="#cb2-74"></a>    <span class="at">sampleRelWidths =</span> <span class="dv">1</span></span>
<span id="cb2-75"><a href="#cb2-75"></a>  )</span>
<span id="cb2-76"><a href="#cb2-76"></a></span>
<span id="cb2-77"><a href="#cb2-77"></a><span class="co"># # adjust to add titles to plots</span></span>
<span id="cb2-78"><a href="#cb2-78"></a><span class="co"># plots &lt;- flatten(plots)</span></span>
<span id="cb2-79"><a href="#cb2-79"></a><span class="co"># plots[[1]] + labs(title="Plot of length \n by dose")</span></span>
<span id="cb2-80"><a href="#cb2-80"></a><span class="co"># gridExtra::grid.arrange(grobs=plots, ncol=3)</span></span>
<span id="cb2-81"><a href="#cb2-81"></a></span>
<span id="cb2-82"><a href="#cb2-82"></a>bibs_qcd <span class="ot">&lt;-</span> </span>
<span id="cb2-83"><a href="#cb2-83"></a>  <span class="fu">runEmptyDrops</span>(</span>
<span id="cb2-84"><a href="#cb2-84"></a>    bibs_qcd,</span>
<span id="cb2-85"><a href="#cb2-85"></a>    <span class="at">lower =</span> <span class="dv">500</span>,</span>
<span id="cb2-86"><a href="#cb2-86"></a>    <span class="at">sample =</span> <span class="fu">colData</span>(bibs)<span class="sc">$</span>sample,</span>
<span id="cb2-87"><a href="#cb2-87"></a>    <span class="at">retain =</span> <span class="dv">1000</span></span>
<span id="cb2-88"><a href="#cb2-88"></a>    <span class="co"># BPPARAM = bp.params</span></span>
<span id="cb2-89"><a href="#cb2-89"></a>  )</span>
<span id="cb2-90"><a href="#cb2-90"></a></span>
<span id="cb2-91"><a href="#cb2-91"></a><span class="fu">table</span>(<span class="fu">duplicated</span>(<span class="fu">rownames</span>(bibs_qcd)))</span>
<span id="cb2-92"><a href="#cb2-92"></a><span class="fu">rownames</span>(bibs_qcd) <span class="ot">&lt;-</span> scuttle<span class="sc">::</span><span class="fu">uniquifyFeatureNames</span>(<span class="fu">rownames</span>(bibs_qcd), <span class="fu">rowData</span>(bibs_qcd)<span class="sc">$</span>feature_name)</span>
<span id="cb2-93"><a href="#cb2-93"></a><span class="fu">rowData</span>(bibs_qcd)<span class="sc">$</span>updated_feature_name <span class="ot">&lt;-</span> <span class="fu">rownames</span>(bibs_qcd)</span>
<span id="cb2-94"><a href="#cb2-94"></a><span class="fu">table</span>(<span class="fu">duplicated</span>(<span class="fu">rownames</span>(bibs_qcd)))</span>
<span id="cb2-95"><a href="#cb2-95"></a><span class="co"># saveRDS(bibs_qcd, "data/bibs_qcd.rds")</span></span>
<span id="cb2-96"><a href="#cb2-96"></a>bibs_qcd <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"data/bibs_qcd.rds"</span>)</span>
<span id="cb2-97"><a href="#cb2-97"></a></span>
<span id="cb2-98"><a href="#cb2-98"></a>bibs_ed <span class="ot">&lt;-</span> bibs_qcd </span>
<span id="cb2-99"><a href="#cb2-99"></a></span>
<span id="cb2-100"><a href="#cb2-100"></a><span class="co"># abbles &lt;-  </span></span>
<span id="cb2-101"><a href="#cb2-101"></a><span class="co"># runDropletQC(</span></span>
<span id="cb2-102"><a href="#cb2-102"></a><span class="co">#   bibs,</span></span>
<span id="cb2-103"><a href="#cb2-103"></a><span class="co">#   algorithms = c("QCMetrics", "emptyDrops", "barcodeRanks"),</span></span>
<span id="cb2-104"><a href="#cb2-104"></a><span class="co">#   sample = colData(bibs)$sample,</span></span>
<span id="cb2-105"><a href="#cb2-105"></a><span class="co">#   useAssay = "counts",</span></span>
<span id="cb2-106"><a href="#cb2-106"></a><span class="co">#   paramsList = NULL</span></span>
<span id="cb2-107"><a href="#cb2-107"></a><span class="co"># )</span></span>
<span id="cb2-108"><a href="#cb2-108"></a></span>
<span id="cb2-109"><a href="#cb2-109"></a></span>
<span id="cb2-110"><a href="#cb2-110"></a><span class="co"># Testing whether emptyDrops is sound -------------------------------------</span></span>
<span id="cb2-111"><a href="#cb2-111"></a></span>
<span id="cb2-112"><a href="#cb2-112"></a><span class="fu">set.seed</span>(<span class="dv">100</span>)</span>
<span id="cb2-113"><a href="#cb2-113"></a>all.out <span class="ot">&lt;-</span> <span class="fu">emptyDrops</span>(<span class="fu">counts</span>(bibs), <span class="at">lower=</span><span class="dv">500</span>, <span class="at">retain=</span><span class="dv">1000</span>, <span class="at">test.ambient=</span><span class="cn">TRUE</span>)</span>
<span id="cb2-114"><a href="#cb2-114"></a><span class="fu">hist</span>(all.out<span class="sc">$</span>PValue[all.out<span class="sc">$</span>Total <span class="sc">&lt;=</span> <span class="dv">500</span> <span class="sc">&amp;</span> all.out<span class="sc">$</span>Total <span class="sc">&gt;</span> <span class="dv">0</span>],</span>
<span id="cb2-115"><a href="#cb2-115"></a>     <span class="at">xlab=</span><span class="st">"P-value"</span>, <span class="at">main=</span><span class="st">""</span>, <span class="at">col=</span><span class="st">"grey80"</span>) </span>
<span id="cb2-116"><a href="#cb2-116"></a></span>
<span id="cb2-117"><a href="#cb2-117"></a><span class="fu">mean</span>(all.out<span class="sc">$</span>PValue[all.out<span class="sc">$</span>Total <span class="sc">&lt;=</span> <span class="dv">500</span> <span class="sc">&amp;</span> all.out<span class="sc">$</span>Total <span class="sc">&gt;</span> <span class="dv">0</span>] <span class="sc">&lt;=</span> <span class="fl">0.05</span>)</span>
<span id="cb2-118"><a href="#cb2-118"></a></span>
<span id="cb2-119"><a href="#cb2-119"></a><span class="co"># is.cell &lt;- abbles$dropletUtils_emptyDrops_fdr &lt;= 0.001</span></span>
<span id="cb2-120"><a href="#cb2-120"></a><span class="co"># sum(is.cell, na.rm=TRUE)</span></span>
<span id="cb2-121"><a href="#cb2-121"></a></span>
<span id="cb2-122"><a href="#cb2-122"></a><span class="co"># saveRDS(bibs_ed, "empty_dropped_sce.rds")</span></span>
<span id="cb2-123"><a href="#cb2-123"></a><span class="co"># bibs_ed &lt;- readRDS("empty_dropped_sce.rds")</span></span>
<span id="cb2-124"><a href="#cb2-124"></a></span>
<span id="cb2-125"><a href="#cb2-125"></a></span>
<span id="cb2-126"><a href="#cb2-126"></a><span class="co"># Plot empty drops results ------------------------------------------------</span></span>
<span id="cb2-127"><a href="#cb2-127"></a></span>
<span id="cb2-128"><a href="#cb2-128"></a><span class="fu">plotEmptyDropsResults</span>(bibs_ed, <span class="at">sample=</span>bibs_ed<span class="sc">$</span>samples)</span>
<span id="cb2-129"><a href="#cb2-129"></a><span class="fu">summary</span>(bibs_ed<span class="sc">$</span>dropletUtils_emptyDrops_fdr <span class="sc">&lt;=</span> <span class="fl">0.001</span>)</span>
<span id="cb2-130"><a href="#cb2-130"></a><span class="fu">table</span>(<span class="at">Sig=</span>bibs_ed<span class="sc">$</span>dropletUtils_emptyDrops_fdr <span class="sc">&lt;=</span> <span class="fl">0.001</span>, <span class="at">Limited=</span>bibs_ed<span class="sc">$</span>dropletUtils_emptyDrops_limited)</span>
<span id="cb2-131"><a href="#cb2-131"></a></span>
<span id="cb2-132"><a href="#cb2-132"></a><span class="fu">colData</span>(bibs_ed) <span class="sc">%&gt;%</span></span>
<span id="cb2-133"><a href="#cb2-133"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb2-134"><a href="#cb2-134"></a>  <span class="fu">filter</span>(dropletUtils_emptyDrops_fdr <span class="sc">&lt;=</span> <span class="fl">0.001</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-135"><a href="#cb2-135"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(dropletUtils_emptyDrops_fdr)) <span class="sc">%&gt;%</span></span>
<span id="cb2-136"><a href="#cb2-136"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb2-137"><a href="#cb2-137"></a>  <span class="fu">summarise</span>(<span class="at">cells=</span><span class="fu">n</span>()) <span class="sc">%T&gt;%</span></span>
<span id="cb2-138"><a href="#cb2-138"></a>  readr<span class="sc">::</span><span class="fu">write_excel_csv</span>(<span class="st">"results/cellno_per_sample.csv"</span>, <span class="at">col_names=</span>T)</span>
<span id="cb2-139"><a href="#cb2-139"></a></span>
<span id="cb2-140"><a href="#cb2-140"></a>bibs_dropped <span class="ot">&lt;-</span> bibs_ed[,<span class="fu">which</span>(bibs_ed<span class="sc">$</span>dropletUtils_emptyDrops_fdr <span class="sc">&lt;=</span> <span class="fl">0.001</span>)]</span>
<span id="cb2-141"><a href="#cb2-141"></a>bibs_dropped <span class="ot">&lt;-</span> bibs_dropped[,<span class="sc">!</span><span class="fu">is.na</span>(bibs_dropped<span class="sc">$</span>dropletUtils_emptyDrops_fdr)]</span>
<span id="cb2-142"><a href="#cb2-142"></a></span>
<span id="cb2-143"><a href="#cb2-143"></a>is.cell <span class="ot">&lt;-</span> bibs_ed<span class="sc">$</span>dropletUtils_emptyDrops_fdr <span class="sc">&lt;=</span> <span class="fl">0.001</span></span>
<span id="cb2-144"><a href="#cb2-144"></a><span class="fu">sum</span>(is.cell, <span class="at">na.rm=</span><span class="cn">TRUE</span>)</span>
<span id="cb2-145"><a href="#cb2-145"></a></span>
<span id="cb2-146"><a href="#cb2-146"></a><span class="co"># set.seed(12345)</span></span>
<span id="cb2-147"><a href="#cb2-147"></a><span class="co"># bibsy &lt;- runCellQC(bibs_dropped, </span></span>
<span id="cb2-148"><a href="#cb2-148"></a><span class="co">#                    algorithms = c(</span></span>
<span id="cb2-149"><a href="#cb2-149"></a><span class="co">#                      "QCMetrics", </span></span>
<span id="cb2-150"><a href="#cb2-150"></a><span class="co">#                      "scrublet", </span></span>
<span id="cb2-151"><a href="#cb2-151"></a><span class="co">#                      "scDblFinder", </span></span>
<span id="cb2-152"><a href="#cb2-152"></a><span class="co">#                      "cxds",</span></span>
<span id="cb2-153"><a href="#cb2-153"></a><span class="co">#                      "bcds",</span></span>
<span id="cb2-154"><a href="#cb2-154"></a><span class="co">#                      "cxds_bcds_hybrid",</span></span>
<span id="cb2-155"><a href="#cb2-155"></a><span class="co">#                      "doubletFinder", </span></span>
<span id="cb2-156"><a href="#cb2-156"></a><span class="co">#                      "decontX"</span></span>
<span id="cb2-157"><a href="#cb2-157"></a><span class="co">#                     #  "soupX"</span></span>
<span id="cb2-158"><a href="#cb2-158"></a><span class="co">#                    ), </span></span>
<span id="cb2-159"><a href="#cb2-159"></a><span class="co">#                    background=bibs_ed,</span></span>
<span id="cb2-160"><a href="#cb2-160"></a><span class="co">#                    mitoPrefix="^MT-",</span></span>
<span id="cb2-161"><a href="#cb2-161"></a><span class="co">#                    sample = colData(bibs_dropped)$sample</span></span>
<span id="cb2-162"><a href="#cb2-162"></a><span class="co">#                    )</span></span>
<span id="cb2-163"><a href="#cb2-163"></a><span class="co"># </span></span>
<span id="cb2-164"><a href="#cb2-164"></a><span class="co"># # saveRDS(bibsy, "data/bibsy.rds")</span></span>
<span id="cb2-165"><a href="#cb2-165"></a><span class="co"># bibsy &lt;- readRDS("data/bibsy.rds")</span></span>
<span id="cb2-166"><a href="#cb2-166"></a><span class="co"># </span></span>
<span id="cb2-167"><a href="#cb2-167"></a><span class="co"># table(colData(bibsy)$scrublet_call)</span></span>
<span id="cb2-168"><a href="#cb2-168"></a><span class="co"># table(colData(bibsy)$scDblFinder_doublet_call)</span></span>
<span id="cb2-169"><a href="#cb2-169"></a><span class="co"># length(which(colData(bibsy)$decontX_contamination &lt; 0.4))</span></span>
<span id="cb2-170"><a href="#cb2-170"></a><span class="co"># </span></span>
<span id="cb2-171"><a href="#cb2-171"></a><span class="co"># hist(colData(bibsy)$decontX_contamination)</span></span>
<span id="cb2-172"><a href="#cb2-172"></a></span>
<span id="cb2-173"><a href="#cb2-173"></a></span>
<span id="cb2-174"><a href="#cb2-174"></a><span class="co"># # Normalization.</span></span>
<span id="cb2-175"><a href="#cb2-175"></a><span class="co"># set.seed(1234)</span></span>
<span id="cb2-176"><a href="#cb2-176"></a><span class="co"># bibsy_umap &lt;- logNormCounts(bibsy)</span></span>
<span id="cb2-177"><a href="#cb2-177"></a><span class="co"># dec &lt;- modelGeneVar(bibsy_umap)</span></span>
<span id="cb2-178"><a href="#cb2-178"></a><span class="co"># hvg &lt;- getTopHVGs(dec, prop=0.1)</span></span>
<span id="cb2-179"><a href="#cb2-179"></a><span class="co"># bibsy_umap &lt;- runPCA(bibsy_umap, ncomponents=25, subset_row=hvg)</span></span>
<span id="cb2-180"><a href="#cb2-180"></a><span class="co"># colLabels(bibsy_umap) &lt;- clusterCells(bibsy_umap, use.dimred='PCA',</span></span>
<span id="cb2-181"><a href="#cb2-181"></a><span class="co">#     BLUSPARAM=NNGraphParam(cluster.fun="louvain"))    </span></span>
<span id="cb2-182"><a href="#cb2-182"></a><span class="co"># bibsy_umap &lt;- runUMAP(bibsy_umap, dimred = 'PCA')</span></span>
<span id="cb2-183"><a href="#cb2-183"></a><span class="co"># plotUMAP(bibsy_umap, colour_by="label")</span></span>
<span id="cb2-184"><a href="#cb2-184"></a></span>
<span id="cb2-185"><a href="#cb2-185"></a></span>
<span id="cb2-186"><a href="#cb2-186"></a><span class="co"># oncle &lt;- </span></span>
<span id="cb2-187"><a href="#cb2-187"></a><span class="co">#   plotDecontXResults(</span></span>
<span id="cb2-188"><a href="#cb2-188"></a><span class="co">#     bibsy,</span></span>
<span id="cb2-189"><a href="#cb2-189"></a><span class="co">#     reducedDimName="decontX_sample_a_178K400_UMAP"</span></span>
<span id="cb2-190"><a href="#cb2-190"></a><span class="co">#   )</span></span>
<span id="cb2-191"><a href="#cb2-191"></a></span>
<span id="cb2-192"><a href="#cb2-192"></a></span>
<span id="cb2-193"><a href="#cb2-193"></a></span>
<span id="cb2-194"><a href="#cb2-194"></a><span class="co"># Diagnostic plots --------------------------------------------------------</span></span>
<span id="cb2-195"><a href="#cb2-195"></a></span>
<span id="cb2-196"><a href="#cb2-196"></a>genesPerCell <span class="ot">&lt;-</span> <span class="fu">colSums</span>(<span class="fu">counts</span>(bibs_dropped) <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb2-197"><a href="#cb2-197"></a><span class="fu">summary</span>(genesPerCell)</span>
<span id="cb2-198"><a href="#cb2-198"></a><span class="fu">plot</span>(<span class="fu">density</span>(genesPerCell), <span class="at">main=</span><span class="st">""</span>, <span class="at">xlab=</span><span class="st">"Genes per cell"</span>)</span>
<span id="cb2-199"><a href="#cb2-199"></a></span>
<span id="cb2-200"><a href="#cb2-200"></a>tmpCounts <span class="ot">&lt;-</span> <span class="fu">counts</span>(bibs_dropped)[,<span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>]</span>
<span id="cb2-201"><a href="#cb2-201"></a></span>
<span id="cb2-202"><a href="#cb2-202"></a><span class="fu">plot</span>(<span class="fu">rowSums</span>(tmpCounts),</span>
<span id="cb2-203"><a href="#cb2-203"></a>     <span class="fu">rowMeans</span>(tmpCounts <span class="sc">&gt;</span> <span class="dv">0</span>),</span>
<span id="cb2-204"><a href="#cb2-204"></a>     <span class="at">log =</span> <span class="st">"x"</span>,</span>
<span id="cb2-205"><a href="#cb2-205"></a>     <span class="at">xlab=</span><span class="st">"total number of UMIs"</span>,</span>
<span id="cb2-206"><a href="#cb2-206"></a>     <span class="at">ylab=</span><span class="st">"proportion of cells expressing the gene"</span></span>
<span id="cb2-207"><a href="#cb2-207"></a>)</span>
<span id="cb2-208"><a href="#cb2-208"></a></span>
<span id="cb2-209"><a href="#cb2-209"></a><span class="fu">rm</span>(tmpCounts)</span>
<span id="cb2-210"><a href="#cb2-210"></a></span>
<span id="cb2-211"><a href="#cb2-211"></a>rel_expression <span class="ot">&lt;-</span> <span class="fu">t</span>( <span class="fu">t</span>(<span class="fu">counts</span>(bibs_dropped)) <span class="sc">/</span> <span class="fu">colSums</span>(<span class="fu">counts</span>(bibs_dropped))) <span class="sc">*</span> <span class="dv">100</span></span>
<span id="cb2-212"><a href="#cb2-212"></a><span class="fu">rownames</span>(rel_expression) <span class="ot">&lt;-</span> <span class="fu">rowData</span>(bibs_dropped)<span class="sc">$</span>feature_name</span>
<span id="cb2-213"><a href="#cb2-213"></a>most_expressed <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">rowSums</span>( rel_expression ),T)[<span class="dv">20</span><span class="sc">:</span><span class="dv">1</span>]</span>
<span id="cb2-214"><a href="#cb2-214"></a>plot_data <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">t</span>(rel_expression[<span class="fu">names</span>(most_expressed),]))</span>
<span id="cb2-215"><a href="#cb2-215"></a></span>
<span id="cb2-216"><a href="#cb2-216"></a><span class="fu">boxplot</span>(plot_data, <span class="at">cex=</span><span class="fl">0.1</span>, <span class="at">las=</span><span class="dv">1</span>, <span class="at">xlab=</span><span class="st">"% total count per cell"</span>, <span class="at">horizontal=</span><span class="cn">TRUE</span>)</span>
<span id="cb2-217"><a href="#cb2-217"></a></span>
<span id="cb2-218"><a href="#cb2-218"></a><span class="co"># Remove genes with zero expression across cell pop.</span></span>
<span id="cb2-219"><a href="#cb2-219"></a>detected_genes <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(bibs_dropped)) <span class="sc">&gt;</span> <span class="dv">0</span></span>
<span id="cb2-220"><a href="#cb2-220"></a><span class="fu">table</span>(detected_genes)</span>
<span id="cb2-221"><a href="#cb2-221"></a>bibs_dropped <span class="ot">&lt;-</span> bibs_dropped[detected_genes,]</span>
<span id="cb2-222"><a href="#cb2-222"></a></span>
<span id="cb2-223"><a href="#cb2-223"></a></span>
<span id="cb2-224"><a href="#cb2-224"></a><span class="co"># ah &lt;- AnnotationHub()</span></span>
<span id="cb2-225"><a href="#cb2-225"></a><span class="co"># ens.mm.98 &lt;- query(ah, c("Homo sapiens", "EnsDb", 98))[[1]] </span></span>
<span id="cb2-226"><a href="#cb2-226"></a></span>
<span id="cb2-227"><a href="#cb2-227"></a><span class="co"># genes &lt;- rowData(bibs_dropped)$feature_ID</span></span>
<span id="cb2-228"><a href="#cb2-228"></a><span class="co"># gene_annot &lt;- AnnotationDbi::select(ens.mm.98, </span></span>
<span id="cb2-229"><a href="#cb2-229"></a><span class="co">#                                     keys = genes,</span></span>
<span id="cb2-230"><a href="#cb2-230"></a><span class="co">#                                     keytype = "GENEID",</span></span>
<span id="cb2-231"><a href="#cb2-231"></a><span class="co">#                                     columns = c("GENEID", "SEQNAME")) %&gt;%</span></span>
<span id="cb2-232"><a href="#cb2-232"></a><span class="co">#     set_names(c("ID", "Chromosome"))</span></span>
<span id="cb2-233"><a href="#cb2-233"></a><span class="co"># rowData(bibs_dropped) &lt;- merge(rowData(bibs_dropped), gene_annot, </span></span>
<span id="cb2-234"><a href="#cb2-234"></a><span class="co">#                                 by.x = "feature_ID", </span></span>
<span id="cb2-235"><a href="#cb2-235"></a><span class="co">#                                 by.y = "ID", </span></span>
<span id="cb2-236"><a href="#cb2-236"></a><span class="co">#                                 all.x=TRUE, </span></span>
<span id="cb2-237"><a href="#cb2-237"></a><span class="co">#                                 sort=FALSE)</span></span>
<span id="cb2-238"><a href="#cb2-238"></a><span class="co"># rownames(rowData(bibs_dropped)) &lt;- rowData(bibs_dropped)$feature_ID</span></span>
<span id="cb2-239"><a href="#cb2-239"></a></span>
<span id="cb2-240"><a href="#cb2-240"></a></span>
<span id="cb2-241"><a href="#cb2-241"></a><span class="co"># QC ----------------------------------------------------------------------</span></span>
<span id="cb2-242"><a href="#cb2-242"></a></span>
<span id="cb2-243"><a href="#cb2-243"></a><span class="co"># Assess number of mitochondrial genes in dataset</span></span>
<span id="cb2-244"><a href="#cb2-244"></a>jobbo <span class="ot">&lt;-</span> bibs_dropped[<span class="fu">grep</span>(<span class="st">"^MT-"</span>,<span class="fu">rownames</span>(bibs_dropped)),]</span>
<span id="cb2-245"><a href="#cb2-245"></a><span class="co"># jobbo &lt;- rowData(bibs_dropped)[grep("^MT",rowData(bibs_dropped)$Chromosome),]</span></span>
<span id="cb2-246"><a href="#cb2-246"></a><span class="fu">table</span>(<span class="fu">rownames</span>(<span class="fu">rowData</span>(bibs_dropped)) <span class="sc">%in%</span> <span class="fu">rownames</span>(jobbo))</span>
<span id="cb2-247"><a href="#cb2-247"></a></span>
<span id="cb2-248"><a href="#cb2-248"></a><span class="fu">head</span>(<span class="fu">rowData</span>(bibs_dropped))</span>
<span id="cb2-249"><a href="#cb2-249"></a></span>
<span id="cb2-250"><a href="#cb2-250"></a><span class="co"># Tag mito ogenes as mito</span></span>
<span id="cb2-251"><a href="#cb2-251"></a>is.mito <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">"^MT-"</span>,<span class="fu">rownames</span>(bibs_dropped))</span>
<span id="cb2-252"><a href="#cb2-252"></a></span>
<span id="cb2-253"><a href="#cb2-253"></a><span class="co"># QC of cells - lib size, features, mito, discard</span></span>
<span id="cb2-254"><a href="#cb2-254"></a>sce <span class="ot">&lt;-</span> <span class="fu">addPerCellQC</span>(bibs_dropped, <span class="at">subsets=</span><span class="fu">list</span>(<span class="at">Mito=</span>is.mito))</span>
<span id="cb2-255"><a href="#cb2-255"></a>cell_qc_results <span class="ot">&lt;-</span> <span class="fu">quickPerCellQC</span>(<span class="fu">colData</span>(sce), <span class="at">percent_subsets=</span><span class="fu">c</span>(<span class="st">"subsets_Mito_percent"</span>))</span>
<span id="cb2-256"><a href="#cb2-256"></a><span class="fu">colSums</span>(<span class="fu">as.data.frame</span>(cell_qc_results))</span>
<span id="cb2-257"><a href="#cb2-257"></a></span>
<span id="cb2-258"><a href="#cb2-258"></a><span class="co"># Add qc results back into sce object</span></span>
<span id="cb2-259"><a href="#cb2-259"></a>sce<span class="sc">$</span>low_lib_size <span class="ot">&lt;-</span> cell_qc_results<span class="sc">$</span>low_lib_size</span>
<span id="cb2-260"><a href="#cb2-260"></a>sce<span class="sc">$</span>low_n_features <span class="ot">&lt;-</span> cell_qc_results<span class="sc">$</span>low_n_features</span>
<span id="cb2-261"><a href="#cb2-261"></a>sce<span class="sc">$</span>high_Mito_percent <span class="ot">&lt;-</span> cell_qc_results<span class="sc">$</span>high_subsets_Mito_percent</span>
<span id="cb2-262"><a href="#cb2-262"></a>sce<span class="sc">$</span>discard <span class="ot">&lt;-</span> cell_qc_results<span class="sc">$</span>discard</span>
<span id="cb2-263"><a href="#cb2-263"></a>sce<span class="sc">$</span>sn_high_Mito_percent <span class="ot">&lt;-</span> <span class="fu">isOutlier</span>(sce<span class="sc">$</span>subsets_Mito_percent, <span class="at">type=</span><span class="st">"higher"</span>, <span class="at">min.diff=</span><span class="fl">0.5</span>) <span class="co"># adjustment for snuclear</span></span>
<span id="cb2-264"><a href="#cb2-264"></a></span>
<span id="cb2-265"><a href="#cb2-265"></a><span class="fu">summary</span>(sce<span class="sc">$</span>subsets_Mito_percent <span class="sc">==</span> <span class="dv">0</span>) <span class="co"># Assessing how many cells have zero mito gene expression</span></span>
<span id="cb2-266"><a href="#cb2-266"></a></span>
<span id="cb2-267"><a href="#cb2-267"></a><span class="co"># Writing no cells with high mito to csv</span></span>
<span id="cb2-268"><a href="#cb2-268"></a><span class="fu">colData</span>(sce) <span class="sc">%&gt;%</span></span>
<span id="cb2-269"><a href="#cb2-269"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb2-270"><a href="#cb2-270"></a>  <span class="fu">filter</span>(sn_high_Mito_percent <span class="sc">==</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-271"><a href="#cb2-271"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb2-272"><a href="#cb2-272"></a>  <span class="fu">summarise</span>(<span class="at">cells=</span><span class="fu">n</span>()) <span class="sc">%T&gt;%</span></span>
<span id="cb2-273"><a href="#cb2-273"></a>  readr<span class="sc">::</span><span class="fu">write_excel_csv</span>(<span class="st">"results/highmito_per_sample.csv"</span>, <span class="at">col_names=</span>T)</span>
<span id="cb2-274"><a href="#cb2-274"></a></span>
<span id="cb2-275"><a href="#cb2-275"></a><span class="co"># Comparing sn nuclear adjustrment to isOutlier mito detection</span></span>
<span id="cb2-276"><a href="#cb2-276"></a><span class="fu">table</span>(sce<span class="sc">$</span>sn_high_Mito_percent)</span>
<span id="cb2-277"><a href="#cb2-277"></a><span class="fu">table</span>(sce<span class="sc">$</span>high_Mito_percent)</span>
<span id="cb2-278"><a href="#cb2-278"></a></span>
<span id="cb2-279"><a href="#cb2-279"></a><span class="co"># sce &lt;- addPerCellQC(bibs_dropped, subsets=list(Mt=grep("^MT-", rownames(bibs_dropped))))</span></span>
<span id="cb2-280"><a href="#cb2-280"></a><span class="co"># summary(sce$subsets_Mt_percent)</span></span>
<span id="cb2-281"><a href="#cb2-281"></a><span class="co"># cell_qc_results &lt;- quickPerCellQC(colData(sce), percent_subsets=c("subsets_Mt_percent"))</span></span>
<span id="cb2-282"><a href="#cb2-282"></a><span class="co"># colSums(as.data.frame(cell_qc_results))</span></span>
<span id="cb2-283"><a href="#cb2-283"></a><span class="co"># sce$high_subsets_Mt_percent &lt;- isOutlier(sce$subsets_Mt_percent, type="higher", min.diff=0.5)</span></span>
<span id="cb2-284"><a href="#cb2-284"></a><span class="co"># cell_qc_results$discard &lt;- Reduce("|", cell_qc_results[,colnames(cell_qc_results)!="discard"])</span></span>
<span id="cb2-285"><a href="#cb2-285"></a><span class="co"># colSums(as.matrix(cell_qc_results))</span></span>
<span id="cb2-286"><a href="#cb2-286"></a></span>
<span id="cb2-287"><a href="#cb2-287"></a><span class="co"># plotColData(sce, y="subsets_Mt_percent",</span></span>
<span id="cb2-288"><a href="#cb2-288"></a><span class="co">#     colour_by=I(cell_qc_results$high_subsets_Mt_percent))</span></span>
<span id="cb2-289"><a href="#cb2-289"></a></span>
<span id="cb2-290"><a href="#cb2-290"></a></span>
<span id="cb2-291"><a href="#cb2-291"></a><span class="co"># MAD thresholds that were chosen for low lib size and features</span></span>
<span id="cb2-292"><a href="#cb2-292"></a></span>
<span id="cb2-293"><a href="#cb2-293"></a><span class="co"># low_lib_size &lt;- isOutlier(sce$sum, log=TRUE, type="lower")</span></span>
<span id="cb2-294"><a href="#cb2-294"></a><span class="co"># table(low_lib_size)</span></span>
<span id="cb2-295"><a href="#cb2-295"></a><span class="fu">attr</span>(sce<span class="sc">$</span>low_lib_size, <span class="st">"thresholds"</span>)[<span class="dv">1</span>]</span>
<span id="cb2-296"><a href="#cb2-296"></a><span class="co"># colData(sce)$low_lib_size &lt;- low_lib_size</span></span>
<span id="cb2-297"><a href="#cb2-297"></a></span>
<span id="cb2-298"><a href="#cb2-298"></a><span class="co"># low_n_features &lt;- isOutlier(sce$detected, log=TRUE, type="lower")</span></span>
<span id="cb2-299"><a href="#cb2-299"></a><span class="co"># table(low_n_features)</span></span>
<span id="cb2-300"><a href="#cb2-300"></a><span class="fu">attr</span>(sce<span class="sc">$</span>low_n_features, <span class="st">"thresholds"</span>)[<span class="dv">1</span>]</span>
<span id="cb2-301"><a href="#cb2-301"></a><span class="co"># colData(sce)$low_n_features &lt;- low_n_features</span></span>
<span id="cb2-302"><a href="#cb2-302"></a></span>
<span id="cb2-303"><a href="#cb2-303"></a><span class="co"># QC plots per sample </span></span>
<span id="cb2-304"><a href="#cb2-304"></a>gridExtra<span class="sc">::</span><span class="fu">grid.arrange</span>(</span>
<span id="cb2-305"><a href="#cb2-305"></a>  <span class="fu">plotColData</span>(sce, </span>
<span id="cb2-306"><a href="#cb2-306"></a>              <span class="at">x=</span><span class="st">"sample"</span>, </span>
<span id="cb2-307"><a href="#cb2-307"></a>              <span class="at">y=</span><span class="st">"sum"</span>,</span>
<span id="cb2-308"><a href="#cb2-308"></a>              <span class="at">colour_by =</span> <span class="st">"low_lib_size"</span>) <span class="sc">+</span> </span>
<span id="cb2-309"><a href="#cb2-309"></a>    <span class="fu">scale_y_log10</span>() <span class="sc">+</span> </span>
<span id="cb2-310"><a href="#cb2-310"></a>    <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Total count"</span>, <span class="at">title =</span> <span class="st">"Total count"</span>) <span class="sc">+</span></span>
<span id="cb2-311"><a href="#cb2-311"></a>    <span class="fu">guides</span>(<span class="at">colour=</span><span class="fu">guide_legend</span>(<span class="at">title=</span><span class="st">"Discarded"</span>)),</span>
<span id="cb2-312"><a href="#cb2-312"></a>  </span>
<span id="cb2-313"><a href="#cb2-313"></a>  <span class="fu">plotColData</span>(sce, </span>
<span id="cb2-314"><a href="#cb2-314"></a>              <span class="at">x=</span><span class="st">"sample"</span>, </span>
<span id="cb2-315"><a href="#cb2-315"></a>              <span class="at">y=</span><span class="st">"detected"</span>,</span>
<span id="cb2-316"><a href="#cb2-316"></a>              <span class="at">colour_by =</span> <span class="st">"low_n_features"</span>) <span class="sc">+</span> </span>
<span id="cb2-317"><a href="#cb2-317"></a>    <span class="fu">scale_y_log10</span>() <span class="sc">+</span> </span>
<span id="cb2-318"><a href="#cb2-318"></a>    <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Genes detected"</span>, <span class="at">title =</span> <span class="st">"Genes detected"</span>) <span class="sc">+</span></span>
<span id="cb2-319"><a href="#cb2-319"></a>    <span class="fu">guides</span>(<span class="at">colour=</span><span class="fu">guide_legend</span>(<span class="at">title=</span><span class="st">"Discarded"</span>)),</span>
<span id="cb2-320"><a href="#cb2-320"></a>  </span>
<span id="cb2-321"><a href="#cb2-321"></a>  <span class="fu">plotColData</span>(sce, </span>
<span id="cb2-322"><a href="#cb2-322"></a>              <span class="at">x=</span><span class="st">"sum"</span>, </span>
<span id="cb2-323"><a href="#cb2-323"></a>              <span class="at">y=</span><span class="st">"subsets_Mito_percent"</span>, </span>
<span id="cb2-324"><a href="#cb2-324"></a>              <span class="at">colour_by=</span><span class="st">"high_Mito_percent"</span>),</span>
<span id="cb2-325"><a href="#cb2-325"></a>  <span class="at">ncol=</span><span class="dv">1</span></span>
<span id="cb2-326"><a href="#cb2-326"></a>)</span>
<span id="cb2-327"><a href="#cb2-327"></a></span>
<span id="cb2-328"><a href="#cb2-328"></a><span class="co"># sn_high_Mito_percent &lt;- isOutlier(sce$subsets_Mito_percent, type="higher", min.diff=0.5) # adjustment for snuclear</span></span>
<span id="cb2-329"><a href="#cb2-329"></a><span class="co"># # discard &lt;- Reduce("|", sce[,colnames(sce)!="discard"])</span></span>
<span id="cb2-330"><a href="#cb2-330"></a><span class="co"># # colSums(as.matrix(sce))</span></span>
<span id="cb2-331"><a href="#cb2-331"></a><span class="co"># # high_Mito_percent &lt;- isOutlier(sce$subsets_Mito_percent, type="higher")</span></span>
<span id="cb2-332"><a href="#cb2-332"></a><span class="co"># table(sn_high_Mito_percent)</span></span>
<span id="cb2-333"><a href="#cb2-333"></a><span class="fu">attr</span>(sce<span class="sc">$</span>high_Mito_percent, <span class="st">"thresholds"</span>)[<span class="dv">2</span>]</span>
<span id="cb2-334"><a href="#cb2-334"></a><span class="fu">attr</span>(sce<span class="sc">$</span>sn_high_Mito_percent, <span class="st">"thresholds"</span>)[<span class="dv">2</span>]</span>
<span id="cb2-335"><a href="#cb2-335"></a><span class="co"># colData(sce)$sn_high_Mito_percent &lt;- sn_high_Mito_percent</span></span>
<span id="cb2-336"><a href="#cb2-336"></a><span class="fu">summary</span>(sce<span class="sc">$</span>high_Mito_percent)</span>
<span id="cb2-337"><a href="#cb2-337"></a><span class="fu">summary</span>(sce<span class="sc">$</span>sn_high_Mito_percent)</span>
<span id="cb2-338"><a href="#cb2-338"></a></span>
<span id="cb2-339"><a href="#cb2-339"></a>gridExtra<span class="sc">::</span><span class="fu">grid.arrange</span>(</span>
<span id="cb2-340"><a href="#cb2-340"></a><span class="fu">plotColData</span>(sce,  </span>
<span id="cb2-341"><a href="#cb2-341"></a>            <span class="at">x=</span><span class="st">"sample"</span>,</span>
<span id="cb2-342"><a href="#cb2-342"></a>            <span class="at">y=</span><span class="st">"subsets_Mito_percent"</span>,</span>
<span id="cb2-343"><a href="#cb2-343"></a>            <span class="at">colour_by =</span> <span class="st">"sn_high_Mito_percent"</span>) <span class="sc">+</span> </span>
<span id="cb2-344"><a href="#cb2-344"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Percentage mitochondrial UMIs"</span>,</span>
<span id="cb2-345"><a href="#cb2-345"></a>       <span class="at">title =</span> <span class="st">"Mitochondrial UMIs"</span>) <span class="sc">+</span></span>
<span id="cb2-346"><a href="#cb2-346"></a>  <span class="fu">guides</span>(<span class="at">colour=</span><span class="fu">guide_legend</span>(<span class="at">title=</span><span class="st">"Discarded"</span>)),</span>
<span id="cb2-347"><a href="#cb2-347"></a></span>
<span id="cb2-348"><a href="#cb2-348"></a><span class="fu">plotColData</span>(sce, </span>
<span id="cb2-349"><a href="#cb2-349"></a>            <span class="at">x=</span><span class="st">"sum"</span>, </span>
<span id="cb2-350"><a href="#cb2-350"></a>            <span class="at">y=</span><span class="st">"subsets_Mito_percent"</span>, </span>
<span id="cb2-351"><a href="#cb2-351"></a>            <span class="at">colour_by=</span><span class="st">"sn_high_Mito_percent"</span>),</span>
<span id="cb2-352"><a href="#cb2-352"></a><span class="at">ncol=</span><span class="dv">2</span></span>
<span id="cb2-353"><a href="#cb2-353"></a>)</span>
<span id="cb2-354"><a href="#cb2-354"></a></span>
<span id="cb2-355"><a href="#cb2-355"></a>mito_thresh <span class="ot">&lt;-</span> <span class="fl">4.5</span></span>
<span id="cb2-356"><a href="#cb2-356"></a></span>
<span id="cb2-357"><a href="#cb2-357"></a>gridExtra<span class="sc">::</span><span class="fu">grid.arrange</span>(</span>
<span id="cb2-358"><a href="#cb2-358"></a>  <span class="co"># Histogram subsets mito percent</span></span>
<span id="cb2-359"><a href="#cb2-359"></a>  <span class="fu">colData</span>(sce) <span class="sc">%&gt;%</span> </span>
<span id="cb2-360"><a href="#cb2-360"></a>    <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb2-361"><a href="#cb2-361"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>subsets_Mito_percent)) <span class="sc">+</span> </span>
<span id="cb2-362"><a href="#cb2-362"></a>    <span class="fu">geom_density</span>(<span class="at">alpha=</span><span class="fl">0.2</span>, <span class="at">fill=</span><span class="st">"#00BFC4"</span>, <span class="at">colour=</span><span class="st">"#00BFC4"</span>) <span class="sc">+</span></span>
<span id="cb2-363"><a href="#cb2-363"></a>    <span class="fu">scale_x_log10</span>() <span class="sc">+</span> </span>
<span id="cb2-364"><a href="#cb2-364"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fl">0.2</span>) <span class="sc">+</span></span>
<span id="cb2-365"><a href="#cb2-365"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept=</span>mito_thresh, <span class="at">color=</span><span class="st">"#F8766D"</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>),</span>
<span id="cb2-366"><a href="#cb2-366"></a>  </span>
<span id="cb2-367"><a href="#cb2-367"></a>  <span class="co"># Visualize the distribution of mitochondrial gene expression detected per cell</span></span>
<span id="cb2-368"><a href="#cb2-368"></a>  <span class="fu">colData</span>(sce) <span class="sc">%&gt;%</span> </span>
<span id="cb2-369"><a href="#cb2-369"></a>    <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb2-370"><a href="#cb2-370"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">color=</span>sample, <span class="at">x=</span>subsets_Mito_percent, <span class="at">fill=</span>sample)) <span class="sc">+</span> </span>
<span id="cb2-371"><a href="#cb2-371"></a>    <span class="fu">geom_density</span>(<span class="at">alpha =</span> <span class="fl">0.2</span>) <span class="sc">+</span> </span>
<span id="cb2-372"><a href="#cb2-372"></a>    <span class="fu">scale_x_log10</span>() <span class="sc">+</span> </span>
<span id="cb2-373"><a href="#cb2-373"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb2-374"><a href="#cb2-374"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fl">0.2</span>) <span class="sc">+</span></span>
<span id="cb2-375"><a href="#cb2-375"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> mito_thresh, <span class="at">color=</span><span class="st">"#F8766D"</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>),</span>
<span id="cb2-376"><a href="#cb2-376"></a>    </span>
<span id="cb2-377"><a href="#cb2-377"></a>  <span class="at">ncol=</span><span class="dv">2</span> </span>
<span id="cb2-378"><a href="#cb2-378"></a>)</span>
<span id="cb2-379"><a href="#cb2-379"></a></span>
<span id="cb2-380"><a href="#cb2-380"></a>gridExtra<span class="sc">::</span><span class="fu">grid.arrange</span>(</span>
<span id="cb2-381"><a href="#cb2-381"></a>  <span class="co"># Complexity plot</span></span>
<span id="cb2-382"><a href="#cb2-382"></a>  <span class="fu">colData</span>(sce) <span class="sc">%&gt;%</span> </span>
<span id="cb2-383"><a href="#cb2-383"></a>    <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb2-384"><a href="#cb2-384"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>sum, <span class="at">y=</span>detected, <span class="at">color=</span>subsets_Mito_percent)) <span class="sc">+</span> </span>
<span id="cb2-385"><a href="#cb2-385"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb2-386"><a href="#cb2-386"></a>    <span class="fu">scale_colour_gradient</span>(<span class="at">low =</span> <span class="st">"gray90"</span>, <span class="at">high =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb2-387"><a href="#cb2-387"></a>    <span class="fu">stat_smooth</span>(<span class="at">method=</span>lm) <span class="sc">+</span></span>
<span id="cb2-388"><a href="#cb2-388"></a>    <span class="fu">scale_x_log10</span>() <span class="sc">+</span> </span>
<span id="cb2-389"><a href="#cb2-389"></a>    <span class="fu">scale_y_log10</span>() <span class="sc">+</span> </span>
<span id="cb2-390"><a href="#cb2-390"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb2-391"><a href="#cb2-391"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">500</span>) <span class="sc">+</span></span>
<span id="cb2-392"><a href="#cb2-392"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">250</span>) ,</span>
<span id="cb2-393"><a href="#cb2-393"></a>  <span class="co"># facet_wrap(~sample)</span></span>
<span id="cb2-394"><a href="#cb2-394"></a></span>
<span id="cb2-395"><a href="#cb2-395"></a>  <span class="fu">plotColData</span>(sce, <span class="at">x=</span><span class="st">"sum"</span>, <span class="at">y=</span><span class="st">"detected"</span>, <span class="at">colour_by=</span><span class="st">"adjusted_discard"</span>),</span>
<span id="cb2-396"><a href="#cb2-396"></a>  <span class="at">ncol=</span><span class="dv">2</span></span>
<span id="cb2-397"><a href="#cb2-397"></a>)</span>
<span id="cb2-398"><a href="#cb2-398"></a></span>
<span id="cb2-399"><a href="#cb2-399"></a><span class="co"># adjusting mito threshold</span></span>
<span id="cb2-400"><a href="#cb2-400"></a>sce<span class="sc">$</span>adjusted_mito_threshold <span class="ot">&lt;-</span> </span>
<span id="cb2-401"><a href="#cb2-401"></a>  <span class="fu">data.frame</span>(<span class="at">hmp =</span> sce<span class="sc">$</span>subsets_Mito_percent <span class="sc">&gt;=</span> mito_thresh) <span class="sc">%&gt;%</span></span>
<span id="cb2-402"><a href="#cb2-402"></a>  <span class="fu">pull</span>(hmp)</span>
<span id="cb2-403"><a href="#cb2-403"></a></span>
<span id="cb2-404"><a href="#cb2-404"></a>sce<span class="sc">$</span>adjusted_discard <span class="ot">&lt;-</span> </span>
<span id="cb2-405"><a href="#cb2-405"></a>  <span class="fu">data.frame</span>(<span class="at">lls =</span> sce<span class="sc">$</span>low_lib_size,</span>
<span id="cb2-406"><a href="#cb2-406"></a>             <span class="at">lnf =</span> sce<span class="sc">$</span>low_n_features,</span>
<span id="cb2-407"><a href="#cb2-407"></a>             <span class="at">hmp =</span> sce<span class="sc">$</span>adjusted_mito_threshold) <span class="sc">%&gt;%</span></span>
<span id="cb2-408"><a href="#cb2-408"></a>  <span class="fu">mutate</span>(<span class="at">adjusted_discard =</span> <span class="fu">case_when</span>(</span>
<span id="cb2-409"><a href="#cb2-409"></a>    <span class="fu">if_any</span>(<span class="fu">everything</span>(), <span class="sc">~</span>. <span class="sc">==</span> <span class="cn">TRUE</span>) <span class="sc">~</span> <span class="cn">TRUE</span>, </span>
<span id="cb2-410"><a href="#cb2-410"></a>    <span class="at">.default=</span><span class="cn">FALSE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb2-411"><a href="#cb2-411"></a>  <span class="fu">pull</span>(adjusted_discard)</span>
<span id="cb2-412"><a href="#cb2-412"></a></span>
<span id="cb2-413"><a href="#cb2-413"></a><span class="co"># Checking effect of changed threshold</span></span>
<span id="cb2-414"><a href="#cb2-414"></a><span class="fu">table</span>(<span class="at">discard=</span>sce<span class="sc">$</span>discard, <span class="at">adjusted_discard=</span>sce<span class="sc">$</span>adjusted_discard)</span>
<span id="cb2-415"><a href="#cb2-415"></a><span class="do">## in this case simply reduced the nbumber of discarded to 41</span></span>
<span id="cb2-416"><a href="#cb2-416"></a></span>
<span id="cb2-417"><a href="#cb2-417"></a><span class="co"># colData(sce)$sn_high_Mito_percent &lt;- sn_high_Mito_percent</span></span>
<span id="cb2-418"><a href="#cb2-418"></a>gridExtra<span class="sc">::</span><span class="fu">grid.arrange</span>(</span>
<span id="cb2-419"><a href="#cb2-419"></a>  <span class="fu">plotColData</span>(sce,  </span>
<span id="cb2-420"><a href="#cb2-420"></a>              <span class="at">x=</span><span class="st">"sample"</span>,</span>
<span id="cb2-421"><a href="#cb2-421"></a>              <span class="at">y=</span><span class="st">"subsets_Mito_percent"</span>,</span>
<span id="cb2-422"><a href="#cb2-422"></a>              <span class="at">colour_by =</span> <span class="st">"adjusted_discard"</span>) <span class="sc">+</span> </span>
<span id="cb2-423"><a href="#cb2-423"></a>    <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Percentage mitochondrial UMIs"</span>,</span>
<span id="cb2-424"><a href="#cb2-424"></a>        <span class="at">title =</span> <span class="st">"Mitochondrial UMIs"</span>) <span class="sc">+</span></span>
<span id="cb2-425"><a href="#cb2-425"></a>    <span class="fu">guides</span>(<span class="at">colour=</span><span class="fu">guide_legend</span>(<span class="at">title=</span><span class="st">"Discarded"</span>)),</span>
<span id="cb2-426"><a href="#cb2-426"></a></span>
<span id="cb2-427"><a href="#cb2-427"></a>  <span class="fu">plotColData</span>(sce, </span>
<span id="cb2-428"><a href="#cb2-428"></a>              <span class="at">x=</span><span class="st">"sum"</span>, </span>
<span id="cb2-429"><a href="#cb2-429"></a>              <span class="at">y=</span><span class="st">"subsets_Mito_percent"</span>, </span>
<span id="cb2-430"><a href="#cb2-430"></a>              <span class="at">colour_by=</span><span class="st">"adjusted_discard"</span>),</span>
<span id="cb2-431"><a href="#cb2-431"></a>  <span class="at">ncol=</span><span class="dv">2</span></span>
<span id="cb2-432"><a href="#cb2-432"></a>)</span>
<span id="cb2-433"><a href="#cb2-433"></a></span>
<span id="cb2-434"><a href="#cb2-434"></a><span class="fu">summary</span>(sce<span class="sc">$</span>high_Mito_percent)</span>
<span id="cb2-435"><a href="#cb2-435"></a><span class="fu">summary</span>(sce<span class="sc">$</span>sn_high_Mito_percent)</span>
<span id="cb2-436"><a href="#cb2-436"></a><span class="fu">summary</span>(sce<span class="sc">$</span>adjusted_mito_threshold)</span>
<span id="cb2-437"><a href="#cb2-437"></a></span>
<span id="cb2-438"><a href="#cb2-438"></a><span class="co"># Final removed for mito</span></span>
<span id="cb2-439"><a href="#cb2-439"></a><span class="fu">colData</span>(sce) <span class="sc">%&gt;%</span></span>
<span id="cb2-440"><a href="#cb2-440"></a><span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb2-441"><a href="#cb2-441"></a><span class="fu">filter</span>(adjusted_mito_threshold <span class="sc">==</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-442"><a href="#cb2-442"></a><span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb2-443"><a href="#cb2-443"></a><span class="fu">summarise</span>(<span class="at">cells=</span><span class="fu">n</span>()) <span class="sc">%T&gt;%</span></span>
<span id="cb2-444"><a href="#cb2-444"></a>readr<span class="sc">::</span><span class="fu">write_excel_csv</span>(<span class="st">"results/adjustedmito_per_sample.csv"</span>, <span class="at">col_names=</span>T)</span>
<span id="cb2-445"><a href="#cb2-445"></a></span>
<span id="cb2-446"><a href="#cb2-446"></a><span class="fu">saveRDS</span>(sce, <span class="st">"data/post_mit_thresh.rds"</span>)</span>
<span id="cb2-447"><a href="#cb2-447"></a></span>
<span id="cb2-448"><a href="#cb2-448"></a><span class="co"># sce.filtered &lt;- sce[, !sce$discard]</span></span>
<span id="cb2-449"><a href="#cb2-449"></a>sce.filtered <span class="ot">&lt;-</span> sce[, <span class="sc">!</span>sce<span class="sc">$</span>adjusted_discard]</span>
<span id="cb2-450"><a href="#cb2-450"></a><span class="fu">colData</span>(sce.filtered) <span class="ot">&lt;-</span> <span class="fu">colData</span>(sce.filtered)[,<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>)]</span>
<span id="cb2-451"><a href="#cb2-451"></a>sce.filtered <span class="ot">&lt;-</span> <span class="fu">addPerCellQC</span>(sce.filtered)</span>
<span id="cb2-452"><a href="#cb2-452"></a><span class="fu">colnames</span>(<span class="fu">colData</span>(sce.filtered))</span>
<span id="cb2-453"><a href="#cb2-453"></a></span>
<span id="cb2-454"><a href="#cb2-454"></a><span class="fu">dim</span>(sce)</span>
<span id="cb2-455"><a href="#cb2-455"></a><span class="fu">dim</span>(sce.filtered)</span>
<span id="cb2-456"><a href="#cb2-456"></a><span class="co"># saveRDS(sce.filtered, "data/sce_filtered.rds")</span></span>
<span id="cb2-457"><a href="#cb2-457"></a>sce.filtered <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"data/sce_filtered.rds"</span>)</span>
<span id="cb2-458"><a href="#cb2-458"></a></span>
<span id="cb2-459"><a href="#cb2-459"></a><span class="co"># the cell sparsity: for each cell, the proportion of genes that are not detected</span></span>
<span id="cb2-460"><a href="#cb2-460"></a><span class="co"># the gene sparsity: for each gene, the proportion of cells in which it is not detected</span></span>
<span id="cb2-461"><a href="#cb2-461"></a></span>
<span id="cb2-462"><a href="#cb2-462"></a><span class="co"># # Sparsity - perhaps a bit harsh</span></span>
<span id="cb2-463"><a href="#cb2-463"></a><span class="co"># sce_sparse &lt;- addPerFeatureQC(sce.filtered)</span></span>
<span id="cb2-464"><a href="#cb2-464"></a><span class="co"># rowData(sce_sparse)</span></span>
<span id="cb2-465"><a href="#cb2-465"></a></span>
<span id="cb2-466"><a href="#cb2-466"></a><span class="co"># colData(sce_sparse)$cell_sparsity &lt;- 1 - (colData(sce_sparse)$detected / nrow(sce_sparse))</span></span>
<span id="cb2-467"><a href="#cb2-467"></a><span class="co"># rowData(sce_sparse)$gene_sparsity &lt;- (100 - rowData(sce_sparse)$detected) / 100</span></span>
<span id="cb2-468"><a href="#cb2-468"></a></span>
<span id="cb2-469"><a href="#cb2-469"></a><span class="co"># hist(sce_sparse$cell_sparsity, breaks=50, col="grey80", xlab="Cell sparsity", main="")</span></span>
<span id="cb2-470"><a href="#cb2-470"></a><span class="co"># hist(rowData(sce_sparse)$gene_sparsity, breaks=50, col="grey80", xlab="Gene sparsity", main="")</span></span>
<span id="cb2-471"><a href="#cb2-471"></a></span>
<span id="cb2-472"><a href="#cb2-472"></a><span class="co"># sce_sparse$sparse_cells &lt;- sce_sparse$cell_sparsity &gt; 0.99</span></span>
<span id="cb2-473"><a href="#cb2-473"></a><span class="co"># table(sce_sparse$sparse_cells)</span></span>
<span id="cb2-474"><a href="#cb2-474"></a></span>
<span id="cb2-475"><a href="#cb2-475"></a><span class="co"># min_cells &lt;- 1 - (10 / ncol(sce_sparse))</span></span>
<span id="cb2-476"><a href="#cb2-476"></a><span class="co"># rowData(sce_sparse)$sparse_genes &lt;- rowData(sce_sparse)$gene_sparsity &gt; min_cells</span></span>
<span id="cb2-477"><a href="#cb2-477"></a><span class="co"># table(rowData(sce_sparse)$sparse_genes)</span></span>
<span id="cb2-478"><a href="#cb2-478"></a></span>
<span id="cb2-479"><a href="#cb2-479"></a><span class="co"># sce_final &lt;- sce_sparse[, !sce_sparse$sparse_cells]</span></span>
<span id="cb2-480"><a href="#cb2-480"></a><span class="co"># dim(sce_final)</span></span>
<span id="cb2-481"><a href="#cb2-481"></a><span class="co"># sce_final &lt;- sce_final[!rowData(sce_final)$sparse_genes, ]</span></span>
<span id="cb2-482"><a href="#cb2-482"></a><span class="co"># dim(sce_final)</span></span>
<span id="cb2-483"><a href="#cb2-483"></a></span>
<span id="cb2-484"><a href="#cb2-484"></a><span class="fu">saveRDS</span>(sce_final, <span class="st">"post_sceqc_object.rds"</span>)</span>
<span id="cb2-485"><a href="#cb2-485"></a></span>
<span id="cb2-486"><a href="#cb2-486"></a><span class="co"># Post-qc seurat ----------------------------------------------------------</span></span>
<span id="cb2-487"><a href="#cb2-487"></a></span>
<span id="cb2-488"><a href="#cb2-488"></a><span class="fu">table</span>(Matrix<span class="sc">::</span><span class="fu">rowSums</span>(<span class="fu">counts</span>(sce.filtered) <span class="sc">&gt;=</span> <span class="dv">1</span>) <span class="sc">&gt;=</span><span class="dv">3</span>)</span>
<span id="cb2-489"><a href="#cb2-489"></a>keep <span class="ot">&lt;-</span> Matrix<span class="sc">::</span><span class="fu">rowSums</span>(<span class="fu">counts</span>(sce.filtered) <span class="sc">&gt;=</span> <span class="dv">1</span>) <span class="sc">&gt;=</span><span class="dv">3</span></span>
<span id="cb2-490"><a href="#cb2-490"></a>final_sce <span class="ot">&lt;-</span> sce.filtered[keep,]</span>
<span id="cb2-491"><a href="#cb2-491"></a>chosen_sce <span class="ot">&lt;-</span> final_sce</span>
<span id="cb2-492"><a href="#cb2-492"></a></span>
<span id="cb2-493"><a href="#cb2-493"></a>preqc <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(<span class="at">counts=</span><span class="fu">counts</span>(chosen_sce),  <span class="co"># change this threshold is too high </span></span>
<span id="cb2-494"><a href="#cb2-494"></a>                            <span class="at">meta.data=</span><span class="fu">as.data.frame</span>(<span class="fu">colData</span>(chosen_sce)),</span>
<span id="cb2-495"><a href="#cb2-495"></a>                            <span class="at">min.cells =</span> <span class="dv">3</span>, </span>
<span id="cb2-496"><a href="#cb2-496"></a>                            <span class="at">min.features  =</span> <span class="dv">200</span>, </span>
<span id="cb2-497"><a href="#cb2-497"></a>                            <span class="at">project =</span> <span class="st">"whole_muscle_myob"</span>, </span>
<span id="cb2-498"><a href="#cb2-498"></a>                            <span class="at">assay =</span> <span class="st">"RNA"</span>)</span>
<span id="cb2-499"><a href="#cb2-499"></a></span>
<span id="cb2-500"><a href="#cb2-500"></a><span class="co"># Making sure rownames of seurat match most recent sce</span></span>
<span id="cb2-501"><a href="#cb2-501"></a>a <span class="ot">&lt;-</span> <span class="fu">rownames</span>(chosen_sce)</span>
<span id="cb2-502"><a href="#cb2-502"></a>b <span class="ot">&lt;-</span> <span class="fu">rownames</span>(preqc)</span>
<span id="cb2-503"><a href="#cb2-503"></a>old <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(a, b)</span>
<span id="cb2-504"><a href="#cb2-504"></a>new <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(b, a)</span>
<span id="cb2-505"><a href="#cb2-505"></a></span>
<span id="cb2-506"><a href="#cb2-506"></a><span class="fu">table</span>(old <span class="sc">%in%</span> <span class="fu">rownames</span>(chosen_sce))</span>
<span id="cb2-507"><a href="#cb2-507"></a><span class="fu">table</span>(old <span class="sc">%in%</span> <span class="fu">rownames</span>(preqc))</span>
<span id="cb2-508"><a href="#cb2-508"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">seq_along</span>(old)) {</span>
<span id="cb2-509"><a href="#cb2-509"></a>  <span class="fu">rownames</span>(chosen_sce) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(old[i], new[i], <span class="fu">rownames</span>(chosen_sce))</span>
<span id="cb2-510"><a href="#cb2-510"></a>}</span>
<span id="cb2-511"><a href="#cb2-511"></a><span class="fu">table</span>(new <span class="sc">%in%</span> <span class="fu">rownames</span>(chosen_sce))</span>
<span id="cb2-512"><a href="#cb2-512"></a><span class="fu">table</span>(new <span class="sc">%in%</span> <span class="fu">rownames</span>(preqc))</span>
<span id="cb2-513"><a href="#cb2-513"></a></span>
<span id="cb2-514"><a href="#cb2-514"></a>preqc[[<span class="st">"percent.mt"</span>]] <span class="ot">&lt;-</span> <span class="fu">PercentageFeatureSet</span>(preqc, <span class="at">pattern =</span> <span class="st">"^MT-"</span>)</span>
<span id="cb2-515"><a href="#cb2-515"></a>preqc[[<span class="st">"percent.rb"</span>]] <span class="ot">&lt;-</span> <span class="fu">PercentageFeatureSet</span>(preqc, <span class="at">pattern =</span> <span class="st">"^RP[SL]"</span>)</span>
<span id="cb2-516"><a href="#cb2-516"></a><span class="fu">VlnPlot</span>(preqc, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"nFeature_RNA"</span>,<span class="st">"nCount_RNA"</span>,<span class="st">"percent.mt"</span>,<span class="st">"percent.rb"</span>),<span class="at">ncol =</span> <span class="dv">4</span>,<span class="at">pt.size =</span> <span class="fl">0.1</span>) <span class="sc">&amp;</span> </span>
<span id="cb2-517"><a href="#cb2-517"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>))</span>
<span id="cb2-518"><a href="#cb2-518"></a>plot_1 <span class="ot">&lt;-</span> <span class="fu">FeatureScatter</span>(preqc, <span class="at">feature1 =</span> <span class="st">"nCount_RNA"</span>, <span class="at">feature2 =</span> <span class="st">"percent.mt"</span>)</span>
<span id="cb2-519"><a href="#cb2-519"></a>plot_2 <span class="ot">&lt;-</span> <span class="fu">FeatureScatter</span>(preqc, <span class="at">feature1 =</span> <span class="st">"nCount_RNA"</span>, <span class="at">feature2 =</span> <span class="st">"nFeature_RNA"</span>)</span>
<span id="cb2-520"><a href="#cb2-520"></a>plot_3 <span class="ot">&lt;-</span> <span class="fu">FeatureScatter</span>(preqc, <span class="at">feature1 =</span> <span class="st">"nCount_RNA"</span>, <span class="at">feature2 =</span> <span class="st">"percent.rb"</span>)</span>
<span id="cb2-521"><a href="#cb2-521"></a>plot_4 <span class="ot">&lt;-</span> <span class="fu">FeatureScatter</span>(preqc, <span class="at">feature1 =</span> <span class="st">"percent.rb"</span>, <span class="at">feature2 =</span> <span class="st">"percent.mt"</span>)</span>
<span id="cb2-522"><a href="#cb2-522"></a></span>
<span id="cb2-523"><a href="#cb2-523"></a>plot_1 <span class="sc">+</span> plot_2</span>
<span id="cb2-524"><a href="#cb2-524"></a>plot_3 <span class="sc">+</span> plot_4</span>
<span id="cb2-525"><a href="#cb2-525"></a></span>
<span id="cb2-526"><a href="#cb2-526"></a>postqc <span class="ot">&lt;-</span> <span class="fu">subset</span>(preqc, </span>
<span id="cb2-527"><a href="#cb2-527"></a>                <span class="at">subset =</span> nFeature_RNA <span class="sc">&gt;</span> <span class="dv">200</span> <span class="sc">&amp;</span> </span>
<span id="cb2-528"><a href="#cb2-528"></a>                nFeature_RNA <span class="sc">&lt;</span> <span class="dv">2350</span> <span class="sc">&amp;</span> </span>
<span id="cb2-529"><a href="#cb2-529"></a>                percent.mt <span class="sc">&lt;</span> <span class="dv">5</span> <span class="sc">&amp;</span></span>
<span id="cb2-530"><a href="#cb2-530"></a>                percent.rb <span class="sc">&lt;</span> <span class="dv">3</span>)</span>
<span id="cb2-531"><a href="#cb2-531"></a><span class="fu">VlnPlot</span>(postqc, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"nFeature_RNA"</span>,<span class="st">"nCount_RNA"</span>,<span class="st">"percent.mt"</span>,<span class="st">"percent.rb"</span>),<span class="at">ncol =</span> <span class="dv">4</span>,<span class="at">pt.size =</span> <span class="fl">0.1</span>) <span class="sc">&amp;</span> </span>
<span id="cb2-532"><a href="#cb2-532"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>))</span>
<span id="cb2-533"><a href="#cb2-533"></a></span>
<span id="cb2-534"><a href="#cb2-534"></a><span class="co"># Processing --------------------------------------------------------------</span></span>
<span id="cb2-535"><a href="#cb2-535"></a></span>
<span id="cb2-536"><a href="#cb2-536"></a><span class="co"># plan(strategy = "multicore", workers = 30)</span></span>
<span id="cb2-537"><a href="#cb2-537"></a></span>
<span id="cb2-538"><a href="#cb2-538"></a>seur_norm <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(postqc)</span>
<span id="cb2-539"><a href="#cb2-539"></a>seur_norm <span class="ot">&lt;-</span> <span class="fu">FindVariableFeatures</span>(seur_norm, <span class="at">selection.method =</span> <span class="st">"vst"</span>)</span>
<span id="cb2-540"><a href="#cb2-540"></a>seur_norm <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(seur_norm, <span class="at">features =</span> <span class="fu">rownames</span>(seur_norm))</span>
<span id="cb2-541"><a href="#cb2-541"></a>s.genes <span class="ot">&lt;-</span> cc.genes<span class="sc">$</span>s.genes</span>
<span id="cb2-542"><a href="#cb2-542"></a>g2m.genes <span class="ot">&lt;-</span> cc.genes<span class="sc">$</span>g2m.genes</span>
<span id="cb2-543"><a href="#cb2-543"></a>seur_norm <span class="ot">&lt;-</span> <span class="fu">CellCycleScoring</span>(</span>
<span id="cb2-544"><a href="#cb2-544"></a>  seur_norm,</span>
<span id="cb2-545"><a href="#cb2-545"></a>  <span class="at">s.features =</span> s.genes,</span>
<span id="cb2-546"><a href="#cb2-546"></a>  <span class="at">g2m.features =</span> g2m.genes,</span>
<span id="cb2-547"><a href="#cb2-547"></a>  <span class="at">set.ident =</span> <span class="cn">TRUE</span></span>
<span id="cb2-548"><a href="#cb2-548"></a>)</span>
<span id="cb2-549"><a href="#cb2-549"></a>seur_norm<span class="sc">$</span>CC.Difference <span class="ot">&lt;-</span> seur_norm<span class="sc">$</span>S.Score <span class="sc">-</span> seur_norm<span class="sc">$</span>G2M.Score</span>
<span id="cb2-550"><a href="#cb2-550"></a>seur_scaled <span class="ot">&lt;-</span> </span>
<span id="cb2-551"><a href="#cb2-551"></a>  <span class="fu">SCTransform</span>(seur_norm, </span>
<span id="cb2-552"><a href="#cb2-552"></a>            <span class="at">vst.flavor =</span> <span class="st">"v2"</span>,</span>
<span id="cb2-553"><a href="#cb2-553"></a>            <span class="at">vars.to.regress =</span> <span class="fu">c</span>(<span class="st">"percent.mt"</span>, </span>
<span id="cb2-554"><a href="#cb2-554"></a>                                <span class="st">"nFeature_RNA"</span>, </span>
<span id="cb2-555"><a href="#cb2-555"></a>                                <span class="st">"CC.Difference"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb2-556"><a href="#cb2-556"></a>  <span class="fu">RunPCA</span>(<span class="at">npcs =</span> <span class="dv">50</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-557"><a href="#cb2-557"></a>  <span class="fu">RunUMAP</span>(<span class="at">reduction =</span> <span class="st">"pca"</span>, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-558"><a href="#cb2-558"></a>  <span class="fu">FindNeighbors</span>(<span class="at">reduction =</span> <span class="st">"pca"</span>, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb2-559"><a href="#cb2-559"></a>  <span class="fu">FindClusters</span>(<span class="at">verbose =</span> <span class="cn">TRUE</span>, <span class="at">resolution=</span><span class="fu">seq</span>(<span class="dv">0</span>, <span class="fl">1.2</span>, <span class="fl">0.1</span>))</span>
<span id="cb2-560"><a href="#cb2-560"></a></span>
<span id="cb2-561"><a href="#cb2-561"></a>cell_plots <span class="ot">&lt;-</span> <span class="cf">function</span>(dataset, reduction, grouping){</span>
<span id="cb2-562"><a href="#cb2-562"></a>  <span class="fu">DimPlot</span>(dataset, <span class="at">reduction =</span> reduction, <span class="at">group.by =</span> grouping, <span class="at">label =</span> T)</span>
<span id="cb2-563"><a href="#cb2-563"></a>}</span>
<span id="cb2-564"><a href="#cb2-564"></a></span>
<span id="cb2-565"><a href="#cb2-565"></a>plots_umap <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb2-566"><a href="#cb2-566"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">paste0</span>(<span class="st">"SCT_snn_res."</span>, <span class="fu">seq</span>(<span class="dv">0</span>, <span class="fl">1.2</span>, <span class="fl">0.1</span>))){</span>
<span id="cb2-567"><a href="#cb2-567"></a>    plots_umap[[i]] <span class="ot">&lt;-</span> <span class="fu">cell_plots</span>(seur_scaled, <span class="at">reduction =</span> <span class="st">"umap"</span>, i)</span>
<span id="cb2-568"><a href="#cb2-568"></a>}</span>
<span id="cb2-569"><a href="#cb2-569"></a><span class="co"># plots_tsne &lt;- list()</span></span>
<span id="cb2-570"><a href="#cb2-570"></a><span class="co">#   for(i in paste0("SCT_snn_res.", seq(0, 1.2, 0.1))){</span></span>
<span id="cb2-571"><a href="#cb2-571"></a><span class="co">#     plots_tsne[[i]] &lt;- cell_plots(post_sct, reduction = "tsne", i)</span></span>
<span id="cb2-572"><a href="#cb2-572"></a><span class="co"># }</span></span>
<span id="cb2-573"><a href="#cb2-573"></a></span>
<span id="cb2-574"><a href="#cb2-574"></a><span class="fu">png</span>(<span class="st">"clustree_diagram.png"</span>, <span class="at">width =</span> <span class="dv">1000</span>, <span class="at">height =</span> <span class="dv">1000</span>)</span>
<span id="cb2-575"><a href="#cb2-575"></a>  <span class="fu">clustree</span>(seur_scaled, <span class="at">prefix =</span> <span class="st">"SCT_snn_res."</span>) <span class="sc">+</span>  <span class="co"># chose 0.6</span></span>
<span id="cb2-576"><a href="#cb2-576"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>)</span>
<span id="cb2-577"><a href="#cb2-577"></a>  <span class="co"># clustree(master, prefix = "RNA_snn_res.", node_colour = "sc3_stability")</span></span>
<span id="cb2-578"><a href="#cb2-578"></a><span class="fu">dev.off</span>()</span>
<span id="cb2-579"><a href="#cb2-579"></a></span>
<span id="cb2-580"><a href="#cb2-580"></a>plots_umap[[<span class="st">"SCT_snn_res.0"</span>  ]] </span>
<span id="cb2-581"><a href="#cb2-581"></a>plots_umap[[<span class="st">"SCT_snn_res.0.1"</span>]] </span>
<span id="cb2-582"><a href="#cb2-582"></a>plots_umap[[<span class="st">"SCT_snn_res.0.2"</span>]] </span>
<span id="cb2-583"><a href="#cb2-583"></a>plots_umap[[<span class="st">"SCT_snn_res.0.3"</span>]]</span>
<span id="cb2-584"><a href="#cb2-584"></a>plots_umap[[<span class="st">"SCT_snn_res.0.4"</span>]] </span>
<span id="cb2-585"><a href="#cb2-585"></a>plots_umap[[<span class="st">"SCT_snn_res.0.5"</span>]] </span>
<span id="cb2-586"><a href="#cb2-586"></a>plots_umap[[<span class="st">"SCT_snn_res.0.6"</span>]] </span>
<span id="cb2-587"><a href="#cb2-587"></a>plots_umap[[<span class="st">"SCT_snn_res.0.7"</span>]]</span>
<span id="cb2-588"><a href="#cb2-588"></a>plots_umap[[<span class="st">"SCT_snn_res.0.8"</span>]] </span>
<span id="cb2-589"><a href="#cb2-589"></a>plots_umap[[<span class="st">"SCT_snn_res.0.9"</span>]] </span>
<span id="cb2-590"><a href="#cb2-590"></a>plots_umap[[<span class="st">"SCT_snn_res.1"</span>  ]] </span>
<span id="cb2-591"><a href="#cb2-591"></a>plots_umap[[<span class="st">"SCT_snn_res.1.1"</span>]]</span>
<span id="cb2-592"><a href="#cb2-592"></a>plots_umap[[<span class="st">"SCT_snn_res.1.2"</span>]]</span>
<span id="cb2-593"><a href="#cb2-593"></a></span>
<span id="cb2-594"><a href="#cb2-594"></a><span class="fu">Idents</span>(<span class="at">object =</span> seur_scaled) <span class="ot">&lt;-</span> <span class="st">"SCT_snn_res.0.6"</span></span>
<span id="cb2-595"><a href="#cb2-595"></a></span>
<span id="cb2-596"><a href="#cb2-596"></a><span class="fu">png</span>(<span class="st">"cluster_diagram.png"</span>, <span class="at">width =</span> <span class="dv">500</span>, <span class="at">height =</span> <span class="dv">500</span>)</span>
<span id="cb2-597"><a href="#cb2-597"></a><span class="fu">DimPlot</span>(seur_scaled, <span class="at">reduction =</span> <span class="st">"umap"</span>, <span class="at">label=</span><span class="cn">TRUE</span>)</span>
<span id="cb2-598"><a href="#cb2-598"></a><span class="fu">dev.off</span>()</span>
<span id="cb2-599"><a href="#cb2-599"></a></span>
<span id="cb2-600"><a href="#cb2-600"></a><span class="fu">saveRDS</span>(seur_scaled, <span class="st">"data/seur_scaled.rds"</span>)</span>
<span id="cb2-601"><a href="#cb2-601"></a>seur_scaled <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"data/seur_scaled.rds"</span>)</span>
<span id="cb2-602"><a href="#cb2-602"></a></span>
<span id="cb2-603"><a href="#cb2-603"></a><span class="co"># Marker genes --------------------------------------------------------------</span></span>
<span id="cb2-604"><a href="#cb2-604"></a><span class="co"># Makes a big feature plot with all the heatmap </span></span>
<span id="cb2-605"><a href="#cb2-605"></a></span>
<span id="cb2-606"><a href="#cb2-606"></a><span class="co"># You need to be able to state the gene symbol exactly - list them here</span></span>
<span id="cb2-607"><a href="#cb2-607"></a>genes_of_interest <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb2-608"><a href="#cb2-608"></a>  <span class="st">"TTN"</span>,</span>
<span id="cb2-609"><a href="#cb2-609"></a>  <span class="st">"PAX7"</span>,</span>
<span id="cb2-610"><a href="#cb2-610"></a>  <span class="st">"ADIPOQ"</span>,</span>
<span id="cb2-611"><a href="#cb2-611"></a>  <span class="st">"MP2"</span>,</span>
<span id="cb2-612"><a href="#cb2-612"></a>  <span class="st">"PCAM1"</span>,</span>
<span id="cb2-613"><a href="#cb2-613"></a>  <span class="st">"MYHL1"</span>,</span>
<span id="cb2-614"><a href="#cb2-614"></a>  <span class="st">"MYH1"</span>,</span>
<span id="cb2-615"><a href="#cb2-615"></a>  <span class="st">"DCN"</span>,</span>
<span id="cb2-616"><a href="#cb2-616"></a>  <span class="st">"PTPRC"</span>,</span>
<span id="cb2-617"><a href="#cb2-617"></a>  <span class="st">"MKX"</span>, </span>
<span id="cb2-618"><a href="#cb2-618"></a>  <span class="st">"TNNT1"</span>,</span>
<span id="cb2-619"><a href="#cb2-619"></a>  <span class="st">"ANKRD1"</span>,</span>
<span id="cb2-620"><a href="#cb2-620"></a>  <span class="st">"MYOZ2"</span>,</span>
<span id="cb2-621"><a href="#cb2-621"></a>  <span class="st">"LAMA2"</span>,</span>
<span id="cb2-622"><a href="#cb2-622"></a>  <span class="st">"FBXO32"</span></span>
<span id="cb2-623"><a href="#cb2-623"></a>)</span>
<span id="cb2-624"><a href="#cb2-624"></a></span>
<span id="cb2-625"><a href="#cb2-625"></a><span class="co"># The function below wil return those with an exact match (i.e. one gene withg that name)</span></span>
<span id="cb2-626"><a href="#cb2-626"></a>affle <span class="ot">&lt;-</span> </span>
<span id="cb2-627"><a href="#cb2-627"></a>  <span class="fu">map</span>(genes_of_interest, <span class="sc">~</span>{</span>
<span id="cb2-628"><a href="#cb2-628"></a>    <span class="fu">rownames</span>(seur_scaled)[<span class="fu">grep</span>(<span class="fu">paste0</span>(<span class="st">"^"</span>, .x), <span class="fu">rownames</span>(seur_scaled), <span class="at">ignore.case=</span>T)]</span>
<span id="cb2-629"><a href="#cb2-629"></a>  }) <span class="sc">%&gt;%</span>  </span>
<span id="cb2-630"><a href="#cb2-630"></a>  <span class="fu">keep</span>(<span class="sc">~</span> <span class="fu">length</span>(.) <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb2-631"><a href="#cb2-631"></a>  <span class="fu">list_c</span>()</span>
<span id="cb2-632"><a href="#cb2-632"></a></span>
<span id="cb2-633"><a href="#cb2-633"></a><span class="fu">setdiff</span>(genes_of_interest, affle)</span>
<span id="cb2-634"><a href="#cb2-634"></a></span>
<span id="cb2-635"><a href="#cb2-635"></a><span class="co"># The rest of the genes you will have to search for in the dataset with the below line, </span></span>
<span id="cb2-636"><a href="#cb2-636"></a><span class="co"># pick the gene and put in the [previously] unknowns </span></span>
<span id="cb2-637"><a href="#cb2-637"></a><span class="fu">rownames</span>(seur_scaled)[<span class="fu">grep</span>(<span class="fu">paste0</span>(<span class="st">"^"</span>, <span class="st">"chrne"</span>), <span class="fu">rownames</span>(seur_scaled), <span class="at">ignore.case=</span>T)]</span>
<span id="cb2-638"><a href="#cb2-638"></a></span>
<span id="cb2-639"><a href="#cb2-639"></a>unknowns <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb2-640"><a href="#cb2-640"></a>  <span class="st">"TTN"</span>,</span>
<span id="cb2-641"><a href="#cb2-641"></a>  <span class="st">"ADIPOR1"</span>,</span>
<span id="cb2-642"><a href="#cb2-642"></a>  <span class="st">"ADIPOR2"</span>,</span>
<span id="cb2-643"><a href="#cb2-643"></a>  <span class="st">"MYH11"</span>,</span>
<span id="cb2-644"><a href="#cb2-644"></a>  <span class="st">"MYH1"</span></span>
<span id="cb2-645"><a href="#cb2-645"></a>)</span>
<span id="cb2-646"><a href="#cb2-646"></a></span>
<span id="cb2-647"><a href="#cb2-647"></a><span class="co"># This will put all the gene names together saved in a dot images folder</span></span>
<span id="cb2-648"><a href="#cb2-648"></a>final <span class="ot">&lt;-</span> <span class="fu">c</span>(unknowns, affle)</span>
<span id="cb2-649"><a href="#cb2-649"></a></span>
<span id="cb2-650"><a href="#cb2-650"></a><span class="co"># This will make individual heatmaps for each one</span></span>
<span id="cb2-651"><a href="#cb2-651"></a>ork <span class="ot">&lt;-</span></span>
<span id="cb2-652"><a href="#cb2-652"></a>  final <span class="sc">%&gt;%</span></span>
<span id="cb2-653"><a href="#cb2-653"></a>  <span class="fu">imap</span>( <span class="sc">~</span> <span class="fu">FeaturePlot</span>(seur_scaled, <span class="at">features =</span> .x, <span class="at">pt.size =</span> <span class="fl">0.35</span>))</span>
<span id="cb2-654"><a href="#cb2-654"></a>ork <span class="ot">&lt;-</span> <span class="fu">setNames</span>(ork, final)</span>
<span id="cb2-655"><a href="#cb2-655"></a><span class="fu">dir.create</span>(<span class="st">"dot_images"</span>)</span>
<span id="cb2-656"><a href="#cb2-656"></a>ork <span class="sc">%&gt;%</span></span>
<span id="cb2-657"><a href="#cb2-657"></a>  <span class="fu">imap</span>(<span class="sc">~</span>{</span>
<span id="cb2-658"><a href="#cb2-658"></a>    namies <span class="ot">&lt;-</span> .y</span>
<span id="cb2-659"><a href="#cb2-659"></a>    <span class="fu">print</span>(namies)</span>
<span id="cb2-660"><a href="#cb2-660"></a>    <span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="fu">paste0</span>(<span class="st">"dot_images/"</span>, namies, <span class="st">".png"</span>), </span>
<span id="cb2-661"><a href="#cb2-661"></a>           <span class="at">plot =</span> .x,</span>
<span id="cb2-662"><a href="#cb2-662"></a>           <span class="at">width =</span> <span class="dv">7</span>,</span>
<span id="cb2-663"><a href="#cb2-663"></a>           <span class="at">height =</span> <span class="dv">6</span>,</span>
<span id="cb2-664"><a href="#cb2-664"></a>           <span class="at">units =</span> <span class="st">"in"</span>)</span>
<span id="cb2-665"><a href="#cb2-665"></a>  })</span>
<span id="cb2-666"><a href="#cb2-666"></a></span>
<span id="cb2-667"><a href="#cb2-667"></a><span class="co"># This will make violin plots for each one saved in a vln images folder</span></span>
<span id="cb2-668"><a href="#cb2-668"></a>ork_vln <span class="ot">&lt;-</span></span>
<span id="cb2-669"><a href="#cb2-669"></a>  final <span class="sc">%&gt;%</span></span>
<span id="cb2-670"><a href="#cb2-670"></a>  <span class="fu">imap</span>( <span class="sc">~</span> <span class="fu">VlnPlot</span>(seur_scaled, <span class="at">features =</span> .x))</span>
<span id="cb2-671"><a href="#cb2-671"></a>ork_vln <span class="ot">&lt;-</span> <span class="fu">setNames</span>(ork_vln, final)</span>
<span id="cb2-672"><a href="#cb2-672"></a><span class="fu">dir.create</span>(<span class="st">"vln_images"</span>)</span>
<span id="cb2-673"><a href="#cb2-673"></a>ork_vln <span class="sc">%&gt;%</span></span>
<span id="cb2-674"><a href="#cb2-674"></a>  <span class="fu">imap</span>(<span class="sc">~</span>{</span>
<span id="cb2-675"><a href="#cb2-675"></a>    namies <span class="ot">&lt;-</span> .y</span>
<span id="cb2-676"><a href="#cb2-676"></a>    <span class="fu">print</span>(namies)</span>
<span id="cb2-677"><a href="#cb2-677"></a>    <span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="fu">paste0</span>(<span class="st">"vln_images/"</span>, namies, <span class="st">".png"</span>), </span>
<span id="cb2-678"><a href="#cb2-678"></a>           <span class="at">plot =</span> .x,</span>
<span id="cb2-679"><a href="#cb2-679"></a>           <span class="at">width =</span> <span class="dv">7</span>,</span>
<span id="cb2-680"><a href="#cb2-680"></a>           <span class="at">height =</span> <span class="dv">6</span>,</span>
<span id="cb2-681"><a href="#cb2-681"></a>           <span class="at">units =</span> <span class="st">"in"</span>)</span>
<span id="cb2-682"><a href="#cb2-682"></a>  })</span>
<span id="cb2-683"><a href="#cb2-683"></a></span>
<span id="cb2-684"><a href="#cb2-684"></a>all_markers <span class="ot">&lt;-</span> <span class="fu">FindAllMarkers</span>(seur_scaled, <span class="at">only.pos =</span> <span class="cn">TRUE</span>, <span class="at">min.pct =</span> <span class="fl">0.25</span>, <span class="at">logfc.threshold =</span> <span class="fl">0.25</span>)</span>
<span id="cb2-685"><a href="#cb2-685"></a>all_markers <span class="sc">%&gt;%</span></span>
<span id="cb2-686"><a href="#cb2-686"></a>    <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span></span>
<span id="cb2-687"><a href="#cb2-687"></a>    <span class="fu">slice_max</span>(<span class="at">n =</span> <span class="dv">20</span>, <span class="at">order_by =</span> avg_log2FC) <span class="sc">%&gt;%</span></span>
<span id="cb2-688"><a href="#cb2-688"></a>    <span class="fu">print</span>(<span class="at">n=</span><span class="cn">Inf</span>)</span>
<span id="cb2-689"><a href="#cb2-689"></a></span>
<span id="cb2-690"><a href="#cb2-690"></a><span class="fu">write_csv</span>(all_markers, <span class="st">"results/wholemuscle_sn_allmarkers.csv"</span>)</span>
<span id="cb2-691"><a href="#cb2-691"></a><span class="fu">saveRDS</span>(all_markers, <span class="st">"data/wholemuscle_sn_allmarkers.rds"</span>)</span>
<span id="cb2-692"><a href="#cb2-692"></a></span>
<span id="cb2-693"><a href="#cb2-693"></a>all_markers <span class="sc">%&gt;%</span></span>
<span id="cb2-694"><a href="#cb2-694"></a>  <span class="fu">filter</span>(cluster <span class="sc">==</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-695"><a href="#cb2-695"></a>  <span class="fu">select</span>(gene) <span class="sc">%&gt;%</span></span>
<span id="cb2-696"><a href="#cb2-696"></a>  <span class="fu">write_tsv</span>(<span class="st">"boops.tsv"</span>)</span>
<span id="cb2-697"><a href="#cb2-697"></a></span>
<span id="cb2-698"><a href="#cb2-698"></a><span class="fu">saveRDS</span>(seur_scaled, <span class="st">"data/23-04-20-wm_sn_master_seurat.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>